// useink/dist/chunk-USPRARIS.mjs
// src/chains/data/chaindata.ts
var Acala = {
  id: "acala",
  name: "Acala",
  account: "*25519",
  subscanUrl: "https://acala.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/acala_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/acala_metadata_latest.apng",
  rpcs: [
    "wss://acala-rpc-0.aca-api.network",
    "wss://acala-rpc-1.aca-api.network",
    "wss://acala-rpc-3.aca-api.network/ws",
    "wss://acala.polkawallet.io",
    "wss://acala-polkadot.api.onfinality.io/public-ws",
    "wss://acala-rpc.dwellir.com"
  ],
  paraId: 2e3,
  relay: { id: "polkadot" },
  balanceModuleConfigs: {
    "substrate-orml": { disable: true },
    "substrate-tokens": {
      tokens: [
        {
          symbol: "aUSD",
          decimals: 12,
          existentialDeposit: "100000000000",
          onChainId: '{"Token":"AUSD"}',
          coingeckoId: "acala-dollar"
        },
        {
          symbol: "TAP",
          decimals: 12,
          existentialDeposit: "1000000000000",
          onChainId: '{"Token":"TAP"}',
          coingeckoId: "tapio-protocol"
        },
        {
          symbol: "lcDOT",
          decimals: 10,
          existentialDeposit: "100000000",
          onChainId: '{"LiquidCrowdloan":13}',
          coingeckoId: "liquid-crowdloan-dot"
        },
        {
          symbol: "LDOT",
          decimals: 10,
          existentialDeposit: "500000000",
          onChainId: '{"Token":"LDOT"}',
          coingeckoId: "liquid-staking-dot"
        },
        {
          symbol: "DOT",
          decimals: 10,
          existentialDeposit: "100000000",
          onChainId: '{"Token":"DOT"}',
          coingeckoId: "polkadot"
        },
        {
          symbol: "EQD",
          decimals: 9,
          existentialDeposit: "1000000000",
          onChainId: '{"ForeignAsset":8}'
        },
        {
          symbol: "INTR",
          decimals: 10,
          existentialDeposit: "1000000000",
          onChainId: '{"ForeignAsset":4}',
          coingeckoId: "interlay"
        },
        {
          symbol: "WETH",
          decimals: 18,
          existentialDeposit: "500000000000000",
          onChainId: '{"ForeignAsset":6}'
        },
        {
          symbol: "ASTR",
          decimals: 18,
          existentialDeposit: "100000000000000000",
          onChainId: '{"ForeignAsset":2}',
          coingeckoId: "astar"
        },
        {
          symbol: "PHA",
          decimals: 12,
          existentialDeposit: "100000000000",
          onChainId: '{"ForeignAsset":9}',
          coingeckoId: "pha"
        },
        {
          symbol: "PARA",
          decimals: 12,
          existentialDeposit: "100000000000",
          onChainId: '{"ForeignAsset":1}'
        },
        {
          symbol: "GLMR",
          decimals: 18,
          existentialDeposit: "100000000000000000",
          onChainId: '{"ForeignAsset":0}',
          coingeckoId: "moonbeam"
        },
        {
          symbol: "WBTC",
          decimals: 8,
          existentialDeposit: "3000",
          onChainId: '{"ForeignAsset":5}'
        },
        {
          symbol: "EQ",
          decimals: 9,
          existentialDeposit: "1000000000",
          onChainId: '{"ForeignAsset":7}',
          coingeckoId: "equilibrium-token"
        },
        {
          symbol: "IBTC",
          decimals: 8,
          existentialDeposit: "100",
          onChainId: '{"ForeignAsset":3}',
          coingeckoId: "interbtc"
        },
        {
          symbol: "tDOT",
          decimals: 10,
          existentialDeposit: "100000000",
          onChainId: '{"StableAssetPoolToken":0}'
        },
        {
          symbol: "USDCet",
          decimals: 6,
          existentialDeposit: "10000",
          onChainId: '{"Erc20":"0x07df96d1341a7d16ba1ad431e2c847d978bc2bce"}'
        },
        {
          symbol: "APE",
          decimals: 18,
          existentialDeposit: "3000000000000000",
          onChainId: '{"Erc20":"0xf4c723e61709d90f89939c1852f516e373d418a8"}'
        },
        {
          symbol: "DAI",
          decimals: 18,
          existentialDeposit: "10000000000000000",
          onChainId: '{"Erc20":"0x54a37a01cd75b616d63e0ab665bffdb0143c52ae"}',
          coingeckoId: "dai"
        },
        {
          symbol: "WETH",
          decimals: 18,
          existentialDeposit: "8500000000000",
          onChainId: '{"Erc20":"0x5a4d6acdc4e3e5ab15717f407afe957f7a242578"}'
        },
        {
          symbol: "WBTC",
          decimals: 8,
          existentialDeposit: "60",
          onChainId: '{"Erc20":"0xc80084af223c8b598536178d9361dc55bfda6818"}'
        },
        {
          symbol: "lp AUSD-LDOT",
          decimals: 12,
          existentialDeposit: "100000000000",
          onChainId: '{"DexShare":[{"Token":"AUSD"},{"Token":"LDOT"}]}'
        },
        {
          symbol: "lp AUSD-IBTC",
          decimals: 12,
          existentialDeposit: "100000000000",
          onChainId: '{"DexShare":[{"Token":"AUSD"},{"ForeignAsset":3}]}'
        },
        {
          symbol: "lp AUSD-lcDOT",
          decimals: 12,
          existentialDeposit: "100000000000",
          onChainId: '{"DexShare":[{"Token":"AUSD"},{"LiquidCrowdloan":13}]}'
        },
        {
          symbol: "lp ACA-AUSD",
          decimals: 12,
          existentialDeposit: "100000000000",
          onChainId: '{"DexShare":[{"Token":"ACA"},{"Token":"AUSD"}]}'
        },
        {
          symbol: "lp DOT-lcDOT",
          decimals: 10,
          existentialDeposit: "100000000",
          onChainId: '{"DexShare":[{"Token":"DOT"},{"LiquidCrowdloan":13}]}'
        },
        {
          symbol: "lp AUSD-INTR",
          decimals: 12,
          existentialDeposit: "100000000000",
          onChainId: '{"DexShare":[{"Token":"AUSD"},{"ForeignAsset":4}]}'
        }
      ]
    }
  }
};
var Ajuna = {
  id: "ajuna",
  name: "Ajuna",
  account: "*25519",
  rpcs: [
    "wss://rpc-parachain.ajuna.network",
    "wss://ajuna.public.curie.radiumblock.co/ws"
  ],
  paraId: 2051,
  relay: { id: "polkadot" }
};
var Aleph = {
  id: "aleph",
  name: "Aleph Zero",
  coingeckoId: "aleph-zero",
  account: "*25519",
  subscanUrl: "https://alephzero.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/aleph-node_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/aleph-node_metadata_latest.apng",
  rpcs: ["wss://ws.azero.dev"]
};
var Altair = {
  id: "altair",
  name: "Altair",
  account: "*25519",
  subscanUrl: "https://altair.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/altair_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/altair_metadata_latest.apng",
  rpcs: [
    "wss://fullnode.altair.centrifuge.io",
    "wss://altair.api.onfinality.io/public-ws",
    "wss://altair-rpc.dwellir.com"
  ],
  paraId: 2088,
  relay: { id: "kusama" }
};
var Amplitude = {
  id: "amplitude",
  name: "Amplitude",
  account: "*25519",
  chainspecQrUrl: "https://metadata.novasama.io/qr/amplitude_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/amplitude_metadata_latest.apng",
  rpcs: [
    "wss://rpc-amplitude.pendulumchain.tech",
    "wss://amplitude-rpc.dwellir.com"
  ],
  paraId: 2124,
  relay: { id: "kusama" }
};
var AresOdyssey = {
  id: "ares-odyssey",
  name: "Ares Odyssey Standalone",
  account: "*25519",
  rpcs: ["wss://odyssey.aresprotocol.io"]
};
var Astar = {
  id: "astar",
  name: "Astar",
  account: "*25519",
  subscanUrl: "https://astar.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/astar_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/astar_metadata_latest.apng",
  rpcs: [
    "wss://rpc.astar.network",
    "wss://astar.public.blastapi.io",
    "wss://astar-rpc.dwellir.com",
    "wss://astar.api.onfinality.io/public-ws",
    "wss://astar.public.curie.radiumblock.co/ws",
    "wss://public-rpc.pinknode.io/astar",
    "wss://1rpc.io/astr"
  ],
  paraId: 2006,
  relay: { id: "polkadot" },
  balanceModuleConfigs: {
    "substrate-assets": {
      tokens: [
        { assetId: "4294969280", symbol: "USDT", coingeckoId: "tether" },
        {
          assetId: "18446744073709551616",
          symbol: "ACA",
          coingeckoId: "acala"
        },
        {
          assetId: "18446744073709551617",
          symbol: "AUSD",
          coingeckoId: "acala-dollar"
        },
        {
          assetId: "18446744073709551618",
          symbol: "LDOT",
          coingeckoId: "liquid-staking-dot"
        },
        {
          assetId: "18446744073709551619",
          symbol: "GLMR",
          coingeckoId: "moonbeam"
        },
        {
          assetId: "18446744073709551620",
          symbol: "IBTC",
          coingeckoId: "interbtc"
        },
        {
          assetId: "18446744073709551621",
          symbol: "INTR",
          coingeckoId: "interlay"
        },
        { assetId: "18446744073709551622", symbol: "PHA", coingeckoId: "pha" },
        {
          assetId: "18446744073709551623",
          symbol: "BNC",
          coingeckoId: "bifrost-native-coin"
        },
        { assetId: "18446744073709551624", symbol: "VDOT" },
        { assetId: "18446744073709551625", symbol: "CLV" },
        { assetId: "18446744073709551626", symbol: "VSDOT" },
        {
          assetId: "18446744073709551627",
          symbol: "RING",
          coingeckoId: "darwinia-network-native-token"
        },
        {
          assetId: "18446744073709551628",
          symbol: "EQ",
          coingeckoId: "equilibrium-token"
        },
        { assetId: "18446744073709551629", symbol: "EQD" },
        {
          assetId: "340282366920938463463374607431768211455",
          symbol: "DOT",
          coingeckoId: "polkadot"
        }
      ]
    }
  }
};
var Automata = {
  id: "automata",
  name: "Automata",
  account: "*25519",
  rpcs: ["wss://api.ata.network", "wss://automata.api.onfinality.io/public-ws"]
};
var Aventus = {
  id: "aventus",
  name: "Aventus",
  account: "*25519",
  rpcs: [],
  paraId: 2056,
  relay: { id: "polkadot" }
};
var Bajun = {
  id: "bajun",
  name: "Bajun",
  account: "*25519",
  chainspecQrUrl: "https://metadata.novasama.io/qr/bajun_specs.png",
  rpcs: [
    "wss://rpc-parachain.bajun.network",
    "wss://bajun-rpc.dwellir.com",
    "wss://bajun.api.onfinality.io/public-ws"
  ],
  paraId: 2119,
  relay: { id: "kusama" }
};
var Basilisk = {
  id: "basilisk",
  name: "Basilisk",
  account: "*25519",
  subscanUrl: "https://basilisk.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/basilisk_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/basilisk_metadata_latest.apng",
  rpcs: ["wss://rpc.basilisk.cloud", "wss://basilisk-rpc.dwellir.com"],
  paraId: 2090,
  relay: { id: "kusama" },
  balanceModuleConfigs: {
    "substrate-tokens": {
      tokens: [
        {
          symbol: "KSM",
          decimals: 12,
          existentialDeposit: "100000000",
          onChainId: 1,
          coingeckoId: "kusama"
        },
        {
          symbol: "aUSD",
          decimals: 12,
          existentialDeposit: "10000000000",
          onChainId: 2,
          coingeckoId: "acala-dollar"
        },
        {
          symbol: "TNKR",
          decimals: 12,
          existentialDeposit: "1000000000",
          onChainId: 6
        },
        {
          symbol: "USDCet",
          decimals: 6,
          existentialDeposit: "10000",
          onChainId: 9,
          coingeckoId: "usd-coin-wormhole-from-ethereum"
        },
        {
          symbol: "wETH",
          decimals: 18,
          existentialDeposit: "10000000",
          onChainId: 10,
          coingeckoId: "ethereum"
        },
        {
          symbol: "wBTC",
          decimals: 8,
          existentialDeposit: "1000000",
          onChainId: 11,
          coingeckoId: "bitcoin"
        },
        {
          symbol: "wUSDT",
          decimals: 6,
          existentialDeposit: "10000000000",
          onChainId: 12,
          coingeckoId: "tether"
        },
        {
          symbol: "DAI",
          decimals: 18,
          existentialDeposit: "10000000000",
          onChainId: 13,
          coingeckoId: "dai"
        },
        {
          symbol: "USDT",
          decimals: 6,
          existentialDeposit: "10000",
          onChainId: 14,
          coingeckoId: "tether"
        }
      ]
    }
  }
};
var BifrostKusama = {
  id: "bifrost-kusama",
  name: "Bifrost Kusama",
  account: "*25519",
  subscanUrl: "https://bifrost-kusama.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/bifrost_specs.png",
  rpcs: [
    "wss://bifrost-rpc.liebi.com/ws",
    "wss://bifrost-parachain.api.onfinality.io/public-ws",
    "wss://bifrost-rpc.dwellir.com"
  ],
  paraId: 2001,
  relay: { id: "kusama" },
  balanceModuleConfigs: {
    "substrate-tokens": {
      tokens: [
        {
          symbol: "aUSD",
          decimals: 12,
          existentialDeposit: "10000000000",
          onChainId: '{"Stable":"KUSD"}',
          coingeckoId: "acala-dollar"
        },
        {
          symbol: "USDT",
          decimals: 6,
          existentialDeposit: "10000",
          onChainId: '{"Token2":0}',
          coingeckoId: "tether"
        },
        {
          symbol: "vBNC",
          decimals: 12,
          existentialDeposit: "8000000000",
          onChainId: '{"VToken":"BNC"}'
        },
        {
          symbol: "vKSM",
          decimals: 12,
          existentialDeposit: "100000000",
          onChainId: '{"VToken":"KSM"}'
        },
        {
          symbol: "vsKSM",
          decimals: 12,
          existentialDeposit: "100000000",
          onChainId: '{"VSToken":"KSM"}'
        }
      ]
    }
  }
};
var BifrostPolkadot = {
  id: "bifrost-polkadot",
  name: "Bifrost Polkadot",
  account: "*25519",
  subscanUrl: "https://bifrost.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/bifrost_polkadot_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/bifrost_polkadot_metadata_latest.apng",
  rpcs: [
    "wss://hk.p.bifrost-rpc.liebi.com/ws",
    "wss://bifrost-polkadot.api.onfinality.io/public-ws"
  ],
  paraId: 2030,
  relay: { id: "polkadot" },
  balanceModuleConfigs: {
    "substrate-tokens": {
      tokens: [
        {
          symbol: "DOT",
          decimals: 10,
          existentialDeposit: "1000000",
          onChainId: '{"Token2":0}',
          coingeckoId: "polkadot"
        },
        {
          symbol: "GLMR",
          decimals: 18,
          existentialDeposit: "1000000000000",
          onChainId: '{"Token2":1}',
          coingeckoId: "moonbeam"
        },
        {
          symbol: "vDOT",
          decimals: 10,
          existentialDeposit: "100000000",
          onChainId: '{"VToken":"DOT"}'
        },
        {
          symbol: "vsDOT",
          decimals: 10,
          existentialDeposit: "100000000",
          onChainId: '{"VSToken":"DOT"}'
        }
      ]
    }
  }
};
var BitcountryPioneer = {
  id: "bitcountry-pioneer",
  name: "Bit.Country Pioneer",
  account: "*25519",
  subscanUrl: "https://pioneer.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/pioneer-runtime_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/pioneer-runtime_metadata_latest.apng",
  rpcs: ["wss://pioneer.api.onfinality.io/public-ws"],
  paraId: 2096,
  relay: { id: "kusama" },
  balanceModuleConfigs: {
    "substrate-tokens": {
      tokens: [
        {
          symbol: "KSM",
          decimals: 12,
          existentialDeposit: "0",
          onChainId: '{"NativeToken":1}',
          coingeckoId: "kusama"
        },
        {
          symbol: "KAR",
          decimals: 12,
          existentialDeposit: "0",
          onChainId: '{"NativeToken":2}',
          coingeckoId: "karura"
        },
        {
          symbol: "aUSD",
          decimals: 12,
          existentialDeposit: "0",
          onChainId: '{"Stable":0}',
          coingeckoId: "acala-dollar"
        },
        {
          symbol: "BIT",
          decimals: 18,
          existentialDeposit: "0",
          onChainId: '{"MiningResource":0}',
          coingeckoId: "bit"
        }
      ]
    }
  }
};
var Bitgreen = {
  id: "bitgreen",
  name: "Bitgreen",
  account: "*25519",
  rpcs: ["wss://mainnet.bitgreen.org"],
  paraId: 2048,
  relay: { id: "polkadot" }
};
var Bittensor = {
  id: "bittensor",
  name: "Bittensor",
  account: "*25519",
  rpcs: ["wss://entrypoint-finney.opentensor.ai"]
};
var Calamari = {
  id: "calamari",
  name: "Calamari",
  account: "*25519",
  subscanUrl: "https://calamari.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/calamari_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/calamari_metadata_latest.apng",
  rpcs: ["wss://ws.calamari.systems/"],
  paraId: 2084,
  relay: { id: "kusama" }
};
var CentrifugePolkadot = {
  id: "centrifuge-polkadot",
  name: "Centrifuge",
  account: "*25519",
  subscanUrl: "https://centrifuge.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/centrifuge_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/centrifuge_metadata_latest.apng",
  rpcs: [
    "wss://fullnode.parachain.centrifuge.io",
    "wss://centrifuge-parachain.api.onfinality.io/public-ws",
    "wss://centrifuge-rpc.dwellir.com"
  ],
  paraId: 2031,
  relay: { id: "polkadot" }
};
var Chainx = {
  id: "chainx",
  name: "ChainX",
  account: "*25519",
  subscanUrl: "https://chainx.subscan.io/",
  rpcs: ["wss://mainnet.chainx.org/ws"]
};
var Clover = {
  id: "clover",
  name: "Clover",
  account: "*25519",
  subscanUrl: "https://clover.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/clover-mainnet_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/clover-mainnet_metadata_latest.apng",
  rpcs: ["wss://rpc-para.clover.finance"],
  paraId: 2002,
  relay: { id: "polkadot" }
};
var Coinversation = {
  id: "coinversation",
  name: "Coinversation",
  account: "*25519",
  rpcs: [],
  paraId: 2027,
  relay: { id: "polkadot" }
};
var CompetitorsClub = {
  id: "competitors-club",
  name: "Competitors Club",
  account: "*25519",
  rpcs: []
};
var ComposableFinance = {
  id: "composable-finance",
  name: "Composable Finance",
  account: "*25519",
  subscanUrl: "https://composable.subscan.io/",
  rpcs: ["wss://rpc.composable.finance"],
  paraId: 2019,
  relay: { id: "polkadot" },
  balanceModuleConfigs: {
    "substrate-orml": { disable: true },
    "substrate-tokens": {
      tokens: [
        {
          symbol: "DOT",
          decimals: 10,
          existentialDeposit: "0",
          onChainId: 6,
          coingeckoId: "polkadot"
        },
        {
          symbol: "KSM",
          decimals: 12,
          existentialDeposit: "0",
          onChainId: 4,
          coingeckoId: "kusama"
        },
        {
          symbol: "KAR",
          decimals: 12,
          existentialDeposit: "0",
          onChainId: 101,
          coingeckoId: "karura"
        },
        {
          symbol: "BNC",
          decimals: 12,
          existentialDeposit: "0",
          onChainId: 102,
          coingeckoId: "bifrost-native-coin"
        },
        {
          symbol: "vKSM",
          decimals: 12,
          existentialDeposit: "0",
          onChainId: 103
        },
        {
          symbol: "MOVR",
          decimals: 18,
          existentialDeposit: "0",
          onChainId: 104,
          coingeckoId: "moonriver"
        },
        {
          symbol: "aUSD",
          decimals: 12,
          existentialDeposit: "0",
          onChainId: 129,
          coingeckoId: "acala-dollar"
        },
        {
          symbol: "USDT",
          decimals: 6,
          existentialDeposit: "0",
          onChainId: 130,
          coingeckoId: "tether"
        },
        {
          symbol: "USDC",
          decimals: 6,
          existentialDeposit: "0",
          onChainId: 131,
          coingeckoId: "usd-coin"
        },
        {
          symbol: "wBTC",
          decimals: 8,
          existentialDeposit: "0",
          onChainId: 132,
          coingeckoId: "bitcoin"
        },
        {
          symbol: "wETH",
          decimals: 18,
          existentialDeposit: "0",
          onChainId: 133,
          coingeckoId: "ethereum"
        }
      ]
    }
  }
};
var CrabKusama = {
  id: "crab-kusama",
  name: "Darwinia Crab",
  account: "*25519",
  subscanUrl: "https://crab-parachain.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/crab%20parachain_specs.png",
  rpcs: ["wss://crab-parachain-rpc.darwinia.network/"],
  paraId: 2105,
  relay: { id: "kusama" }
};
var CrabStandalone = {
  id: "crab-standalone",
  name: "Darwinia Crab Standalone",
  account: "*25519",
  subscanUrl: "https://crab.subscan.io/",
  rpcs: [
    "wss://crab-rpc.darwinia.network",
    "wss://darwiniacrab-rpc.dwellir.com",
    "wss://crab.api.onfinality.io/public-ws"
  ]
};
var Creditcoin = {
  id: "creditcoin",
  name: "Creditcoin",
  account: "*25519",
  rpcs: ["wss://rpc.mainnet.creditcoin.network/ws"]
};
var CrownSterling = {
  id: "crown-sterling",
  name: "Crown Sterling",
  account: "*25519",
  rpcs: ["wss://blockchain.crownsterling.io"]
};
var Crust = {
  id: "crust",
  name: "Crust Standalone",
  account: "*25519",
  subscanUrl: "https://crust.subscan.io/",
  rpcs: [
    "wss://rpc.crust.network",
    "wss://crust.api.onfinality.io/public-ws",
    "wss://rpc-crust-mainnet.decoo.io"
  ]
};
var CrustParachain = {
  id: "crust-parachain",
  name: "Crust",
  account: "*25519",
  rpcs: ["wss://crust-parachain.crustapps.net"],
  paraId: 2008,
  relay: { id: "polkadot" }
};
var Custom = {
  id: "custom",
  name: "Custom",
  account: "*25519",
  rpcs: []
};
var DarwiniaPolkadot = {
  id: "darwinia-polkadot",
  name: "Darwinia",
  account: "*25519",
  subscanUrl: "https://darwinia.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/darwinia%20parachain_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/darwinia%20parachain_metadata_latest.apng",
  rpcs: ["wss://parachain-rpc.darwinia.network"],
  paraId: 2046,
  relay: { id: "polkadot" }
};
var DarwiniaPolkadot2 = {
  id: "darwinia-polkadot-2",
  name: "Darwinia Backup",
  account: "*25519",
  rpcs: [],
  paraId: 2003,
  relay: { id: "polkadot" }
};
var DarwiniaStandalone = {
  id: "darwinia-standalone",
  name: "Darwinia Standalone",
  account: "*25519",
  rpcs: ["wss://rpc.darwinia.network", "wss://darwinia-rpc.dwellir.com"]
};
var Debio = {
  id: "debio",
  name: "DeBio",
  account: "*25519",
  rpcs: ["wss://ws-rpc.debio.network"]
};
var DockPosMainnet = {
  id: "dock-pos-mainnet",
  name: "Dock",
  account: "*25519",
  subscanUrl: "https://dock.subscan.io/",
  rpcs: ["wss://mainnet-node.dock.io"]
};
var Dorafactory = {
  id: "dorafactory",
  name: "Dora Factory",
  account: "*25519",
  chainspecQrUrl: "https://metadata.novasama.io/qr/dora%20ksm%20parachain_metadata_latest.apng",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/dora%20ksm%20parachain_metadata_latest.apng",
  rpcs: ["wss://kusama.dorafactory.org"],
  paraId: 2115,
  relay: { id: "kusama" }
};
var Edgeware = {
  id: "edgeware",
  name: "Edgeware",
  account: "*25519",
  subscanUrl: "https://edgeware.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/edgeware_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/edgeware_metadata_latest.apng",
  rpcs: [
    "wss://edgeware.api.onfinality.io/public-ws",
    "wss://mainnet2.edgewa.re",
    "wss://edgeware.jelliedowl.net",
    "wss://mainnet3.edgewa.re",
    "wss://mainnet4.edgewa.re",
    "wss://mainnet5.edgewa.re",
    "wss://edgeware-rpc.dwellir.com"
  ]
};
var EfinityPolkadot = {
  id: "efinity-polkadot",
  name: "Efinity",
  account: "Sr25519",
  subscanUrl: "https://efinity.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/efinity_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/efinity_metadata_latest.apng",
  rpcs: [
    "wss://rpc.efinity.io",
    "wss://efinity-rpc.dwellir.com",
    "wss://efinity.api.onfinality.io/public-ws"
  ],
  paraId: 2021,
  relay: { id: "polkadot" }
};
var EfinityStandalone = {
  id: "efinity-standalone",
  name: "Efinity Standalone",
  account: "*25519",
  rpcs: []
};
var Encointer = {
  id: "encointer",
  name: "Encointer",
  account: "*25519",
  subscanUrl: "https://encointer.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/encointer-parachain_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/encointer-parachain_metadata_latest.apng",
  rpcs: [
    "wss://kusama.api.encointer.org",
    "wss://encointer.api.onfinality.io/public-ws"
  ],
  paraId: 1001,
  relay: { id: "kusama" }
};
var EquilibriumPolkadot = {
  id: "equilibrium-polkadot",
  name: "Equilibrium",
  account: "*25519",
  subscanUrl: "https://equilibrium.subscan.io/",
  overrideNativeTokenId: "equilibrium-polkadot-substrate-equilibrium-eq",
  rpcs: ["wss://node.pol.equilibrium.io/", "wss://equilibrium-rpc.dwellir.com"],
  paraId: 2011,
  relay: { id: "polkadot" },
  balanceModuleConfigs: {
    "substrate-native": { disable: true },
    "substrate-equilibrium": { disable: false }
  }
};
var Frequency = {
  id: "frequency",
  name: "Frequency",
  account: "*25519",
  rpcs: ["wss://0.rpc.frequency.xyz", "wss://1.rpc.frequency.xyz"],
  paraId: 2091,
  relay: { id: "polkadot" }
};
var Geminis = {
  id: "geminis",
  name: "Geminis",
  account: "*25519",
  rpcs: ["wss://rpc.geminis.network"],
  paraId: 2038,
  relay: { id: "polkadot" }
};
var GenshiroKusama = {
  id: "genshiro-kusama",
  name: "Genshiro",
  account: "*25519",
  rpcs: ["wss://node.genshiro.io"],
  paraId: 2024,
  relay: { id: "kusama" },
  balanceModuleConfigs: {
    "substrate-native": { disable: true },
    "substrate-equilibrium": { disable: false }
  }
};
var Gm = {
  id: "gm",
  name: "GM",
  account: "*25519",
  chainspecQrUrl: "https://metadata.novasama.io/qr/gm_parachain_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/gm_parachain_metadata_latest.apng",
  rpcs: [
    "wss://kusama.gmordie.com",
    "wss://ws.gm.bldnodes.org",
    "wss://ws-node-gm.terrabiodao.org",
    "wss://leemo.gmordie.com",
    "wss://intern.gmordie.com"
  ],
  paraId: 2123,
  relay: { id: "kusama" },
  balanceModuleConfigs: {
    "substrate-tokens": {
      tokens: [
        { symbol: "GM", decimals: 0, existentialDeposit: "0", onChainId: 1 },
        { symbol: "GN", decimals: 0, existentialDeposit: "0", onChainId: 2 }
      ]
    }
  }
};
var Hanonycash = {
  id: "hanonycash",
  name: "Hanonycash",
  account: "*25519",
  rpcs: []
};
var Hashed = {
  id: "hashed",
  name: "Hashed",
  account: "*25519",
  rpcs: ["wss://c1.hashed.network"],
  paraId: 2093,
  relay: { id: "polkadot" }
};
var HeikoKusama = {
  id: "heiko-kusama",
  name: "Parallel Heiko",
  account: "*25519",
  subscanUrl: "https://parallel-heiko.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/heiko_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/heiko_metadata_latest.apng",
  rpcs: [
    "wss://parallel-heiko.api.onfinality.io/public-ws",
    "wss://heiko-rpc.parallel.fi"
  ],
  paraId: 2085,
  relay: { id: "kusama" }
};
var Hydra = {
  id: "hydra",
  name: "HydraDX",
  account: "*25519",
  subscanUrl: "https://hydradx.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/hydradx_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/hydradx_metadata_latest.apng",
  rpcs: [
    "wss://rpc.hydradx.cloud",
    "wss://hydradx.api.onfinality.io/public-ws",
    "wss://hydradx-rpc.dwellir.com"
  ],
  paraId: 2034,
  relay: { id: "polkadot" },
  balanceModuleConfigs: {
    "substrate-tokens": {
      tokens: [
        {
          symbol: "LRNA",
          decimals: 12,
          existentialDeposit: "400000000",
          onChainId: 1
        },
        {
          symbol: "DAI",
          decimals: 18,
          existentialDeposit: "10000000000000000",
          onChainId: 2,
          coingeckoId: "dai"
        },
        {
          symbol: "WBTC",
          decimals: 8,
          existentialDeposit: "44",
          onChainId: 3,
          coingeckoId: "bitcoin"
        },
        {
          symbol: "WETH",
          decimals: 18,
          existentialDeposit: "7000000000000",
          onChainId: 4,
          coingeckoId: "ethereum"
        },
        {
          symbol: "DOT",
          decimals: 10,
          existentialDeposit: "17540000",
          onChainId: 5,
          coingeckoId: "polkadot"
        },
        {
          symbol: "APE",
          decimals: 18,
          existentialDeposit: "2518891687657430",
          onChainId: 6,
          coingeckoId: "apecoin"
        },
        {
          symbol: "USDC",
          decimals: 6,
          existentialDeposit: "10000",
          onChainId: 7,
          coingeckoId: "usd-coin"
        }
      ]
    }
  }
};
var Imbue = {
  id: "imbue",
  name: "Imbue",
  account: "*25519",
  chainspecQrUrl: "https://metadata.novasama.io/qr/imbue_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/imbue_metadata_latest.apng",
  rpcs: ["wss://imbue-kusama.imbue.network"],
  paraId: 2121,
  relay: { id: "kusama" }
};
var IntegriteeKusama = {
  id: "integritee-kusama",
  name: "Integritee",
  account: "*25519",
  subscanUrl: "https://integritee.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/integritee-parachain_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/integritee-parachain_metadata_latest.apng",
  rpcs: [
    "wss://kusama.api.integritee.network",
    "wss://integritee-kusama.api.onfinality.io/public-ws",
    "wss://integritee-ksm-rpc.dwellir.com"
  ],
  paraId: 2015,
  relay: { id: "kusama" }
};
var IntegriteePolkadot = {
  id: "integritee-polkadot",
  name: "Integritee Shell",
  account: "*25519",
  rpcs: ["wss://polkadot.api.integritee.network"],
  paraId: 2039,
  relay: { id: "polkadot" }
};
var Interlay = {
  id: "interlay",
  name: "Interlay",
  account: "*25519",
  subscanUrl: "https://interlay.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/interlay-parachain_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/interlay-parachain_metadata_latest.apng",
  overrideNativeTokenId: "interlay-substrate-orml-intr",
  rpcs: [
    "wss://api.interlay.io/parachain",
    "wss://interlay.api.onfinality.io/public-ws"
  ],
  paraId: 2032,
  relay: { id: "polkadot" }
};
var Ipci = {
  id: "ipci",
  name: "DAO IPCI",
  account: "*25519",
  rpcs: ["wss://kusama.rpc.ipci.io"],
  chainspecQrUrl: "https://metadata.novasama.io/qr/ipci_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/ipci_metadata_latest.apng",
  paraId: 2222,
  relay: { id: "kusama" }
};
var Joystream = {
  id: "joystream",
  name: "Joystream",
  account: "*25519",
  rpcs: ["wss://rpc.joystream.org"]
};
var Kabocha = {
  id: "kabocha",
  name: "Kabocha",
  account: "*25519",
  chainspecQrUrl: "https://metadata.novasama.io/qr/kabocha-parachain_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/kabocha-parachain_metadata_latest.apng",
  rpcs: ["wss://kabocha.jelliedowl.net"],
  paraId: 2113,
  relay: { id: "kusama" }
};
var Kapex = {
  id: "kapex",
  name: "Kapex",
  account: "*25519",
  chainspecQrUrl: "https://metadata.novasama.io/qr/totem-parachain_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/totem-parachain_metadata_latest.apng",
  rpcs: ["wss://k-ui.kapex.network"],
  paraId: 2007,
  relay: { id: "polkadot" }
};
var Karura = {
  id: "karura",
  name: "Karura",
  account: "*25519",
  subscanUrl: "https://karura.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/karura_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/karura_metadata_latest.apng",
  rpcs: [
    "wss://karura-rpc-0.aca-api.network",
    "wss://karura-rpc-1.aca-api.network",
    "wss://karura-rpc-2.aca-api.network/ws",
    "wss://karura-rpc-3.aca-api.network/ws",
    "wss://karura.polkawallet.io",
    "wss://karura.api.onfinality.io/public-ws",
    "wss://karura-rpc.dwellir.com"
  ],
  paraId: 2e3,
  relay: { id: "kusama" },
  balanceModuleConfigs: {
    "substrate-orml": { disable: true },
    "substrate-tokens": {
      tokens: [
        {
          symbol: "LKSM",
          decimals: 12,
          existentialDeposit: "500000000",
          onChainId: '{"Token":"LKSM"}',
          coingeckoId: "liquid-ksm"
        },
        {
          symbol: "BNC",
          decimals: 12,
          existentialDeposit: "8000000000",
          onChainId: '{"Token":"BNC"}',
          coingeckoId: "bifrost-native-coin"
        },
        {
          symbol: "TAI",
          decimals: 12,
          existentialDeposit: "1000000000000",
          onChainId: '{"Token":"TAI"}',
          coingeckoId: "taiga"
        },
        {
          symbol: "PHA",
          decimals: 12,
          existentialDeposit: "40000000000",
          onChainId: '{"Token":"PHA"}',
          coingeckoId: "pha"
        },
        {
          symbol: "KINT",
          decimals: 12,
          existentialDeposit: "133330000",
          onChainId: '{"Token":"KINT"}',
          coingeckoId: "kintsugi"
        },
        {
          symbol: "VSKSM",
          decimals: 12,
          existentialDeposit: "100000000",
          onChainId: '{"Token":"VSKSM"}'
        },
        {
          symbol: "KSM",
          decimals: 12,
          existentialDeposit: "100000000",
          onChainId: '{"Token":"KSM"}',
          coingeckoId: "kusama"
        },
        {
          symbol: "aUSD",
          decimals: 12,
          existentialDeposit: "10000000000",
          onChainId: '{"Token":"KUSD"}',
          coingeckoId: "acala-dollar"
        },
        {
          symbol: "KBTC",
          decimals: 8,
          existentialDeposit: "66",
          onChainId: '{"Token":"KBTC"}',
          coingeckoId: "kintsugi-btc"
        },
        {
          symbol: "KMA",
          decimals: 12,
          existentialDeposit: "100000000000",
          onChainId: '{"ForeignAsset":10}',
          coingeckoId: "calamari-network"
        },
        {
          symbol: "GENS",
          decimals: 9,
          existentialDeposit: "1000000000000",
          onChainId: '{"ForeignAsset":14}',
          coingeckoId: "genshiro"
        },
        {
          symbol: "LT",
          decimals: 12,
          existentialDeposit: "1000000000000",
          onChainId: '{"ForeignAsset":19}'
        },
        {
          symbol: "TEER",
          decimals: 12,
          existentialDeposit: "100000000000",
          onChainId: '{"ForeignAsset":8}',
          coingeckoId: "integritee"
        },
        {
          symbol: "TUR",
          decimals: 10,
          existentialDeposit: "40000000000",
          onChainId: '{"ForeignAsset":16}'
        },
        {
          symbol: "HKO",
          decimals: 12,
          existentialDeposit: "100000000000",
          onChainId: '{"ForeignAsset":4}'
        },
        {
          symbol: "KICO",
          decimals: 14,
          existentialDeposit: "100000000000000",
          onChainId: '{"ForeignAsset":6}'
        },
        {
          symbol: "QTZ",
          decimals: 18,
          existentialDeposit: "1000000000000000000",
          onChainId: '{"ForeignAsset":2}',
          coingeckoId: "quartz"
        },
        {
          symbol: "LIT",
          decimals: 12,
          existentialDeposit: "100000000000",
          onChainId: '{"ForeignAsset":20}',
          coingeckoId: "litentry"
        },
        {
          symbol: "NEER",
          decimals: 18,
          existentialDeposit: "100000000000000000",
          onChainId: '{"ForeignAsset":9}',
          coingeckoId: "metaverse-network-pioneer"
        },
        {
          symbol: "ARIS",
          decimals: 8,
          existentialDeposit: "1000000000000",
          onChainId: '{"ForeignAsset":1}'
        },
        {
          symbol: "RMRK",
          decimals: 10,
          existentialDeposit: "100000000",
          onChainId: '{"ForeignAsset":0}',
          coingeckoId: "rmrk"
        },
        {
          symbol: "CSM",
          decimals: 12,
          existentialDeposit: "1000000000000",
          onChainId: '{"ForeignAsset":5}',
          coingeckoId: "crust-storage-market"
        },
        {
          symbol: "PCHU",
          decimals: 18,
          existentialDeposit: "100000000000000000",
          onChainId: '{"ForeignAsset":17}'
        },
        {
          symbol: "CRAB",
          decimals: 18,
          existentialDeposit: "1000000000000000000",
          onChainId: '{"ForeignAsset":13}'
        },
        {
          symbol: "BSX",
          decimals: 12,
          existentialDeposit: "1000000000000",
          onChainId: '{"ForeignAsset":11}',
          coingeckoId: "basilisk"
        },
        {
          symbol: "AIR",
          decimals: 18,
          existentialDeposit: "100000000000000000",
          onChainId: '{"ForeignAsset":12}',
          coingeckoId: "altair"
        },
        {
          symbol: "USDT",
          decimals: 6,
          existentialDeposit: "10000",
          onChainId: '{"ForeignAsset":7}',
          coingeckoId: "tether"
        },
        {
          symbol: "SDN",
          decimals: 18,
          existentialDeposit: "10000000000000000",
          onChainId: '{"ForeignAsset":18}',
          coingeckoId: "shiden"
        },
        {
          symbol: "EQD",
          decimals: 9,
          existentialDeposit: "10000000000",
          onChainId: '{"ForeignAsset":15}'
        },
        {
          symbol: "MOVR",
          decimals: 18,
          existentialDeposit: "1000000000000000",
          onChainId: '{"ForeignAsset":3}',
          coingeckoId: "moonriver"
        },
        {
          symbol: "taiKSM",
          decimals: 12,
          existentialDeposit: "100000000",
          onChainId: '{"StableAssetPoolToken":0}'
        },
        {
          symbol: "3USD",
          decimals: 12,
          existentialDeposit: "10000000000",
          onChainId: '{"StableAssetPoolToken":1}'
        },
        {
          symbol: "USDCet",
          decimals: 6,
          existentialDeposit: "10000",
          onChainId: '{"Erc20":"0x1f3a10587a20114ea25ba1b388ee2dd4a337ce27"}'
        },
        {
          symbol: "DAI",
          decimals: 18,
          existentialDeposit: "10000000000000000",
          onChainId: '{"Erc20":"0x4bb6afb5fa2b07a5d1c499e1c3ddb5a15e709a71"}'
        },
        {
          symbol: "waUSD",
          decimals: 12,
          existentialDeposit: "100000000000",
          onChainId: '{"Erc20":"0xe20683ad1ed8bbeed7e1ae74be10f19d8045b530"}'
        },
        {
          symbol: "lp KSM-RMRK",
          decimals: 12,
          existentialDeposit: "100000000",
          onChainId: '{"DexShare":[{"Token":"KSM"},{"ForeignAsset":0}]}'
        },
        {
          symbol: "lp KSM-LKSM",
          decimals: 12,
          existentialDeposit: "100000000",
          onChainId: '{"DexShare":[{"Token":"KSM"},{"Token":"LKSM"}]}'
        },
        {
          symbol: "lp KAR-KUSD",
          decimals: 12,
          existentialDeposit: "100000000000",
          onChainId: '{"DexShare":[{"Token":"KAR"},{"Token":"KUSD"}]}'
        },
        {
          symbol: "lp KAR-QTZ",
          decimals: 12,
          existentialDeposit: "100000000000",
          onChainId: '{"DexShare":[{"Token":"KAR"},{"ForeignAsset":2}]}'
        },
        {
          symbol: "lp KUSD-KSM",
          decimals: 12,
          existentialDeposit: "10000000000",
          onChainId: '{"DexShare":[{"Token":"KUSD"},{"Token":"KSM"}]}'
        },
        {
          symbol: "lp KUSD-BNC",
          decimals: 12,
          existentialDeposit: "10000000000",
          onChainId: '{"DexShare":[{"Token":"KUSD"},{"Token":"BNC"}]}'
        },
        {
          symbol: "lp KUSD-CSM",
          decimals: 12,
          existentialDeposit: "10000000000",
          onChainId: '{"DexShare":[{"Token":"KUSD"},{"ForeignAsset":5}]}'
        },
        {
          symbol: "lp KUSD-RMRK",
          decimals: 12,
          existentialDeposit: "10000000000",
          onChainId: '{"DexShare":[{"Token":"KUSD"},{"ForeignAsset":0}]}'
        },
        {
          symbol: "lp KSM-ARIS",
          decimals: 12,
          existentialDeposit: "100000000",
          onChainId: '{"DexShare":[{"Token":"KSM"},{"ForeignAsset":1}]}'
        },
        {
          symbol: "lp KUSD-KINT",
          decimals: 12,
          existentialDeposit: "10000000000",
          onChainId: '{"DexShare":[{"Token":"KUSD"},{"Token":"KINT"}]}'
        },
        {
          symbol: "lp TAI-taiKSM",
          decimals: 12,
          existentialDeposit: "1000000000000",
          onChainId: '{"DexShare":[{"Token":"TAI"},{"StableAssetPoolToken":0}]}'
        },
        {
          symbol: "lp KAR-KSM",
          decimals: 12,
          existentialDeposit: "100000000000",
          onChainId: '{"DexShare":[{"Token":"KAR"},{"Token":"KSM"}]}'
        },
        {
          symbol: "lp KUSD-AIR",
          decimals: 12,
          existentialDeposit: "10000000000",
          onChainId: '{"DexShare":[{"Token":"KUSD"},{"ForeignAsset":12}]}'
        },
        {
          symbol: "lp KAR-LKSM",
          decimals: 12,
          existentialDeposit: "100000000000",
          onChainId: '{"DexShare":[{"Token":"KAR"},{"Token":"LKSM"}]}'
        },
        {
          symbol: "lp KUSD-QTZ",
          decimals: 12,
          existentialDeposit: "10000000000",
          onChainId: '{"DexShare":[{"Token":"KUSD"},{"ForeignAsset":2}]}'
        },
        {
          symbol: "lp KUSD-PHA",
          decimals: 12,
          existentialDeposit: "10000000000",
          onChainId: '{"DexShare":[{"Token":"KUSD"},{"Token":"PHA"}]}'
        },
        {
          symbol: "lp KUSD-LKSM",
          decimals: 12,
          existentialDeposit: "10000000000",
          onChainId: '{"DexShare":[{"Token":"KUSD"},{"Token":"LKSM"}]}'
        },
        {
          symbol: "lp KUSD-KBTC",
          decimals: 12,
          existentialDeposit: "10000000000",
          onChainId: '{"DexShare":[{"Token":"KUSD"},{"Token":"KBTC"}]}'
        }
      ]
    }
  }
};
var Khala = {
  id: "khala",
  name: "Khala",
  account: "*25519",
  subscanUrl: "https://khala.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/khala_specs.png",
  rpcs: [
    "wss://khala-api.phala.network/ws",
    "wss://khala.api.onfinality.io/public-ws",
    "wss://khala-rpc.dwellir.com",
    "wss://public-rpc.pinknode.io/khala"
  ],
  paraId: 2004,
  relay: { id: "kusama" }
};
var Kico = {
  id: "kico",
  name: "Kico",
  account: "*25519",
  rpcs: ["wss://rpc.kico.dico.io"],
  paraId: 2107,
  relay: { id: "kusama" }
};
var KiltSpiritnet = {
  id: "kilt-spiritnet",
  name: "KILT Spiritnet",
  account: "*25519",
  subscanUrl: "https://spiritnet.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/kilt-spiritnet_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/kilt-spiritnet_metadata_latest.apng",
  rpcs: [
    "wss://spiritnet.kilt.io/",
    "wss://spiritnet.api.onfinality.io/public-ws",
    "wss://kilt-rpc.dwellir.com"
  ],
  paraId: 2086,
  relay: { id: "polkadot" }
};
var Kintsugi = {
  id: "kintsugi",
  name: "Kintsugi",
  account: "*25519",
  subscanUrl: "https://kintsugi.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/kintsugi-parachain_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/kintsugi-parachain_metadata_latest.apng",
  overrideNativeTokenId: "kintsugi-substrate-orml-kint",
  rpcs: [
    "wss://api-kusama.interlay.io/parachain",
    "wss://kintsugi.api.onfinality.io/public-ws",
    "wss://kintsugi-rpc.dwellir.com"
  ],
  paraId: 2092,
  relay: { id: "kusama" }
};
var Kpron = {
  id: "kpron",
  name: "Kpron",
  account: "*25519",
  rpcs: ["wss://kusama-kpron-rpc.apron.network/"],
  paraId: 2019,
  relay: { id: "kusama" }
};
var Kulupu = {
  id: "kulupu",
  name: "Kulupu",
  account: "*25519",
  subscanUrl: "https://kulupu.subscan.io/",
  rpcs: ["wss://rpc.kulupu.corepaper.org/ws"]
};
var Kusama = {
  id: "kusama",
  name: "Kusama",
  account: "*25519",
  subscanUrl: "https://kusama.subscan.io/",
  chainspecQrUrl: "https://metadata.parity.io/qr/kusama_specs.png",
  latestMetadataQrUrl: "https://metadata.parity.io/qr/kusama_metadata_latest.apng",
  rpcs: [
    "wss://kusama-rpc.polkadot.io",
    "wss://kusama.api.onfinality.io/public-ws",
    "wss://kusama-rpc.dwellir.com",
    "wss://kusama-rpc-tn.dwellir.com",
    "wss://kusama.public.curie.radiumblock.co/ws",
    "wss://public-rpc.pinknode.io/kusama",
    "wss://1rpc.io/ksm",
    "wss://rpc.dotters.network/kusama"
  ]
};
var KusamaBridgeHub = {
  id: "kusama-bridge-hub",
  name: "BridgeHub",
  account: "*25519",
  rpcs: ["wss://kusama-bridge-hub-rpc.polkadot.io"],
  paraId: 1002,
  relay: { id: "kusama" }
};
var Kusari = {
  id: "kusari",
  name: "Kusari",
  account: "*25519",
  rpcs: ["wss://ws.kusari.network"]
};
var Kylin = {
  id: "kylin",
  name: "Kylin",
  account: "*25519",
  chainspecQrUrl: "https://metadata.novasama.io/qr/kylin_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/kylin_metadata_latest.apng",
  rpcs: ["wss://polkadot.kylin-node.co.uk"],
  paraId: 2052,
  relay: { id: "polkadot" }
};
var Listen = {
  id: "listen",
  name: "Listen",
  account: "*25519",
  rpcs: ["wss://rpc.mainnet.listen.io", "wss://wss.mainnet.listen.io"],
  paraId: 2118,
  relay: { id: "kusama" }
};
var Litentry = {
  id: "litentry",
  name: "Litentry",
  account: "*25519",
  chainspecQrUrl: "https://metadata.novasama.io/qr/litentry-parachain_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/litentry-parachain_metadata_latest.apng",
  rpcs: [
    "wss://rpc.litentry-parachain.litentry.io",
    "wss://litentry-rpc.dwellir.com"
  ],
  paraId: 2013,
  relay: { id: "polkadot" }
};
var Litmus = {
  id: "litmus",
  name: "Litmus",
  account: "*25519",
  subscanUrl: "https://litmus.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/litmus-parachain_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/litmus-parachain_metadata_latest.apng",
  rpcs: ["wss://rpc.litmus-parachain.litentry.io"],
  paraId: 2106,
  relay: { id: "kusama" }
};
var Logion = {
  id: "logion",
  name: "Logion",
  account: "*25519",
  rpcs: ["wss://rpc01.logion.network"]
};
var LoomNetwork = {
  id: "loom-network",
  name: "Loom",
  account: "*25519",
  rpcs: ["wss://kusama.dappchains.com"],
  paraId: 2080,
  relay: { id: "kusama" }
};
var Luhn = {
  id: "luhn",
  name: "Luhn",
  account: "*25519",
  rpcs: ["wss://c1.luhn.network"],
  paraId: 2232,
  relay: { id: "kusama" }
};
var Mangata = {
  id: "mangata",
  name: "MangataX",
  coingeckoId: null,
  account: "*25519",
  subscanUrl: "https://mangatax.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/mangata-parachain_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/mangata-parachain_metadata_latest.apng",
  overrideNativeTokenId: "mangata-substrate-tokens-mgx",
  rpcs: [
    "wss://prod-kusama-collator-01.mangatafinance.cloud",
    "wss://mangata-x.api.onfinality.io/public-ws"
  ],
  paraId: 2110,
  relay: { id: "kusama" },
  balanceModuleConfigs: {
    "substrate-tokens": {
      tokens: [
        { symbol: "MGX", decimals: 18, existentialDeposit: "0", onChainId: 0 }
      ]
    }
  }
};
var Manta = {
  id: "manta",
  name: "Manta",
  account: "*25519",
  rpcs: [],
  paraId: 2015,
  relay: { id: "polkadot" }
};
var Mars = {
  id: "mars",
  name: "Mars",
  account: "*25519",
  rpcs: ["wss://wss.mars.aresprotocol.io"],
  paraId: 2008,
  relay: { id: "kusama" }
};
var Mathchain = {
  id: "mathchain",
  name: "MathChain",
  account: "*25519",
  rpcs: [
    "wss://mathchain-asia.maiziqianbao.net/ws",
    "wss://mathchain-us.maiziqianbao.net/ws"
  ]
};
var Minix = {
  id: "minix",
  name: "MiniX",
  account: "*25519",
  rpcs: []
};
var Moonbeam = {
  id: "moonbeam",
  name: "Moonbeam",
  account: "secp256k1",
  subscanUrl: "https://moonbeam.subscan.io/",
  rpcs: [
    "wss://wss.api.moonbeam.network",
    "wss://moonbeam.public.blastapi.io",
    "wss://moonbeam.api.onfinality.io/public-ws",
    "wss://public-rpc.pinknode.io/moonbeam",
    "wss://1rpc.io/glmr"
  ],
  paraId: 2004,
  relay: { id: "polkadot" }
};
var Moonriver = {
  id: "moonriver",
  name: "Moonriver",
  account: "secp256k1",
  subscanUrl: "https://moonriver.subscan.io/",
  rpcs: [
    "wss://wss.api.moonriver.moonbeam.network",
    "wss://moonriver.public.blastapi.io",
    "wss://moonriver.api.onfinality.io/public-ws",
    "wss://public-rpc.pinknode.io/moonriver"
  ],
  paraId: 2023,
  relay: { id: "kusama" }
};
var Myriad = {
  id: "myriad",
  name: "Myriad",
  chainspecQrUrl: "https://metadata.novasama.io/qr/appchain_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/appchain_metadata_latest.apng",
  account: "*25519",
  rpcs: ["wss://ws-rpc.myriad.social"]
};
var Neatcoin = {
  id: "neatcoin",
  name: "Neatcoin",
  account: "*25519",
  rpcs: ["wss://rpc.neatcoin.org/ws"]
};
var Nftmart = {
  id: "nftmart",
  name: "NFTMart",
  account: "*25519",
  rpcs: ["wss://mainnet.nftmart.io/rpc/ws"]
};
var NodlePolkadot = {
  id: "nodle-polkadot",
  name: "Nodle",
  account: "*25519",
  subscanUrl: "https://nodle.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/nodle-para_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/nodle-para_metadata_latest.apng",
  rpcs: [
    "wss://nodle-parachain.api.onfinality.io/public-ws",
    "wss://eden-rpc.dwellir.com",
    "wss://public-rpc.pinknode.io/nodle"
  ],
  paraId: 2026,
  relay: { id: "polkadot" }
};
var Oak = {
  id: "oak",
  name: "Oak",
  account: "*25519",
  rpcs: ["wss://rpc.oak.tech"],
  paraId: 2090,
  relay: { id: "polkadot" }
};
var Odyssey = {
  id: "odyssey",
  name: "Ares Odyssey",
  account: "*25519",
  rpcs: ["wss://wss.odyssey.aresprotocol.io"],
  paraId: 2028,
  relay: { id: "polkadot" }
};
var Omnibtc = {
  id: "omnibtc",
  name: "OmniBTC",
  account: "*25519",
  rpcs: ["wss://psc-parachain.coming.chat"],
  paraId: 2053,
  relay: { id: "polkadot" }
};
var OrigintrailParachain = {
  id: "origintrail-parachain",
  name: "OriginTrail",
  account: "*25519",
  chainspecQrUrl: "https://metadata.novasama.io/qr/origintrail-parachain_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/origintrail-parachain_metadata_latest.apng",
  rpcs: ["wss://parachain-rpc.origin-trail.network"],
  paraId: 2043,
  relay: { id: "polkadot" }
};
var Parallel = {
  id: "parallel",
  name: "Parallel",
  account: "*25519",
  subscanUrl: "https://parallel.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/parallel_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/parallel_metadata_latest.apng",
  rpcs: ["wss://rpc.parallel.fi"],
  paraId: 2012,
  relay: { id: "polkadot" }
};
var Pendulum = {
  id: "pendulum",
  name: "Pendulum",
  account: "*25519",
  rpcs: ["wss://rpc-pendulum.prd.pendulumchain.tech"],
  paraId: 2094,
  relay: { id: "polkadot" }
};
var Phala = {
  id: "phala",
  name: "Phala",
  account: "*25519",
  subscanUrl: "https://phala.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/phala_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/phala_metadata_latest.apng",
  rpcs: [
    "wss://api.phala.network/ws",
    "wss://phala.api.onfinality.io/public-ws"
  ],
  paraId: 2035,
  relay: { id: "polkadot" }
};
var Picasso = {
  id: "picasso",
  name: "Picasso",
  account: "*25519",
  subscanUrl: "https://picasso.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/picasso_specs.png",
  rpcs: ["wss://rpc.composablenodes.tech"],
  paraId: 2087,
  relay: { id: "kusama" },
  balanceModuleConfigs: {
    "substrate-orml": { disable: true },
    "substrate-tokens": {
      tokens: [
        {
          symbol: "DOT",
          decimals: 10,
          existentialDeposit: "0",
          onChainId: 6,
          coingeckoId: "polkadot"
        },
        {
          symbol: "KSM",
          decimals: 12,
          existentialDeposit: "0",
          onChainId: 4,
          coingeckoId: "kusama"
        },
        {
          symbol: "KAR",
          decimals: 12,
          existentialDeposit: "0",
          onChainId: 101,
          coingeckoId: "karura"
        },
        {
          symbol: "BNC",
          decimals: 12,
          existentialDeposit: "0",
          onChainId: 102,
          coingeckoId: "bifrost-native-coin"
        },
        {
          symbol: "vKSM",
          decimals: 12,
          existentialDeposit: "0",
          onChainId: 103
        },
        {
          symbol: "MOVR",
          decimals: 18,
          existentialDeposit: "0",
          onChainId: 104,
          coingeckoId: "moonriver"
        },
        {
          symbol: "aUSD",
          decimals: 12,
          existentialDeposit: "0",
          onChainId: 129,
          coingeckoId: "acala-dollar"
        },
        {
          symbol: "USDT",
          decimals: 6,
          existentialDeposit: "0",
          onChainId: 130,
          coingeckoId: "tether"
        },
        {
          symbol: "USDC",
          decimals: 6,
          existentialDeposit: "0",
          onChainId: 131,
          coingeckoId: "usd-coin"
        },
        {
          symbol: "wBTC",
          decimals: 8,
          existentialDeposit: "0",
          onChainId: 132,
          coingeckoId: "bitcoin"
        },
        {
          symbol: "wETH",
          decimals: 18,
          existentialDeposit: "0",
          onChainId: 133,
          coingeckoId: "ethereum"
        }
      ]
    }
  }
};
var Pichiu = {
  id: "pichiu",
  name: "Pichiu",
  account: "*25519",
  chainspecQrUrl: "https://metadata.novasama.io/qr/pichiu_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/pichiu_metadata_latest.apng",
  rpcs: ["wss://kusama.kylin-node.co.uk"],
  paraId: 2102,
  relay: { id: "kusama" }
};
var PolkadexPolkadot = {
  id: "polkadex-polkadot",
  name: "Polkadex",
  account: "*25519",
  rpcs: [],
  paraId: 2040,
  relay: { id: "polkadot" }
};
var PolkadexStandalone = {
  id: "polkadex-standalone",
  name: "Polkadex Standalone",
  account: "*25519",
  subscanUrl: "https://polkadex.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/node_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/node_metadata_latest.apng",
  rpcs: [
    "wss://mainnet.polkadex.trade",
    "wss://polkadex.api.onfinality.io/public-ws"
  ]
};
var Polkadot = {
  id: "polkadot",
  name: "Polkadot",
  account: "*25519",
  subscanUrl: "https://polkadot.subscan.io/",
  chainspecQrUrl: "https://metadata.parity.io/qr/polkadot_specs.png",
  latestMetadataQrUrl: "https://metadata.parity.io/qr/polkadot_metadata_latest.apng",
  rpcs: [
    "wss://rpc.polkadot.io",
    "wss://polkadot.api.onfinality.io/public-ws",
    "wss://polkadot-rpc.dwellir.com",
    "wss://polkadot-rpc-tn.dwellir.com",
    "wss://public-rpc.pinknode.io/polkadot",
    "wss://polkadot.public.curie.radiumblock.co/ws",
    "wss://1rpc.io/dot",
    "wss://rpc.dotters.network/polkadot"
  ]
};
var PolkadotCollectives = {
  id: "polkadot-collectives",
  name: "Collectives",
  account: "*25519",
  rpcs: [
    "wss://polkadot-collectives-rpc.polkadot.io",
    "wss://collectives.api.onfinality.io/public-ws"
  ],
  paraId: 1001,
  relay: { id: "polkadot" }
};
var Polkasmith = {
  id: "polkasmith",
  name: "PolkaSmith",
  account: "*25519",
  rpcs: ["wss://wss-polkasmith.polkafoundry.com"],
  paraId: 2009,
  relay: { id: "kusama" }
};
var Polymesh = {
  id: "polymesh",
  name: "Polymesh",
  account: "*25519",
  subscanUrl: "https://polymesh.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/polymesh_mainnet_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/polymesh_mainnet_metadata_latest.apng",
  rpcs: ["wss://mainnet-rpc.polymesh.network"]
};
var Quartz = {
  id: "quartz",
  name: "Quartz",
  account: "*25519",
  subscanUrl: "https://quartz.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/quartz_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/quartz_metadata_latest.apng",
  rpcs: [
    "wss://quartz.api.onfinality.io/public-ws",
    "wss://us-ws-quartz.unique.network",
    "wss://asia-ws-quartz.unique.network",
    "wss://eu-ws-quartz.unique.network"
  ],
  paraId: 2095,
  relay: { id: "kusama" }
};
var Riochain = {
  id: "riochain",
  name: "RioChain",
  account: "*25519",
  rpcs: ["wss://node.v1.riochain.io"]
};
var Riodefi = {
  id: "riodefi",
  name: "RioDeFi",
  account: "*25519",
  rpcs: ["wss://rio-kusama.riocorenetwork.com"],
  paraId: 2227,
  relay: { id: "kusama" }
};
var RobonomicsKusama = {
  id: "robonomics-kusama",
  name: "Robonomics",
  account: "*25519",
  subscanUrl: "https://robonomics.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/robonomics_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/robonomics_metadata_latest.apng",
  rpcs: [
    "wss://kusama.rpc.robonomics.network/",
    "wss://robonomics.api.onfinality.io/public-ws",
    "wss://robonomics-rpc.dwellir.com",
    "wss://robonomics.0xsamsara.com",
    "wss://robonomics.leemo.me"
  ],
  paraId: 2048,
  relay: { id: "kusama" }
};
var Sakura = {
  id: "sakura",
  name: "Sakura",
  account: "*25519",
  subscanUrl: "https://sakura.subscan.io/",
  rpcs: ["wss://api-sakura.clover.finance"],
  paraId: 2016,
  relay: { id: "kusama" }
};
var ShadowKusama = {
  id: "shadow-kusama",
  name: "Crust Shadow",
  account: "*25519",
  subscanUrl: "https://shadow.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/crust-collator_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/crust-collator_metadata_latest.apng",
  rpcs: ["wss://rpc-shadow.crust.network/", "wss://rpc2-shadow.crust.network"],
  paraId: 2012,
  relay: { id: "kusama" }
};
var Sherpax = {
  id: "sherpax",
  name: "SherpaX",
  account: "*25519",
  rpcs: ["wss://mainnet.sherpax.io"]
};
var ShidenKusama = {
  id: "shiden-kusama",
  name: "Shiden",
  account: "*25519",
  subscanUrl: "https://shiden.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/shiden_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/shiden_metadata_latest.apng",
  rpcs: [
    "wss://rpc.shiden.astar.network",
    "wss://shiden.public.blastapi.io",
    "wss://shiden-rpc.dwellir.com",
    "wss://shiden.api.onfinality.io/public-ws",
    "wss://public-rpc.pinknode.io/shiden"
  ],
  paraId: 2007,
  relay: { id: "kusama" }
};
var Snow = {
  id: "snow",
  name: "Snow",
  account: "*25519",
  chainspecQrUrl: "https://metadata.novasama.io/qr/snow_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/snow_metadata_latest.apng",
  rpcs: ["wss://snow-rpc.icenetwork.io"],
  paraId: 2129,
  relay: { id: "kusama" }
};
var SoraKsm = {
  id: "sora-ksm",
  name: "Sora",
  account: "*25519",
  rpcs: ["wss://ws.parachain-collator-1.c1.sora2.soramitsu.co.jp"],
  paraId: 2011,
  relay: { id: "kusama" }
};
var SoraSubstrate = {
  id: "sora-substrate",
  name: "Sora Standalone",
  account: "*25519",
  subscanUrl: "https://sora.subscan.io/",
  rpcs: [
    "wss://mof2.sora.org",
    "wss://ws.mof.sora.org",
    "wss://mof3.sora.org",
    "wss://sora.api.onfinality.io/public-ws"
  ]
};
var Spanner = {
  id: "spanner",
  name: "Spanner",
  account: "*25519",
  rpcs: []
};
var Stafi = {
  id: "stafi",
  name: "Stafi",
  account: "*25519",
  subscanUrl: "https://stafi.subscan.io/",
  rpcs: ["wss://mainnet-rpc.stafi.io"]
};
var Statemine = {
  id: "statemine",
  name: "Statemine",
  account: "*25519",
  subscanUrl: "https://statemine.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/statemine_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/statemine_metadata_latest.apng",
  rpcs: [
    "wss://statemine-rpc.polkadot.io",
    "wss://statemine.api.onfinality.io/public-ws",
    "wss://statemine-rpc.dwellir.com",
    "wss://statemine-rpc-tn.dwellir.com",
    "wss://public-rpc.pinknode.io/statemine",
    "wss://statemine.public.curie.radiumblock.co/ws"
  ],
  paraId: 1e3,
  relay: { id: "kusama" },
  balanceModuleConfigs: {
    "substrate-assets": {
      tokens: [
        { assetId: 8, symbol: "RMRK", coingeckoId: "rmrk" },
        { assetId: 16, symbol: "ARIS" },
        { assetId: 1984, symbol: "USDT", coingeckoId: "tether" }
      ]
    }
  }
};
var Statemint = {
  id: "statemint",
  name: "Statemint",
  account: "*25519",
  subscanUrl: "https://statemint.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/statemint_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/statemint_metadata_latest.apng",
  rpcs: [
    "wss://statemint-rpc.polkadot.io",
    "wss://statemint.api.onfinality.io/public-ws",
    "wss://statemint-rpc.dwellir.com",
    "wss://statemint-rpc-tn.dwellir.com",
    "wss://public-rpc.pinknode.io/statemint",
    "wss://statemint.public.curie.radiumblock.co/ws"
  ],
  paraId: 1e3,
  relay: { id: "polkadot" },
  balanceModuleConfigs: {
    "substrate-assets": {
      tokens: [{ assetId: 1984, symbol: "USDT", coingeckoId: "tether" }]
    }
  }
};
var Subdao = {
  id: "subdao",
  name: "SubDAO",
  account: "*25519",
  rpcs: ["wss://parachain-rpc.subdao.org"],
  paraId: 2018,
  relay: { id: "polkadot" }
};
var SubgameKusama = {
  id: "subgame-kusama",
  name: "SubGame Gamma Kusama",
  account: "*25519",
  rpcs: ["wss://gamma.subgame.org/"],
  paraId: 2018,
  relay: { id: "kusama" }
};
var SubgamePolkadot = {
  id: "subgame-polkadot",
  name: "SubGame Gamma Polkadot",
  account: "*25519",
  rpcs: [],
  paraId: 2017,
  relay: { id: "polkadot" }
};
var SubgameStandalone = {
  id: "subgame-standalone",
  name: "SubGame",
  account: "*25519",
  subscanUrl: "https://subgame.subscan.io/",
  rpcs: ["wss://mainnet.subgame.org/"]
};
var SubsocialX = {
  id: "subsocial-x",
  name: "SubsocialX",
  coingeckoId: null,
  account: "*25519",
  chainspecQrUrl: "https://metadata.novasama.io/qr/subsocial-parachain_specs.png",
  rpcs: ["wss://para.f3joule.network", "wss://para.subsocial.network"],
  paraId: 2100,
  relay: { id: "kusama" }
};
var Swapdex = {
  id: "swapdex",
  name: "Swapdex",
  account: "*25519",
  rpcs: ["wss://ws.swapdex.network"]
};
var Tanganika = {
  id: "tanganika",
  name: "Tanganika",
  account: "*25519",
  subscanUrl: "https://datahighway.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/datahighway-parachain_specs.png",
  rpcs: ["wss://tanganika.datahighway.com"],
  paraId: 2116,
  relay: { id: "kusama" }
};
var Ternoa = {
  id: "ternoa",
  name: "Ternoa",
  account: "*25519",
  chainspecQrUrl: "https://metadata.novasama.io/qr/mainnet_specs.png",
  rpcs: ["wss://mainnet.ternoa.network"]
};
var ThebifrostMainnet = {
  id: "thebifrost-mainnet",
  name: "The Bifrost",
  account: "*25519",
  rpcs: [
    "wss://public-01.mainnet.thebifrost.io/wss",
    "wss://public-02.mainnet.thebifrost.io/wss"
  ]
};
var Tinker = {
  id: "tinker",
  name: "InvArch Tinkernet",
  account: "*25519",
  chainspecQrUrl: "https://metadata.novasama.io/qr/tinkernet_node_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/tinkernet_node_metadata_latest.apng",
  rpcs: [
    "wss://tinker.invarch.network",
    "wss://invarch-tinkernet.api.onfinality.io/public-ws"
  ],
  paraId: 2125,
  relay: { id: "kusama" }
};
var Trustbase = {
  id: "trustbase",
  name: "TrustBase",
  account: "*25519",
  rpcs: [],
  paraId: 2078,
  relay: { id: "kusama" }
};
var Turing = {
  id: "turing",
  name: "Turing",
  account: "*25519",
  subscanUrl: "https://turing.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/turing_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/turing_metadata_latest.apng",
  coingeckoId: null,
  rpcs: ["wss://rpc.turing.oak.tech", "wss://turing-rpc.dwellir.com"],
  paraId: 2114,
  relay: { id: "kusama" }
};
var Uniarts = {
  id: "uniarts",
  name: "UniArts",
  account: "*25519",
  subscanUrl: "https://uniarts.subscan.io/",
  rpcs: ["wss://mainnet.uniarts.vip:9443"]
};
var Unique = {
  id: "unique",
  name: "Unique",
  account: "*25519",
  subscanUrl: "https://unique.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/unique_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/unique_metadata_latest.apng",
  rpcs: [
    "wss://us-ws.unique.network/",
    "wss://asia-ws.unique.network/",
    "wss://eu-ws.unique.network/"
  ],
  paraId: 2037,
  relay: { id: "polkadot" }
};
var Unorthodox = {
  id: "unorthodox",
  name: "Unorthodox",
  account: "*25519",
  rpcs: ["wss://rpc.kusama.standard.tech"],
  paraId: 2094,
  relay: { id: "kusama" }
};
var Westlake = {
  id: "westlake",
  name: "Westlake",
  account: "*25519",
  rpcs: []
};
var Xx = {
  id: "xx",
  name: "XX",
  account: "*25519",
  chainspecQrUrl: "https://metadata.novasama.io/qr/xxnetwork_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/xxnetwork_metadata_latest.apng",
  rpcs: ["wss://rpc.xx.network"]
};
var Zeitgeist = {
  id: "zeitgeist",
  name: "Zeitgeist",
  account: "*25519",
  subscanUrl: "https://zeitgeist.subscan.io/",
  chainspecQrUrl: "https://metadata.novasama.io/qr/zeitgeist_specs.png",
  latestMetadataQrUrl: "https://metadata.novasama.io/qr/zeitgeist_metadata_latest.apng",
  rpcs: [
    "wss://zeitgeist-rpc.dwellir.com",
    "wss://zeitgeist.api.onfinality.io/public-ws"
  ],
  paraId: 2101,
  relay: { id: "kusama" }
};
var Zero = {
  id: "zero",
  name: "Subzero",
  account: "*25519",
  rpcs: ["wss://rpc-1.kusama.node.zero.io"],
  paraId: 2236,
  relay: { id: "kusama" }
};
var _3Dpass = {
  id: "3-dpass",
  name: "3DPass",
  account: "*25519",
  rpcs: ["wss://rpc2.3dpass.org"]
};

// src/chains/data/testnet-chaindata.ts
var AjunaTestnet = {
  id: "ajuna-testnet",
  name: "Ajuna Testnet",
  account: "*25519",
  rpcs: ["wss://rpc-test.ajuna.network"]
};
var AlephTestnet = {
  id: "aleph-testnet",
  name: "Aleph Zero Testnet",
  account: "*25519",
  rpcs: ["wss://ws.test.azero.dev", "wss://aleph-zero-testnet-rpc.dwellir.com"]
};
var ArcticTestnetRococo = {
  id: "arctic-testnet-rococo",
  name: "Arctic Testnet",
  account: "*25519",
  rpcs: ["wss://arctic-rococo-rpc.icenetwork.io"],
  paraId: 3015,
  relay: { id: "rococo-testnet" }
};
var ArcticTestnetStandalone = {
  id: "arctic-testnet-standalone",
  name: "Arctic Testnet Standalone",
  account: "*25519",
  subscanUrl: "https://arctic.subscan.io/",
  rpcs: ["wss://arctic-rpc.icenetwork.io:9944"]
};
var AresGladiosTestnet = {
  id: "ares-gladios-testnet",
  name: "Ares Gladios",
  account: "*25519",
  rpcs: ["wss://gladios.aresprotocol.io"]
};
var AutomataContextfreeTestnet = {
  id: "automata-contextfree-testnet",
  name: "Automata ContextFree",
  account: "*25519",
  rpcs: [
    "wss://cf-api.ata.network",
    "wss://contextfree.api.onfinality.io/public-ws"
  ]
};
var BifrostTestnet = {
  id: "bifrost-testnet",
  name: "Bifrost Testnet",
  account: "*25519",
  subscanUrl: "https://bifrost-testnet.subscan.io/",
  rpcs: []
};
var BitcountryTestnet = {
  id: "bitcountry-testnet",
  name: "Bit.Country",
  account: "*25519",
  rpcs: ["wss://tewai-rpc.bit.country"]
};
var BrainstormTestnet = {
  id: "brainstorm-testnet",
  name: "InvArch Brainstorm",
  account: "*25519",
  rpcs: []
};
var CessTestnet = {
  id: "cess-testnet",
  name: "Cess",
  account: "*25519",
  rpcs: ["wss://testnet-rpc0.cess.cloud/ws/"]
};
var ChainoliTestnet = {
  id: "chainoli-testnet",
  name: "Oli",
  account: "*25519",
  rpcs: [],
  paraId: 4023,
  relay: { id: "rococo-testnet" }
};
var CharcoalTestnet = {
  id: "charcoal-testnet",
  name: "Charcoal",
  account: "*25519",
  rpcs: [],
  paraId: 2086,
  relay: { id: "westend-testnet" }
};
var CloverTestnet = {
  id: "clover-testnet",
  name: "Clover Testnet",
  account: "*25519",
  subscanUrl: "https://clover-testnet.subscan.io/",
  rpcs: []
};
var CoinversationTestnet = {
  id: "coinversation-testnet",
  name: "Coinversation Testnet",
  account: "*25519",
  rpcs: ["wss://rpc.coinversation.io/"]
};
var CreditcoinTestnet = {
  id: "creditcoin-testnet",
  name: "Creditcoin Testnet",
  account: "*25519",
  rpcs: ["wss://rpc.testnet.creditcoin.network/ws"]
};
var CrustMaxwellTestnet = {
  id: "crust-maxwell-testnet",
  name: "Crust Maxwell",
  account: "*25519",
  rpcs: []
};
var DatahighwayTestnet = {
  id: "datahighway-testnet",
  name: "Spreehafen",
  account: "*25519",
  rpcs: []
};
var DebioTestnet = {
  id: "debio-testnet",
  name: "DeBio Testnet",
  account: "*25519",
  rpcs: ["wss://ws-rpc.testnet.debio.network"]
};
var Development = {
  id: "development",
  name: "Development",
  account: "*25519",
  rpcs: ["ws://127.0.0.1:9944"]
};
var DockTestnet = {
  id: "dock-testnet",
  name: "Dock Testnet",
  account: "*25519",
  rpcs: []
};
var DolphinTestnet = {
  id: "dolphin-testnet",
  name: "Dolphin",
  account: "*25519",
  subscanUrl: "https://dolphin.subscan.io/",
  rpcs: []
};
var DotmogTestnet = {
  id: "dotmog-testnet",
  name: "DOTMog",
  account: "*25519",
  rpcs: ["wss://mogiway-01.dotmog.com"]
};
var EdgewareTestnet = {
  id: "edgeware-testnet",
  name: "Beresheet",
  account: "*25519",
  rpcs: ["wss://beresheet.jelliedowl.net"]
};
var EggnetTestnet = {
  id: "eggnet-testnet",
  name: "Eggnet",
  account: "*25519",
  rpcs: [],
  paraId: 4006,
  relay: { id: "rococo-testnet" }
};
var EncointerTestnetRococo = {
  id: "encointer-testnet-rococo",
  name: "Encointer Lietaer",
  account: "*25519",
  rpcs: ["wss://rococo.api.encointer.org"],
  paraId: 1003,
  relay: { id: "rococo-testnet" }
};
var EncointerTestnetStandalone = {
  id: "encointer-testnet-standalone",
  name: "Encointer Gesell",
  account: "*25519",
  rpcs: ["wss://gesell.encointer.org"]
};
var EquilibriumTestnet = {
  id: "equilibrium-testnet",
  name: "Equilibrium Testnet",
  account: "*25519",
  rpcs: []
};
var FantourTestnet = {
  id: "fantour-testnet",
  name: "Fantour",
  account: "*25519",
  rpcs: []
};
var FerrumTestnet = {
  id: "ferrum-testnet",
  name: "Ferrum",
  account: "*25519",
  rpcs: ["wss://testnet.dev.svcs.ferrumnetwork.io"]
};
var GalitalTestnet = {
  id: "galital-testnet",
  name: "Galital PC2",
  account: "*25519",
  rpcs: []
};
var GaloisTestnet = {
  id: "galois-testnet",
  name: "Galois",
  account: "*25519",
  rpcs: [
    "wss://galois-hk.maiziqianbao.net/ws",
    "wss://galois.maiziqianbao.net/ws"
  ]
};
var GamepowerTestnet = {
  id: "gamepower-testnet",
  name: "GamePower",
  account: "*25519",
  rpcs: []
};
var GeekTestnet = {
  id: "geek-testnet",
  name: "GeekCash",
  account: "*25519",
  rpcs: ["wss://testnet.geekcash.org"]
};
var GuildTestnet = {
  id: "guild-testnet",
  name: "Guild Testnet",
  account: "*25519",
  rpcs: ["wss://1.oracle.network.guild.xyz"]
};
var HalongbayTestnet = {
  id: "halongbay-testnet",
  name: "Halongbay",
  account: "*25519",
  rpcs: []
};
var HelixstreetTestnet = {
  id: "helixstreet-testnet",
  name: "Helixstreet",
  account: "*25519",
  rpcs: ["wss://rpc-rococo.helixstreet.io"],
  paraId: 3025,
  relay: { id: "rococo-testnet" }
};
var IntegriteeTestnet = {
  id: "integritee-testnet",
  name: "Integritee Testnet",
  account: "*25519",
  rpcs: ["wss://teerw1.integritee.network"],
  paraId: 2081,
  relay: { id: "westend-testnet" }
};
var InterlayTestnetStandalone = {
  id: "interlay-testnet-standalone",
  name: "Interlay Testnet",
  account: "*25519",
  rpcs: ["wss://api-testnet.interlay.io/parachain/"]
};
var InterlayTestnetWestend = {
  id: "interlay-testnet-westend",
  name: "Interlay Testnet",
  account: "*25519",
  rpcs: [],
  paraId: 2094,
  relay: { id: "westend-testnet" }
};
var IpseTestnet = {
  id: "ipse-testnet",
  name: "Ipse",
  account: "*25519",
  rpcs: [
    "wss://testnet-china.ipse.io",
    "wss://testnet-usa.ipse.io",
    "wss://testnet-europe.ipse.io"
  ]
};
var JazTestnet = {
  id: "jaz-testnet",
  name: "Artio",
  account: "*25519",
  rpcs: ["wss://ws0.jaz.network"]
};
var JoystreamTestnet = {
  id: "joystream-testnet",
  name: "Joystream Testnet",
  account: "*25519",
  rpcs: ["wss://rpc.joystream.org:9944", "wss://joystream-rpc.dwellir.com"]
};
var JupiterTestnet = {
  id: "jupiter-testnet",
  name: "Jupiter",
  account: "*25519",
  rpcs: []
};
var KaruraTestnet = {
  id: "karura-testnet",
  name: "Wendala",
  account: "*25519",
  rpcs: [],
  paraId: 2005,
  relay: { id: "westend-testnet" }
};
var KhalaTestnet = {
  id: "khala-testnet",
  name: "Khala Testnet",
  account: "*25519",
  rpcs: []
};
var KiltTestnetStandalone = {
  id: "kilt-testnet-standalone",
  name: "KILT Mashnet",
  account: "*25519",
  rpcs: ["wss://full-nodes.kilt.io:9944/"]
};
var KiltTestnetStandalone2 = {
  id: "kilt-testnet-standalone-2",
  name: "KILT Peregrine",
  account: "*25519",
  subscanUrl: "https://kilt-testnet.subscan.io/",
  rpcs: ["wss://peregrine.kilt.io/parachain-public-ws/"]
};
var KiltTestnetWestend = {
  id: "kilt-testnet-westend",
  name: "Wilt",
  account: "*25519",
  rpcs: ["wss://westend.kilt.io:9977"],
  paraId: 2085,
  relay: { id: "westend-testnet" }
};
var KintsugiTestnet = {
  id: "kintsugi-testnet",
  name: "Kintsugi Testnet",
  account: "*25519",
  rpcs: ["wss://api-dev-kintsugi.interlay.io/parachain"]
};
var KlugdossierTestnet = {
  id: "klugdossier-testnet",
  name: "Klug Dossier",
  account: "*25519",
  rpcs: []
};
var KylinTestnet = {
  id: "kylin-testnet",
  name: "Kylin Testnet",
  account: "*25519",
  rpcs: []
};
var LaminarTestnet = {
  id: "laminar-testnet",
  name: "Turbulence",
  account: "*25519",
  rpcs: []
};
var LitentryTestnet = {
  id: "litentry-testnet",
  name: "Litentry Testnet",
  account: "*25519",
  rpcs: ["wss://testnet.litentry.io"]
};
var LogionTestnetStandalone = {
  id: "logion-testnet-standalone",
  name: "Logion Testnet Parachain",
  account: "*25519",
  rpcs: ["wss://chimay.logion.network"]
};
var LogionTestnetStandalone2 = {
  id: "logion-testnet-standalone-2",
  name: "Logion Testnet",
  account: "*25519",
  rpcs: ["wss://test-rpc01.logion.network"]
};
var MandalaTestnet = {
  id: "mandala-testnet",
  name: "Mandala",
  account: "*25519",
  subscanUrl: "https://acala-testnet.subscan.io/",
  rpcs: ["wss://mandala.polkawallet.io"]
};
var MantaTestnet = {
  id: "manta-testnet",
  name: "Manta Testnet",
  account: "*25519",
  rpcs: []
};
var MinixTestnet = {
  id: "minix-testnet",
  name: "MiniX Testnet",
  account: "*25519",
  rpcs: []
};
var MoonbaseAlphaTestnet = {
  id: "moonbase-alpha-testnet",
  name: "Moonbase Alpha",
  account: "secp256k1",
  subscanUrl: "https://moonbase.subscan.io/",
  rpcs: [
    "wss://wss.api.moonbase.moonbeam.network",
    "wss://moonbase-alpha.public.blastapi.io",
    "wss://moonbeam-alpha.api.onfinality.io/public-ws",
    "wss://public-rpc.pinknode.io/alphanet"
  ]
};
var MoonshadowTestnet = {
  id: "moonshadow-testnet",
  name: "Moonshadow",
  account: "*25519",
  rpcs: [],
  paraId: 2002,
  relay: { id: "westend-testnet" }
};
var MybankTestnet = {
  id: "mybank-testnet",
  name: "mybank.network",
  account: "*25519",
  rpcs: []
};
var MyriadTestnet = {
  id: "myriad-testnet",
  name: "Myriad Testnet",
  account: "*25519",
  rpcs: ["wss://ws-rpc.testnet.myriad.social"]
};
var NftmartTestnet = {
  id: "nftmart-testnet",
  name: "NFTMart Testnet",
  account: "*25519",
  rpcs: ["wss://dev-ws.nftmart.io", "wss://staging-ws.nftmart.io"]
};
var NodleTestnet = {
  id: "nodle-testnet",
  name: "Arcadia",
  account: "*25519",
  rpcs: []
};
var OpalTestnet = {
  id: "opal-testnet",
  name: "Opal",
  account: "*25519",
  rpcs: [
    "wss://ws-opal.unique.network",
    "wss://eu-ws-opal.unique.network",
    "wss://us-ws-opal.unique.network",
    "wss://asia-ws-opal.unique.network"
  ]
};
var OpportunityTestnet = {
  id: "opportunity-testnet",
  name: "Opportunity",
  account: "*25519",
  rpcs: []
};
var PangolinTestnet = {
  id: "pangolin-testnet",
  name: "Pangolin",
  account: "*25519",
  subscanUrl: "https://pangolin.subscan.io/",
  rpcs: ["wss://pangolin-rpc.darwinia.network"]
};
var PangoroTestnetStandalone = {
  id: "pangoro-testnet-standalone",
  name: "Pangoro",
  account: "*25519",
  subscanUrl: "https://pangoro.subscan.io/",
  rpcs: ["wss://pangoro-rpc.darwinia.network"]
};
var PangoroTestnetWestend = {
  id: "pangoro-testnet-westend",
  name: "Pangoro",
  account: "*25519",
  rpcs: [],
  paraId: 2102,
  relay: { id: "westend-testnet" }
};
var PhalaTestnet = {
  id: "phala-testnet",
  name: "Phala Testnet",
  account: "*25519",
  rpcs: ["wss://poc5.phala.network/ws"]
};
var PhoenixTestnet = {
  id: "phoenix-testnet",
  name: "Phoenix Mashnet",
  account: "*25519",
  rpcs: []
};
var PichiuTestnet = {
  id: "pichiu-testnet",
  name: "Pichiu Testnet",
  account: "*25519",
  rpcs: []
};
var PolkadexTestnet = {
  id: "polkadex-testnet",
  name: "Polkadex Testnet",
  account: "*25519",
  rpcs: ["wss://blockchain.polkadex.trade"]
};
var PolymeshTestnet = {
  id: "polymesh-testnet",
  name: "Polymesh Testnet",
  account: "*25519",
  subscanUrl: "https://polymesh-testnet.subscan.io/",
  rpcs: ["wss://testnet-rpc.polymesh.live"]
};
var PontemTestnet = {
  id: "pontem-testnet",
  name: "Pontem",
  account: "*25519",
  rpcs: []
};
var PrismTestnet = {
  id: "prism-testnet",
  name: "Prism",
  account: "*25519",
  rpcs: []
};
var RealisTestnet = {
  id: "realis-testnet",
  name: "Realis.Network",
  account: "*25519",
  rpcs: []
};
var RiochainTestnet = {
  id: "riochain-testnet",
  name: "RioChain Testnet",
  account: "*25519",
  rpcs: []
};
var RobonomicsTestnet = {
  id: "robonomics-testnet",
  name: "Robonomics Testnet",
  account: "*25519",
  rpcs: ["wss://rococo.rpc.robonomics.network"],
  paraId: 2048,
  relay: { id: "rococo-testnet" }
};
var RocfinityTestnet = {
  id: "rocfinity-testnet",
  name: "Rocfinity",
  account: "*25519",
  rpcs: ["wss://rpc.rococo.efinity.io"],
  paraId: 2021,
  relay: { id: "rococo-testnet" }
};
var RococoAcurastTestnet = {
  id: "rococo-acurast-testnet",
  name: "Acurast",
  account: "*25519",
  rpcs: ["wss://ws.acurast-rococo.diamond.papers.tech"],
  paraId: 4191,
  relay: { id: "rococo-testnet" }
};
var RococoAmplitudeTestnet = {
  id: "rococo-amplitude-testnet",
  name: "Amplitude Testnet",
  account: "*25519",
  rpcs: ["wss://rpc-foucoco.pendulumchain.tech"],
  paraId: 2124,
  relay: { id: "rococo-testnet" }
};
var RococoAventusTestnet = {
  id: "rococo-aventus-testnet",
  name: "Aventus Testnet",
  account: "*25519",
  rpcs: [],
  paraId: 2056,
  relay: { id: "rococo-testnet" }
};
var RococoBajunTestnet = {
  id: "rococo-bajun-testnet",
  name: "Bajun Testnet",
  account: "*25519",
  rpcs: ["wss://rpc-rococo.bajun.network"],
  paraId: 2119,
  relay: { id: "rococo-testnet" }
};
var RococoBasiliskTestnet = {
  id: "rococo-basilisk-testnet",
  name: "Basilisk Testnet",
  account: "*25519",
  rpcs: ["wss://rococo-basilisk-rpc.hydration.dev"],
  paraId: 2090,
  relay: { id: "rococo-testnet" }
};
var RococoBifrostTestnet = {
  id: "rococo-bifrost-testnet",
  name: "Bifrost Testnet",
  account: "*25519",
  rpcs: ["wss://bifrost-rpc.rococo.liebi.com/ws"],
  paraId: 2030,
  relay: { id: "rococo-testnet" }
};
var RococoBitgreenTestnet = {
  id: "rococo-bitgreen-testnet",
  name: "Bitgreen Testnet",
  account: "*25519",
  rpcs: ["wss://staging.bitgreen.org"],
  paraId: 20048,
  relay: { id: "rococo-testnet" }
};
var RococoBridgehubTestnet = {
  id: "rococo-bridgehub-testnet",
  name: "Bridgehub Rococo",
  account: "*25519",
  rpcs: ["wss://rococo-bridge-hub-rpc.polkadot.io"],
  paraId: 1013,
  relay: { id: "rococo-testnet" }
};
var RococoCatalystTestnet = {
  id: "rococo-catalyst-testnet",
  name: "Catalyst",
  account: "*25519",
  rpcs: ["wss://fullnode.catalyst.cntrfg.com"],
  paraId: 2031,
  relay: { id: "rococo-testnet" }
};
var RococoConftiTestnet = {
  id: "rococo-confti-testnet",
  name: "Confti",
  account: "*25519",
  rpcs: [],
  paraId: 4094,
  relay: { id: "rococo-testnet" }
};
var RococoContractsTestnet = {
  id: "rococo-contracts-testnet",
  name: "Contracts",
  account: "*25519",
  rpcs: ["wss://rococo-contracts-rpc.polkadot.io"],
  paraId: 1002,
  relay: { id: "rococo-testnet" }
};
var RococoCrustTestnet = {
  id: "rococo-crust-testnet",
  name: "Crust Testnet",
  account: "*25519",
  rpcs: ["wss://rococo-csm.crustcode.com/"],
  paraId: 2012,
  relay: { id: "rococo-testnet" }
};
var RococoDaliTestnet = {
  id: "rococo-dali-testnet",
  name: "Dali",
  account: "*25519",
  subscanUrl: "https://dali.subscan.io/",
  rpcs: ["wss://rpc.composablefinance.ninja"],
  paraId: 2087,
  relay: { id: "rococo-testnet" }
};
var RococoDolphinTestnet = {
  id: "rococo-dolphin-testnet",
  name: "Dolphin",
  account: "*25519",
  subscanUrl: "https://dolphin.subscan.io/",
  rpcs: ["wss://ws.rococo.dolphin.engineering"],
  paraId: 2084,
  relay: { id: "rococo-testnet" }
};
var RococoEthosTestnet = {
  id: "rococo-ethos-testnet",
  name: "Ethos",
  account: "*25519",
  rpcs: ["wss://ethos.jur.io"],
  paraId: 2095,
  relay: { id: "rococo-testnet" }
};
var RococoFrequencyTestnet = {
  id: "rococo-frequency-testnet",
  name: "Frequency Testnet",
  account: "*25519",
  rpcs: ["wss://rpc.rococo.frequency.xyz"],
  paraId: 4044,
  relay: { id: "rococo-testnet" }
};
var RococoGenshiroTestnet = {
  id: "rococo-genshiro-testnet",
  name: "Genshiro Testnet",
  account: "*25519",
  rpcs: ["wss://parachain-testnet.equilab.io/rococo/collator/node1/wss"],
  paraId: 2024,
  relay: { id: "rococo-testnet" }
};
var RococoHydraDxTestnet = {
  id: "rococo-hydra-dx-testnet",
  name: "HydraDX Testnet",
  account: "*25519",
  rpcs: ["wss://rococo-hydradx-rpc.hydration.dev"],
  paraId: 2034,
  relay: { id: "rococo-testnet" }
};
var RococoImbueTestnet = {
  id: "rococo-imbue-testnet",
  name: "Imbue Testnet",
  account: "*25519",
  rpcs: ["wss://rococo.imbue.network"],
  paraId: 2121,
  relay: { id: "rococo-testnet" }
};
var RococoIntegriteeTestnet = {
  id: "rococo-integritee-testnet",
  name: "Integritee Testnet",
  account: "*25519",
  rpcs: ["wss://rococo.api.integritee.network"],
  paraId: 3002,
  relay: { id: "rococo-testnet" }
};
var RococoKabochaTestnet = {
  id: "rococo-kabocha-testnet",
  name: "Kabocha Testnet",
  account: "*25519",
  rpcs: ["wss://kabsoup1.jelliedowl.com"],
  paraId: 2113,
  relay: { id: "rococo-testnet" }
};
var RococoKiltTestnet = {
  id: "rococo-kilt-testnet",
  name: "Rilt",
  account: "*25519",
  rpcs: ["wss://rococo.kilt.io"],
  paraId: 2086,
  relay: { id: "rococo-testnet" }
};
var RococoLitentryTestnet = {
  id: "rococo-litentry-testnet",
  name: "Litentry Testnet",
  account: "*25519",
  rpcs: ["wss://rpc.rococo-parachain-sg.litentry.io"],
  paraId: 2106,
  relay: { id: "rococo-testnet" }
};
var RococoMangataTestnet = {
  id: "rococo-mangata-testnet",
  name: "MangataX Testnet",
  account: "*25519",
  subscanUrl: "https://mangata-testnet.subscan.io/",
  rpcs: ["wss://roccoco-testnet-collator-01.mangatafinance.cloud"],
  paraId: 2110,
  relay: { id: "rococo-testnet" }
};
var RococoMd5Testnet = {
  id: "rococo-md-5-testnet",
  name: "MD5",
  account: "*25519",
  rpcs: ["wss://c1.md5.network"],
  paraId: 2089,
  relay: { id: "rococo-testnet" }
};
var RococoMoonsamaTestnet = {
  id: "rococo-moonsama-testnet",
  name: "Moonsama",
  account: "*25519",
  rpcs: [],
  paraId: 2055,
  relay: { id: "rococo-testnet" }
};
var RococoNodleTestnet = {
  id: "rococo-nodle-testnet",
  name: "Nodle Testnet",
  account: "*25519",
  rpcs: ["wss://nodle-paradis.api.onfinality.io/public-ws"],
  paraId: 2026,
  relay: { id: "rococo-testnet" }
};
var RococoOriginTrailParachainTestnet = {
  id: "rococo-origin-trail-parachain-testnet",
  name: "OriginTrail Testnet",
  account: "*25519",
  rpcs: ["wss://parachain-testnet-rpc.origin-trail.network/"],
  paraId: 2043,
  relay: { id: "rococo-testnet" }
};
var RococoPangolinTestnet = {
  id: "rococo-pangolin-testnet",
  name: "Pangolin",
  account: "*25519",
  subscanUrl: "https://pangolin-parachain.subscan.io/",
  rpcs: ["wss://pangolin-parachain-rpc.darwinia.network"],
  paraId: 2105,
  relay: { id: "rococo-testnet" }
};
var RococoSoraTestnet = {
  id: "rococo-sora-testnet",
  name: "Sora Rococo",
  account: "*25519",
  rpcs: ["wss://ws.parachain-collator-1.c1.stg1.sora2.soramitsu.co.jp"],
  paraId: 2011,
  relay: { id: "rococo-testnet" }
};
var RococoSpreehafenTestnet = {
  id: "rococo-spreehafen-testnet",
  name: "Spreehafen",
  account: "*25519",
  rpcs: ["wss://spreehafen.datahighway.com"],
  paraId: 2116,
  relay: { id: "rococo-testnet" }
};
var RococoStatemintTestnet = {
  id: "rococo-statemint-testnet",
  name: "Rockmine",
  account: "*25519",
  rpcs: ["wss://rococo-rockmine-rpc.polkadot.io"],
  paraId: 1e3,
  relay: { id: "rococo-testnet" }
};
var RococoSubsocialTestnet = {
  id: "rococo-subsocial-testnet",
  name: "SoonsocialX",
  account: "*25519",
  rpcs: ["wss://rco-para.subsocial.network"],
  paraId: 2100,
  relay: { id: "rococo-testnet" }
};
var RococoSubzeroTestnet = {
  id: "rococo-subzero-testnet",
  name: "Subzero Testnet",
  account: "*25519",
  rpcs: ["wss://staging.para.sub.zero.io"],
  paraId: 4040,
  relay: { id: "rococo-testnet" }
};
var RococoTestnet = {
  id: "rococo-testnet",
  name: "Rococo",
  account: "*25519",
  subscanUrl: "https://rococo.subscan.io/",
  rpcs: ["wss://rococo-rpc.polkadot.io"]
};
var RococoTinkernetTestnet = {
  id: "rococo-tinkernet-testnet",
  name: "Tinkernet",
  account: "*25519",
  rpcs: [],
  paraId: 2125,
  relay: { id: "rococo-testnet" }
};
var RococoTuringTestnet = {
  id: "rococo-turing-testnet",
  name: "Turing Testnet",
  account: "*25519",
  rpcs: ["wss://rpc.turing-staging.oak.tech"],
  paraId: 2114,
  relay: { id: "rococo-testnet" }
};
var RococoUnitNetworkTestnet = {
  id: "rococo-unit-network-testnet",
  name: "Unit",
  account: "*25519",
  rpcs: ["wss://www.unitnode3.info:443"],
  paraId: 4168,
  relay: { id: "rococo-testnet" }
};
var RococoVirtoTestnet = {
  id: "rococo-virto-testnet",
  name: "Virto",
  account: "*25519",
  rpcs: [],
  paraId: 3003,
  relay: { id: "rococo-testnet" }
};
var RococoWatrTestnet = {
  id: "rococo-watr-testnet",
  name: "Watr",
  account: "*25519",
  rpcs: ["wss://rpc.dev.watr.org"],
  paraId: 2058,
  relay: { id: "rococo-testnet" }
};
var RococoZeitgeistTestnet = {
  id: "rococo-zeitgeist-testnet",
  name: "Zeitgeist Rococo Battery Station",
  account: "*25519",
  rpcs: ["wss://roc.zeitgeist.pm"],
  paraId: 2101,
  relay: { id: "rococo-testnet" }
};
var SherpaxTestnet = {
  id: "sherpax-testnet",
  name: "Sherpax Testnet",
  account: "*25519",
  rpcs: ["wss://sherpax-testnet.chainx.org"]
};
var ShibuyaTestnet = {
  id: "shibuya-testnet",
  name: "Shibuya",
  account: "*25519",
  subscanUrl: "https://shibuya.subscan.io/",
  rpcs: ["wss://rpc.shibuya.astar.network", "wss://shibuya-rpc.dwellir.com"]
};
var SkyekiwiTestnet = {
  id: "skyekiwi-testnet",
  name: "SkyeKiwi",
  account: "*25519",
  rpcs: ["wss://staging.rpc.skye.kiwi"]
};
var SnowbridgeTestnet = {
  id: "snowbridge-testnet",
  name: "Snowbridge",
  account: "*25519",
  rpcs: ["wss://rococo-rpc.snowbridge.network"],
  paraId: 3016,
  relay: { id: "rococo-testnet" }
};
var SoonsocialTestnet = {
  id: "soonsocial-testnet",
  name: "Soonsocial",
  account: "*25519",
  rpcs: []
};
var SoraSubstrateTestnet = {
  id: "sora-substrate-testnet",
  name: "Sora Testnet",
  account: "*25519",
  rpcs: [
    "wss://ws.framenode-1.s1.stg1.sora2.soramitsu.co.jp",
    "wss://ws.framenode-2.s1.stg1.sora2.soramitsu.co.jp",
    "wss://ws.framenode-3.s2.stg1.sora2.soramitsu.co.jp",
    "wss://ws.framenode-4.s2.stg1.sora2.soramitsu.co.jp"
  ]
};
var StagexTestnet = {
  id: "stagex-testnet",
  name: "Stagex",
  account: "*25519",
  rpcs: ["wss://s-ui.kapex.network"],
  paraId: 2007,
  relay: { id: "rococo-testnet" }
};
var SubdaoTestnet = {
  id: "subdao-testnet",
  name: "SubDAO Testnet",
  account: "*25519",
  rpcs: []
};
var SubgameTestnet = {
  id: "subgame-testnet",
  name: "SubGame Testnet",
  account: "*25519",
  rpcs: []
};
var SubspaceFarmnetTestnet = {
  id: "subspace-farmnet-testnet",
  name: "Subspace Farmnet",
  account: "*25519",
  rpcs: []
};
var SubspaceGemini1Testnet = {
  id: "subspace-gemini-1-testnet",
  name: "Subspace Gemini",
  account: "*25519",
  rpcs: ["wss://eu.gemini-1b.subspace.network/ws"]
};
var SubspaceGemini2ATestnet = {
  id: "subspace-gemini-2-a-testnet",
  name: "Subspace Gemini 2a",
  account: "*25519",
  rpcs: [
    "wss://eu-0.gemini-2a.subspace.network/ws",
    "wss://eu-1.gemini-2a.subspace.network/ws",
    "wss://eu-2.gemini-2a.subspace.network/ws",
    "wss://subspace-gemini-2a-rpc.dwellir.com"
  ]
};
var SubspaceTestnet = {
  id: "subspace-testnet",
  name: "Subspace",
  account: "*25519",
  rpcs: ["wss://test-rpc.subspace.network"]
};
var SubstrateTestnet = {
  id: "substrate-testnet",
  name: "Flaming Fir",
  account: "*25519",
  rpcs: []
};
var SumiStagingTestnet = {
  id: "sumi-staging-testnet",
  name: "SUMI staging testnet",
  account: "*25519",
  rpcs: ["wss://testnet1.sumi.xyz"]
};
var T0RnTestnet = {
  id: "t-0-rn-testnet",
  name: "t0rn",
  account: "*25519",
  rpcs: ["wss://ws.t0rn.io"],
  paraId: 3333,
  relay: { id: "rococo-testnet" }
};
var TernoaAlphanetTestnet = {
  id: "ternoa-alphanet-testnet",
  name: "Ternoa Alphanet",
  account: "*25519",
  rpcs: ["wss://alphanet.ternoa.com"]
};
var TernoaTestnet = {
  id: "ternoa-testnet",
  name: "Ternoa Testnet",
  account: "*25519",
  rpcs: ["wss://testnet.ternoa.com/"]
};
var ThebifrostTestnet = {
  id: "thebifrost-testnet",
  name: "The Bifrost Testnet",
  account: "*25519",
  rpcs: [
    "wss://public-01.testnet.thebifrost.io/ws",
    "wss://public-02.testnet.thebifrost.io/ws"
  ]
};
var UniartsTestnet = {
  id: "uniarts-testnet",
  name: "UniArts Testnet",
  account: "*25519",
  rpcs: ["wss://testnet.uniarts.network"]
};
var UniqueTestnet = {
  id: "unique-testnet",
  name: "Unique Testnet",
  account: "*25519",
  rpcs: []
};
var UnitvTestnet = {
  id: "unitv-testnet",
  name: "Unit",
  account: "*25519",
  rpcs: []
};
var VaraTestnet = {
  id: "vara-testnet",
  name: "Vara",
  account: "*25519",
  rpcs: ["wss://rpc.vara-network.io"]
};
var VodkaTestnet = {
  id: "vodka-testnet",
  name: "Vodka",
  account: "*25519",
  rpcs: []
};
var Web3GamesTestnet = {
  id: "web-3-games-testnet",
  name: "Web3Games",
  account: "*25519",
  rpcs: ["wss://devnet.web3games.org"]
};
var WestendBridgeHubTestnet = {
  id: "westend-bridge-hub-testnet",
  name: "BridgeHub Westend",
  account: "*25519",
  rpcs: ["wss://westend-bridge-hub-rpc.polkadot.io"],
  paraId: 1002,
  relay: { id: "westend-testnet" }
};
var WestendCollectivesTestnet = {
  id: "westend-collectives-testnet",
  name: "Collectives Testnet",
  account: "*25519",
  rpcs: ["wss://westend-collectives-rpc.polkadot.io"],
  paraId: 1001,
  relay: { id: "westend-testnet" }
};
var WestendPichiuTestnet = {
  id: "westend-pichiu-testnet",
  name: "Pichiu Testnet",
  account: "*25519",
  rpcs: ["wss://westend.kylin-node.co.uk"],
  paraId: 2112,
  relay: { id: "westend-testnet" }
};
var WestendStandardTestnet = {
  id: "westend-standard-testnet",
  name: "Standard",
  account: "*25519",
  rpcs: ["wss://rpc.westend.standard.tech"],
  paraId: 2094,
  relay: { id: "westend-testnet" }
};
var WestendTestnet = {
  id: "westend-testnet",
  name: "Westend",
  account: "*25519",
  subscanUrl: "https://westend.subscan.io/",
  chainspecQrUrl: "https://metadata.parity.io/qr/westend_specs.png",
  latestMetadataQrUrl: "https://metadata.parity.io/qr/westend_metadata_latest.apng",
  rpcs: [
    "wss://westend-rpc.polkadot.io",
    "wss://westend.api.onfinality.io/public-ws",
    "wss://rpc.pinknode.io/westend/explorer",
    "wss://westend-rpc.dwellir.com",
    "wss://westend-rpc-tn.dwellir.com",
    "wss://rpc.dotters.network/westend",
    "wss://rpc.ibp.network/westend"
  ]
};
var WestmintTestnet = {
  id: "westmint-testnet",
  name: "Westmint",
  account: "*25519",
  rpcs: [
    "wss://westmint-rpc.polkadot.io",
    "wss://westmint-rpc.dwellir.com",
    "wss://westmint-rpc-tn.dwellir.com"
  ],
  paraId: 1e3,
  relay: { id: "westend-testnet" }
};
var WhalaTestnet = {
  id: "whala-testnet",
  name: "Whala",
  account: "*25519",
  rpcs: [],
  paraId: 2013,
  relay: { id: "westend-testnet" }
};
var ZCloakTestnet = {
  id: "z-cloak-testnet",
  name: "zCloak",
  account: "*25519",
  rpcs: []
};
var ZeitgeistTestnet = {
  id: "zeitgeist-testnet",
  name: "Zeitgeist Battery Station",
  account: "*25519",
  rpcs: ["wss://bsr.zeitgeist.pm"]
};
var ZeroTestnet = {
  id: "zero-testnet",
  name: "Zero Alphaville",
  account: "*25519",
  rpcs: []
};
var _3DpassTestnet = {
  id: "3-dpass-testnet",
  name: "3DPass Testnet",
  account: "*25519",
  rpcs: ["wss://test-rpc.3dpass.org"]
};

export {
  Acala,
  Ajuna,
  Aleph,
  Altair,
  Amplitude,
  AresOdyssey,
  Astar,
  Automata,
  Aventus,
  Bajun,
  Basilisk,
  BifrostKusama,
  BifrostPolkadot,
  BitcountryPioneer,
  Bitgreen,
  Bittensor,
  Calamari,
  CentrifugePolkadot,
  Chainx,
  Clover,
  Coinversation,
  CompetitorsClub,
  ComposableFinance,
  CrabKusama,
  CrabStandalone,
  Creditcoin,
  CrownSterling,
  Crust,
  CrustParachain,
  Custom,
  DarwiniaPolkadot,
  DarwiniaPolkadot2,
  DarwiniaStandalone,
  Debio,
  DockPosMainnet,
  Dorafactory,
  Edgeware,
  EfinityPolkadot,
  EfinityStandalone,
  Encointer,
  EquilibriumPolkadot,
  Frequency,
  Geminis,
  GenshiroKusama,
  Gm,
  Hanonycash,
  Hashed,
  HeikoKusama,
  Hydra,
  Imbue,
  IntegriteeKusama,
  IntegriteePolkadot,
  Interlay,
  Ipci,
  Joystream,
  Kabocha,
  Kapex,
  Karura,
  Khala,
  Kico,
  KiltSpiritnet,
  Kintsugi,
  Kpron,
  Kulupu,
  Kusama,
  KusamaBridgeHub,
  Kusari,
  Kylin,
  Listen,
  Litentry,
  Litmus,
  Logion,
  LoomNetwork,
  Luhn,
  Mangata,
  Manta,
  Mars,
  Mathchain,
  Minix,
  Moonbeam,
  Moonriver,
  Myriad,
  Neatcoin,
  Nftmart,
  NodlePolkadot,
  Oak,
  Odyssey,
  Omnibtc,
  OrigintrailParachain,
  Parallel,
  Pendulum,
  Phala,
  Picasso,
  Pichiu,
  PolkadexPolkadot,
  PolkadexStandalone,
  Polkadot,
  PolkadotCollectives,
  Polkasmith,
  Polymesh,
  Quartz,
  Riochain,
  Riodefi,
  RobonomicsKusama,
  Sakura,
  ShadowKusama,
  Sherpax,
  ShidenKusama,
  Snow,
  SoraKsm,
  SoraSubstrate,
  Spanner,
  Stafi,
  Statemine,
  Statemint,
  Subdao,
  SubgameKusama,
  SubgamePolkadot,
  SubgameStandalone,
  SubsocialX,
  Swapdex,
  Tanganika,
  Ternoa,
  ThebifrostMainnet,
  Tinker,
  Trustbase,
  Turing,
  Uniarts,
  Unique,
  Unorthodox,
  Westlake,
  Xx,
  Zeitgeist,
  Zero,
  _3Dpass,
  AjunaTestnet,
  AlephTestnet,
  ArcticTestnetRococo,
  ArcticTestnetStandalone,
  AresGladiosTestnet,
  AutomataContextfreeTestnet,
  BifrostTestnet,
  BitcountryTestnet,
  BrainstormTestnet,
  CessTestnet,
  ChainoliTestnet,
  CharcoalTestnet,
  CloverTestnet,
  CoinversationTestnet,
  CreditcoinTestnet,
  CrustMaxwellTestnet,
  DatahighwayTestnet,
  DebioTestnet,
  Development,
  DockTestnet,
  DolphinTestnet,
  DotmogTestnet,
  EdgewareTestnet,
  EggnetTestnet,
  EncointerTestnetRococo,
  EncointerTestnetStandalone,
  EquilibriumTestnet,
  FantourTestnet,
  FerrumTestnet,
  GalitalTestnet,
  GaloisTestnet,
  GamepowerTestnet,
  GeekTestnet,
  GuildTestnet,
  HalongbayTestnet,
  HelixstreetTestnet,
  IntegriteeTestnet,
  InterlayTestnetStandalone,
  InterlayTestnetWestend,
  IpseTestnet,
  JazTestnet,
  JoystreamTestnet,
  JupiterTestnet,
  KaruraTestnet,
  KhalaTestnet,
  KiltTestnetStandalone,
  KiltTestnetStandalone2,
  KiltTestnetWestend,
  KintsugiTestnet,
  KlugdossierTestnet,
  KylinTestnet,
  LaminarTestnet,
  LitentryTestnet,
  LogionTestnetStandalone,
  LogionTestnetStandalone2,
  MandalaTestnet,
  MantaTestnet,
  MinixTestnet,
  MoonbaseAlphaTestnet,
  MoonshadowTestnet,
  MybankTestnet,
  MyriadTestnet,
  NftmartTestnet,
  NodleTestnet,
  OpalTestnet,
  OpportunityTestnet,
  PangolinTestnet,
  PangoroTestnetStandalone,
  PangoroTestnetWestend,
  PhalaTestnet,
  PhoenixTestnet,
  PichiuTestnet,
  PolkadexTestnet,
  PolymeshTestnet,
  PontemTestnet,
  PrismTestnet,
  RealisTestnet,
  RiochainTestnet,
  RobonomicsTestnet,
  RocfinityTestnet,
  RococoAcurastTestnet,
  RococoAmplitudeTestnet,
  RococoAventusTestnet,
  RococoBajunTestnet,
  RococoBasiliskTestnet,
  RococoBifrostTestnet,
  RococoBitgreenTestnet,
  RococoBridgehubTestnet,
  RococoCatalystTestnet,
  RococoConftiTestnet,
  RococoContractsTestnet,
  RococoCrustTestnet,
  RococoDaliTestnet,
  RococoDolphinTestnet,
  RococoEthosTestnet,
  RococoFrequencyTestnet,
  RococoGenshiroTestnet,
  RococoHydraDxTestnet,
  RococoImbueTestnet,
  RococoIntegriteeTestnet,
  RococoKabochaTestnet,
  RococoKiltTestnet,
  RococoLitentryTestnet,
  RococoMangataTestnet,
  RococoMd5Testnet,
  RococoMoonsamaTestnet,
  RococoNodleTestnet,
  RococoOriginTrailParachainTestnet,
  RococoPangolinTestnet,
  RococoSoraTestnet,
  RococoSpreehafenTestnet,
  RococoStatemintTestnet,
  RococoSubsocialTestnet,
  RococoSubzeroTestnet,
  RococoTestnet,
  RococoTinkernetTestnet,
  RococoTuringTestnet,
  RococoUnitNetworkTestnet,
  RococoVirtoTestnet,
  RococoWatrTestnet,
  RococoZeitgeistTestnet,
  SherpaxTestnet,
  ShibuyaTestnet,
  SkyekiwiTestnet,
  SnowbridgeTestnet,
  SoonsocialTestnet,
  SoraSubstrateTestnet,
  StagexTestnet,
  SubdaoTestnet,
  SubgameTestnet,
  SubspaceFarmnetTestnet,
  SubspaceGemini1Testnet,
  SubspaceGemini2ATestnet,
  SubspaceTestnet,
  SubstrateTestnet,
  SumiStagingTestnet,
  T0RnTestnet,
  TernoaAlphanetTestnet,
  TernoaTestnet,
  ThebifrostTestnet,
  UniartsTestnet,
  UniqueTestnet,
  UnitvTestnet,
  VaraTestnet,
  VodkaTestnet,
  Web3GamesTestnet,
  WestendBridgeHubTestnet,
  WestendCollectivesTestnet,
  WestendPichiuTestnet,
  WestendStandardTestnet,
  WestendTestnet,
  WestmintTestnet,
  WhalaTestnet,
  ZCloakTestnet,
  ZeitgeistTestnet,
  ZeroTestnet,
  _3DpassTestnet
};
