// useink/dist/notifications.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/notifications/index.ts
var notifications_exports = {};
__export(notifications_exports, {
  DEFAULT_NOTIFICATIONS: () => DEFAULT_NOTIFICATIONS,
  NotificationsContext: () => NotificationsContext,
  NotificationsProvider: () => NotificationsProvider,
  toNotificationLevel: () => toNotificationLevel,
  useNotifications: () => useNotifications,
  useTxNotifications: () => useTxNotifications
});
module.exports = __toCommonJS(notifications_exports);

// src/notifications/model.ts
var DEFAULT_NOTIFICATIONS = [];

// src/notifications/context.ts
var import_react = require("react");
var NotificationsContext = (0, import_react.createContext)({
  notifications: DEFAULT_NOTIFICATIONS,
  config: void 0,
  addNotification: () => null,
  removeNotification: () => null
});

// src/react/constants.ts
var HALF_A_SECOND = 500;

// src/react/hooks/internal/useInterval.ts
var import_react2 = require("react");
function useInterval(callback, delay) {
  const savedCallback = (0, import_react2.useRef)(callback);
  (0, import_react2.useEffect)(() => {
    savedCallback.current = callback;
  }, [callback]);
  (0, import_react2.useEffect)(() => {
    if (delay === null)
      return;
    const id = setInterval(() => savedCallback.current(), delay);
    return () => clearInterval(id);
  }, [delay]);
}

// ../../node_modules/.pnpm/@talismn+connect-wallets@1.2.3_@polkadot+api@10.9.1_@polkadot+extension-inject@0.46.5/node_modules/@talismn/connect-wallets/dist/esm/index.js
var import_react3 = require("react");
function __awaiter(thisArg, _arguments, P, generator) {
  function adopt(value) {
    return value instanceof P ? value : new P(function(resolve) {
      resolve(value);
    });
  }
  return new (P || (P = Promise))(function(resolve, reject) {
    function fulfilled(value) {
      try {
        step(generator.next(value));
      } catch (e) {
        reject(e);
      }
    }
    function rejected(value) {
      try {
        step(generator["throw"](value));
      } catch (e) {
        reject(e);
      }
    }
    function step(result) {
      result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected);
    }
    step((generator = generator.apply(thisArg, _arguments || [])).next());
  });
}
var BaseWalletError = class extends Error {
  constructor(message, wallet) {
    super(message);
    this.name = "WalletError";
    Object.setPrototypeOf(this, new.target.prototype);
    if (Error.captureStackTrace) {
      Error.captureStackTrace(this, BaseWalletError);
    }
    this.wallet = wallet;
  }
};
var AuthError = class extends BaseWalletError {
  constructor() {
    super(...arguments);
    this.name = "AuthError";
  }
};
var NotInstalledError = class extends BaseWalletError {
  constructor() {
    super(...arguments);
    this.name = "NotInstalledError";
  }
};
var BaseDotsamaWallet = class {
  constructor() {
    this.extensionName = "";
    this.title = "";
    this.installUrl = "";
    this.logo = {
      src: "",
      alt: ""
    };
    this.transformError = (err) => {
      if (err.message.includes("pending authorization request")) {
        return new AuthError(err.message, this);
      }
      return err;
    };
    this.enable = (dappName) => __awaiter(this, void 0, void 0, function* () {
      if (!dappName) {
        throw new Error("MissingParamsError: Dapp name is required.");
      }
      if (!this.installed) {
        throw new NotInstalledError(`Refresh the browser if ${this.title} is already installed.`, this);
      }
      try {
        const injectedExtension = this.rawExtension;
        const rawExtension = yield injectedExtension === null || injectedExtension === void 0 ? void 0 : injectedExtension.enable(dappName);
        if (!rawExtension) {
          throw new NotInstalledError(`${this.title} is installed but is not returned by the 'Wallet.enable(dappname)' function`, this);
        }
        const extension = Object.assign(Object.assign({}, rawExtension), { name: this.extensionName, version: injectedExtension.version });
        this._extension = extension;
        this._signer = extension === null || extension === void 0 ? void 0 : extension.signer;
      } catch (err) {
        throw this.transformError(err);
      }
    });
    this.getAccounts = (anyType) => __awaiter(this, void 0, void 0, function* () {
      if (!this._extension) {
        throw new NotInstalledError(`The 'Wallet.enable(dappname)' function should be called first.`, this);
      }
      const accounts = yield this._extension.accounts.get(anyType);
      const accountsWithWallet = accounts.map((account) => {
        var _a, _b;
        return Object.assign(Object.assign({}, account), { source: (_a = this._extension) === null || _a === void 0 ? void 0 : _a.name, wallet: this, signer: (_b = this._extension) === null || _b === void 0 ? void 0 : _b.signer });
      });
      return accountsWithWallet;
    });
    this.subscribeAccounts = (callback) => __awaiter(this, void 0, void 0, function* () {
      if (!this._extension) {
        throw new NotInstalledError(`The 'Wallet.enable(dappname)' function should be called first.`, this);
      }
      const unsubscribe = this._extension.accounts.subscribe((accounts) => {
        const accountsWithWallet = accounts.map((account) => {
          var _a, _b;
          return Object.assign(Object.assign({}, account), { source: (_a = this._extension) === null || _a === void 0 ? void 0 : _a.name, wallet: this, signer: (_b = this._extension) === null || _b === void 0 ? void 0 : _b.signer });
        });
        callback(accountsWithWallet);
      });
      return unsubscribe;
    });
  }
  get extension() {
    return this._extension;
  }
  get signer() {
    return this._signer;
  }
  get installed() {
    var _a;
    const injectedWindow = window;
    const injectedExtension = (_a = injectedWindow === null || injectedWindow === void 0 ? void 0 : injectedWindow.injectedWeb3) === null || _a === void 0 ? void 0 : _a[this.extensionName];
    return !!injectedExtension;
  }
  get rawExtension() {
    var _a;
    const injectedWindow = window;
    const injectedExtension = (_a = injectedWindow === null || injectedWindow === void 0 ? void 0 : injectedWindow.injectedWeb3) === null || _a === void 0 ? void 0 : _a[this.extensionName];
    return injectedExtension;
  }
};
var logo$3 = "data:image/svg+xml,%3Csvg%20width%3D%2224%22%20height%3D%2224%22%20viewBox%3D%220%200%2024%2024%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%3Crect%20width%3D%2224%22%20height%3D%2224%22%20rx%3D%224%22%20fill%3D%22%23D5FF5C%22%2F%3E%3Cpath%20fill-rule%3D%22evenodd%22%20clip-rule%3D%22evenodd%22%20d%3D%22M16.5349%2012.9159C16.6871%2013.2472%2017.135%2013.3643%2017.3929%2013.1065L17.8651%2012.6345C18.3535%2012.1464%2019.1453%2012.1464%2019.6337%2012.6345C20.1221%2013.1227%2020.1221%2013.9141%2019.6337%2014.4023L15.8097%2018.2246C14.8921%2019.3104%2013.5198%2020%2011.9865%2020C10.3851%2020%208.95942%2019.2478%208.04385%2018.0777L4.36629%2014.4018C3.87791%2013.9136%203.87791%2013.1222%204.36629%2012.634C4.85467%2012.1459%205.64649%2012.1459%206.13487%2012.634L6.60044%2013.0994C6.85253%2013.3514%207.29002%2013.238%207.43894%2012.9141V12.9141C7.46838%2012.8501%207.48439%2012.7814%207.48439%2012.711L7.48438%207.00059C7.48438%206.30991%208.04428%205.75001%208.73496%205.75001C9.42563%205.75001%209.98553%206.30991%209.98553%207.00058L9.98553%209.88892C9.98553%2010.1376%2010.2403%2010.3065%2010.4774%2010.2315V10.2315C10.6276%2010.1841%2010.736%2010.0474%2010.736%209.89001L10.736%205.25041C10.736%204.55974%2011.2959%203.99984%2011.9866%203.99984C12.6773%203.99984%2013.2372%204.55974%2013.2372%205.25041L13.2372%209.89018C13.2372%2010.0476%2013.3456%2010.1842%2013.4957%2010.2316V10.2316C13.7327%2010.3065%2013.9874%2010.1377%2013.9874%209.88909L13.9874%207.00059C13.9874%206.30991%2014.5473%205.75001%2015.2379%205.75001C15.9286%205.75001%2016.4885%206.30991%2016.4885%207.00058L16.4885%2012.7086C16.4885%2012.7805%2016.5049%2012.8506%2016.5349%2012.9159V12.9159Z%22%20fill%3D%22%23FD4848%22%2F%3E%3Cpath%20d%3D%22M15.9885%2015.5C15.9885%2015.5%2014.1969%2018%2011.9867%2018C9.77655%2018%207.98486%2015.5%207.98486%2015.5C7.98486%2015.5%209.77655%2013%2011.9867%2013C14.1969%2013%2015.9885%2015.5%2015.9885%2015.5Z%22%20fill%3D%22%23D5FF5C%22%2F%3E%3Cpath%20d%3D%22M13.8543%2015.5C13.8543%2016.5311%2013.018%2017.3671%2011.9863%2017.3671C10.9545%2017.3671%2010.1183%2016.5311%2010.1183%2015.5C10.1183%2014.4689%2010.9545%2013.6329%2011.9863%2013.6329C13.018%2013.6329%2013.8543%2014.4689%2013.8543%2015.5Z%22%20stroke%3D%22%23FD4848%22%20stroke-width%3D%220.265831%22%2F%3E%3Cpath%20d%3D%22M13.1041%2015.5C13.1041%2016.1169%2012.6037%2016.6171%2011.9864%2016.6171C11.3691%2016.6171%2010.8688%2016.1169%2010.8688%2015.5C10.8688%2014.8831%2011.3691%2014.3829%2011.9864%2014.3829C12.6037%2014.3829%2013.1041%2014.8831%2013.1041%2015.5Z%22%20stroke%3D%22%23FD4848%22%20stroke-width%3D%220.265831%22%2F%3E%3Cpath%20d%3D%22M14.605%2015.5C14.605%2016.9453%2013.4327%2018.1171%2011.9866%2018.1171C10.5405%2018.1171%209.36827%2016.9453%209.36827%2015.5C9.36827%2014.0547%2010.5405%2012.8829%2011.9866%2012.8829C13.4327%2012.8829%2014.605%2014.0547%2014.605%2015.5Z%22%20stroke%3D%22%23FD4848%22%20stroke-width%3D%220.265831%22%2F%3E%3Cpath%20d%3D%22M15.3552%2015.5C15.3552%2017.3595%2013.847%2018.8671%2011.9865%2018.8671C10.1259%2018.8671%208.61778%2017.3595%208.61778%2015.5C8.61778%2013.6405%2010.1259%2012.1329%2011.9865%2012.1329C13.847%2012.1329%2015.3552%2013.6405%2015.3552%2015.5Z%22%20stroke%3D%22%23FD4848%22%20stroke-width%3D%220.265831%22%2F%3E%3Cpath%20d%3D%22M12.3534%2015.5C12.3534%2015.7027%2012.1891%2015.8671%2011.9863%2015.8671C11.7836%2015.8671%2011.6192%2015.7027%2011.6192%2015.5C11.6192%2015.2973%2011.7836%2015.1329%2011.9863%2015.1329C12.1891%2015.1329%2012.3534%2015.2973%2012.3534%2015.5Z%22%20fill%3D%22%23162BEB%22%20stroke%3D%22%23FD4848%22%20stroke-width%3D%220.265831%22%2F%3E%3Cellipse%20cx%3D%2211.9863%22%20cy%3D%2215.5%22%20rx%3D%220.5%22%20ry%3D%220.5%22%20fill%3D%22%23FD4848%22%2F%3E%3Cmask%20id%3D%22path-10-inside-1_4684_17034%22%20fill%3D%22white%22%3E%3Cpath%20d%3D%22M15.9885%2015.5C15.9885%2015.5%2014.1969%2018%2011.9867%2018C9.77655%2018%207.98486%2015.5%207.98486%2015.5C7.98486%2015.5%209.77655%2013%2011.9867%2013C14.1969%2013%2015.9885%2015.5%2015.9885%2015.5Z%22%2F%3E%3C%2Fmask%3E%3Cpath%20d%3D%22M15.9885%2015.5C15.9885%2015.5%2014.1969%2018%2011.9867%2018C9.77655%2018%207.98486%2015.5%207.98486%2015.5C7.98486%2015.5%209.77655%2013%2011.9867%2013C14.1969%2013%2015.9885%2015.5%2015.9885%2015.5Z%22%20stroke%3D%22%23D5FF5C%22%20stroke-width%3D%220.531663%22%20mask%3D%22url%28%23path-10-inside-1_4684_17034%29%22%2F%3E%3C%2Fsvg%3E";
var TalismanWallet = class extends BaseDotsamaWallet {
  constructor() {
    super(...arguments);
    this.extensionName = "talisman";
    this.title = "Talisman";
    this.installUrl = "https://talisman.xyz/download";
    this.noExtensionMessage = "You can use any Polkadot compatible wallet but we recommend using Talisman";
    this.logo = {
      src: logo$3,
      alt: "Talisman Logo"
    };
  }
};
var logo$2 = "data:image/svg+xml,%3C%3Fxml%20version%3D%221.0%22%20encoding%3D%22utf-8%22%20standalone%3D%22yes%22%3F%3E%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20xmlns%3Axlink%3D%22http%3A%2F%2Fwww.w3.org%2F1999%2Fxlink%22%20version%3D%221.1%22%20id%3D%22Layer_1%22%20x%3D%220px%22%20y%3D%220px%22%20viewBox%3D%2215%2015%20140%20140%22%20style%3D%22enable-background%3Anew%200%200%20170%20170%3Bzoom%3A%201%3B%22%20xml%3Aspace%3D%22preserve%22%3E%3Cstyle%20type%3D%22text%2Fcss%22%3E.bg0%7Bfill%3A%23FF8C00%7D%20.st0%7Bfill%3A%23FFFFFF%7D%3C%2Fstyle%3E%3Cg%3E%3Ccircle%20class%3D%22bg0%22%20cx%3D%2285%22%20cy%3D%2285%22%20r%3D%2270%22%3E%3C%2Fcircle%3E%3Cg%3E%3Cpath%20class%3D%22st0%22%20d%3D%22M85%2C34.7c-20.8%2C0-37.8%2C16.9-37.8%2C37.8c0%2C4.2%2C0.7%2C8.3%2C2%2C12.3c0.9%2C2.7%2C3.9%2C4.2%2C6.7%2C3.3c2.7-0.9%2C4.2-3.9%2C3.3-6.7%20c-1.1-3.1-1.6-6.4-1.5-9.7C58.1%2C57.6%2C69.5%2C46%2C83.6%2C45.3c15.7-0.8%2C28.7%2C11.7%2C28.7%2C27.2c0%2C14.5-11.4%2C26.4-25.7%2C27.2%20c0%2C0-5.3%2C0.3-7.9%2C0.7c-1.3%2C0.2-2.3%2C0.4-3%2C0.5c-0.3%2C0.1-0.6-0.2-0.5-0.5l0.9-4.4L81%2C73.4c0.6-2.8-1.2-5.6-4-6.2%20c-2.8-0.6-5.6%2C1.2-6.2%2C4c0%2C0-11.8%2C55-11.9%2C55.6c-0.6%2C2.8%2C1.2%2C5.6%2C4%2C6.2c2.8%2C0.6%2C5.6-1.2%2C6.2-4c0.1-0.6%2C1.7-7.9%2C1.7-7.9%20c1.2-5.6%2C5.8-9.7%2C11.2-10.4c1.2-0.2%2C5.9-0.5%2C5.9-0.5c19.5-1.5%2C34.9-17.8%2C34.9-37.7C122.8%2C51.6%2C105.8%2C34.7%2C85%2C34.7z%20M87.7%2C121.7%20c-3.4-0.7-6.8%2C1.4-7.5%2C4.9c-0.7%2C3.4%2C1.4%2C6.8%2C4.9%2C7.5c3.4%2C0.7%2C6.8-1.4%2C7.5-4.9C93.3%2C125.7%2C91.2%2C122.4%2C87.7%2C121.7z%22%3E%3C%2Fpath%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E";
var PolkadotjsWallet = class extends BaseDotsamaWallet {
  constructor() {
    super(...arguments);
    this.extensionName = "polkadot-js";
    this.title = "Polkadot.js";
    this.noExtensionMessage = "You can use any Polkadot compatible wallet but we recommend using Talisman";
    this.installUrl = "https://polkadot.js.org/extension/";
    this.logo = {
      src: logo$2,
      alt: "Polkadotjs Logo"
    };
  }
};
var logo$1 = "data:image/svg+xml,%3Csvg%20width%3D%22134%22%20height%3D%22134%22%20viewBox%3D%220%200%20134%20134%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%3Cmask%20id%3D%22mask0_699_5101%22%20style%3D%22mask-type%3Aalpha%22%20maskUnits%3D%22userSpaceOnUse%22%20x%3D%220%22%20y%3D%220%22%20width%3D%22134%22%20height%3D%22134%22%3E%3Crect%20width%3D%22134%22%20height%3D%22134%22%20fill%3D%22%23C4C4C4%22%2F%3E%3C%2Fmask%3E%3Cg%20mask%3D%22url%28%23mask0_699_5101%29%22%3E%3Cpath%20d%3D%22M87.9615%2064.3201L87.9456%2047.7455L27.1191%2016.2236V64.3041L66.0589%2085.106L80.2884%2078.8367L37.4403%2056.1046L37.4722%2037.887L87.9615%2064.3201Z%22%20fill%3D%22url%28%23paint0_linear_699_5101%29%22%2F%3E%3Cpath%20d%3D%22M50.7607%2044.8421V50.5052L37.3926%2056.2321L37.4883%2037.6636L50.7607%2044.8421Z%22%20fill%3D%22url%28%23paint1_linear_699_5101%29%22%2F%3E%3Cpath%20d%3D%22M50.8095%2091.822L80.2895%2078.8368L37.4414%2056.2163L50.6819%2050.5054L105.765%2079.2835L50.9212%20103.212L50.8095%2091.822Z%22%20fill%3D%22url%28%23paint2_linear_699_5101%29%22%2F%3E%3Cpath%20d%3D%22M37.4886%2087.9773L50.6493%2082.2982L50.9365%20103.196L105.765%2079.2832V97.118L37.377%20127.077L37.4886%2087.9773Z%22%20fill%3D%22url%28%23paint3_linear_699_5101%29%22%2F%3E%3Cpath%20d%3D%22M27.1191%2082.5857L37.4403%2087.9776L37.3765%20127.013L27.1191%20121.86V82.5857Z%22%20fill%3D%22url%28%23paint4_linear_699_5101%29%22%2F%3E%3Cpath%20d%3D%22M40.1522%2076.7791L50.6489%2082.2986L37.4403%2087.9776L27.1191%2082.5857L40.1522%2076.7791Z%22%20fill%3D%22url%28%23paint5_linear_699_5101%29%22%2F%3E%3Cpath%20d%3D%22M105.765%2056.5993L105.702%2039.9131L87.9785%2047.7457V64.3362L105.765%2056.5993Z%22%20fill%3D%22url%28%23paint6_linear_699_5101%29%22%2F%3E%3Cpath%20d%3D%22M27.1191%2016.2237L45.0337%207.97632L105.732%2039.8811L87.9775%2047.7456L27.1191%2016.2237Z%22%20fill%3D%22url%28%23paint7_linear_699_5101%29%22%2F%3E%3C%2Fg%3E%3Cdefs%3E%3ClinearGradient%20id%3D%22paint0_linear_699_5101%22%20x1%3D%2211.9006%22%20y1%3D%2250.6648%22%20x2%3D%22119.372%22%20y2%3D%2250.6648%22%20gradientUnits%3D%22userSpaceOnUse%22%3E%3Cstop%20stop-color%3D%22%23FFD4B2%22%2F%3E%3Cstop%20offset%3D%220.36%22%20stop-color%3D%22%239ACEB7%22%2F%3E%3Cstop%20offset%3D%220.67%22%20stop-color%3D%22%2347C8BB%22%2F%3E%3Cstop%20offset%3D%220.89%22%20stop-color%3D%22%2314C5BE%22%2F%3E%3Cstop%20offset%3D%221%22%20stop-color%3D%22%2300C4BF%22%2F%3E%3C%2FlinearGradient%3E%3ClinearGradient%20id%3D%22paint1_linear_699_5101%22%20x1%3D%2244.0766%22%20y1%3D%2262.8524%22%20x2%3D%2244.0766%22%20y2%3D%2221.2167%22%20gradientUnits%3D%22userSpaceOnUse%22%3E%3Cstop%20stop-color%3D%22%2300FECF%22%2F%3E%3Cstop%20offset%3D%220.08%22%20stop-color%3D%22%2300E5D0%22%2F%3E%3Cstop%20offset%3D%220.24%22%20stop-color%3D%22%2300A5D1%22%2F%3E%3Cstop%20offset%3D%220.48%22%20stop-color%3D%22%230040D4%22%2F%3E%3Cstop%20offset%3D%220.54%22%20stop-color%3D%22%230025D5%22%2F%3E%3Cstop%20offset%3D%221%22%2F%3E%3C%2FlinearGradient%3E%3ClinearGradient%20id%3D%22paint2_linear_699_5101%22%20x1%3D%2237.4414%22%20y1%3D%2276.8587%22%20x2%3D%22146.891%22%20y2%3D%2276.8587%22%20gradientUnits%3D%22userSpaceOnUse%22%3E%3Cstop%20stop-color%3D%22%23FDEC9F%22%2F%3E%3Cstop%20offset%3D%220.08%22%20stop-color%3D%22%23E4D8A4%22%2F%3E%3Cstop%20offset%3D%220.24%22%20stop-color%3D%22%23A4A6B2%22%2F%3E%3Cstop%20offset%3D%220.47%22%20stop-color%3D%22%233F57C8%22%2F%3E%3Cstop%20offset%3D%220.61%22%20stop-color%3D%22%230025D5%22%2F%3E%3Cstop%20offset%3D%221%22%2F%3E%3C%2FlinearGradient%3E%3ClinearGradient%20id%3D%22paint3_linear_699_5101%22%20x1%3D%2215.0596%22%20y1%3D%22103.18%22%20x2%3D%22155.01%22%20y2%3D%22103.18%22%20gradientUnits%3D%22userSpaceOnUse%22%3E%3Cstop%20offset%3D%220.05%22%20stop-color%3D%22%2362A5FF%22%2F%3E%3Cstop%20offset%3D%220.45%22%20stop-color%3D%22%231032D1%22%2F%3E%3Cstop%20offset%3D%221%22%2F%3E%3C%2FlinearGradient%3E%3ClinearGradient%20id%3D%22paint4_linear_699_5101%22%20x1%3D%22628.741%22%20y1%3D%223244.93%22%20x2%3D%22797.782%22%20y2%3D%223247.12%22%20gradientUnits%3D%22userSpaceOnUse%22%3E%3Cstop%20stop-color%3D%22%23FFD4B2%22%2F%3E%3Cstop%20offset%3D%220.36%22%20stop-color%3D%22%239ACEB7%22%2F%3E%3Cstop%20offset%3D%220.67%22%20stop-color%3D%22%2347C8BB%22%2F%3E%3Cstop%20offset%3D%220.89%22%20stop-color%3D%22%2314C5BE%22%2F%3E%3Cstop%20offset%3D%221%22%20stop-color%3D%22%2300C4BF%22%2F%3E%3C%2FlinearGradient%3E%3ClinearGradient%20id%3D%22paint5_linear_699_5101%22%20x1%3D%2224.5987%22%20y1%3D%2282.3783%22%20x2%3D%2272.5834%22%20y2%3D%2282.3783%22%20gradientUnits%3D%22userSpaceOnUse%22%3E%3Cstop%20stop-color%3D%22%2300FECF%22%2F%3E%3Cstop%20offset%3D%220.08%22%20stop-color%3D%22%2300E5D0%22%2F%3E%3Cstop%20offset%3D%220.25%22%20stop-color%3D%22%2300A5D1%22%2F%3E%3Cstop%20offset%3D%220.49%22%20stop-color%3D%22%230040D4%22%2F%3E%3Cstop%20offset%3D%220.56%22%20stop-color%3D%22%230025D5%22%2F%3E%3C%2FlinearGradient%3E%3ClinearGradient%20id%3D%22paint6_linear_699_5101%22%20x1%3D%2270.9573%22%20y1%3D%2252.5952%22%20x2%3D%22189.069%22%20y2%3D%2250.4576%22%20gradientUnits%3D%22userSpaceOnUse%22%3E%3Cstop%20stop-color%3D%22%2300FECF%22%2F%3E%3Cstop%20offset%3D%220.05%22%20stop-color%3D%22%2300E5D0%22%2F%3E%3Cstop%20offset%3D%220.15%22%20stop-color%3D%22%2300A5D1%22%2F%3E%3Cstop%20offset%3D%220.29%22%20stop-color%3D%22%230040D4%22%2F%3E%3Cstop%20offset%3D%220.33%22%20stop-color%3D%22%230025D5%22%2F%3E%3C%2FlinearGradient%3E%3ClinearGradient%20id%3D%22paint7_linear_699_5101%22%20x1%3D%2227.1191%22%20y1%3D%2227.8689%22%20x2%3D%22173.642%22%20y2%3D%2227.8689%22%20gradientUnits%3D%22userSpaceOnUse%22%3E%3Cstop%20stop-color%3D%22%23FFD4AF%22%2F%3E%3Cstop%20offset%3D%220.1%22%20stop-color%3D%22%23E6D5BA%22%2F%3E%3Cstop%20offset%3D%220.31%22%20stop-color%3D%22%23A7D6D5%22%2F%3E%3Cstop%20offset%3D%220.61%22%20stop-color%3D%22%2343D9FF%22%2F%3E%3Cstop%20offset%3D%220.63%22%20stop-color%3D%22%2337B1D0%22%2F%3E%3Cstop%20offset%3D%220.65%22%20stop-color%3D%22%232B8CA5%22%2F%3E%3Cstop%20offset%3D%220.67%22%20stop-color%3D%22%23216B7D%22%2F%3E%3Cstop%20offset%3D%220.7%22%20stop-color%3D%22%23184E5B%22%2F%3E%3Cstop%20offset%3D%220.72%22%20stop-color%3D%22%2310353F%22%2F%3E%3Cstop%20offset%3D%220.75%22%20stop-color%3D%22%230A2228%22%2F%3E%3Cstop%20offset%3D%220.78%22%20stop-color%3D%22%23061316%22%2F%3E%3Cstop%20offset%3D%220.82%22%20stop-color%3D%22%23020809%22%2F%3E%3Cstop%20offset%3D%220.88%22%20stop-color%3D%22%23010202%22%2F%3E%3Cstop%20offset%3D%221%22%2F%3E%3C%2FlinearGradient%3E%3C%2Fdefs%3E%3C%2Fsvg%3E";
var SubWallet = class extends BaseDotsamaWallet {
  constructor() {
    super(...arguments);
    this.extensionName = "subwallet-js";
    this.title = "SubWallet";
    this.installUrl = "https://chrome.google.com/webstore/detail/subwallet/onhogfjeacnfoofkfgppdlbmlmnplgbn?hl=en&authuser=0";
    this.noExtensionMessage = "You can use any Polkadot compatible wallet but we recommend using Talisman";
    this.logo = {
      src: logo$1,
      alt: "Subwallet Logo"
    };
  }
};
var logo = "data:image/svg+xml,%3Csvg%20id%3D%22Layer_1%22%20data-name%3D%22Layer%201%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20xmlns%3Axlink%3D%22http%3A%2F%2Fwww.w3.org%2F1999%2Fxlink%22%20viewBox%3D%220%200%20120%20120%22%3E%3Cdefs%3E%3Cstyle%3E.cls-1%7Bfill-rule%3Aevenodd%3Bfill%3Aurl%28%23radial-gradient%29%3B%7D%3C%2Fstyle%3E%3CradialGradient%20id%3D%22radial-gradient%22%20cx%3D%22-164.98%22%20cy%3D%22566.58%22%20r%3D%221%22%20gradientTransform%3D%22matrix%28-15%2C%20136.5%2C%20135.36%2C%2014.88%2C%20-79032.62%2C%2014076.16%29%22%20gradientUnits%3D%22userSpaceOnUse%22%3E%3Cstop%20offset%3D%220%22%20stop-color%3D%22%23704bff%22%2F%3E%3Cstop%20offset%3D%220.21%22%20stop-color%3D%22%23c549ff%22%2F%3E%3C%2FradialGradient%3E%3C%2Fdefs%3E%3Ctitle%3Eenkrypt-logo-color%3C%2Ftitle%3E%3Cpath%20class%3D%22cls-1%22%20d%3D%22M0%2C23.09A23.09%2C23.09%2C0%2C0%2C1%2C23.09%2C0H120V17.72a16.6%2C16.6%2C0%2C0%2C1-16.6%2C16.6H57.84A23.09%2C23.09%2C0%2C0%2C0%2C34.75%2C57.41v5.87a23.1%2C23.1%2C0%2C0%2C0%2C23.09%2C23.1H103.4A16.6%2C16.6%2C0%2C0%2C1%2C120%2C103v17H23.09A23.09%2C23.09%2C0%2C0%2C1%2C0%2C96.91ZM58.84%2C44h46.73A14.43%2C14.43%2C0%2C0%2C1%2C120%2C58.4v3.89a14.43%2C14.43%2C0%2C0%2C1-14.43%2C14.43H58.84A14.44%2C14.44%2C0%2C0%2C1%2C44.4%2C62.29V58.4A14.44%2C14.44%2C0%2C0%2C1%2C58.84%2C44Z%22%2F%3E%3C%2Fsvg%3E";
var EnkryptWallet = class extends BaseDotsamaWallet {
  constructor() {
    super(...arguments);
    this.extensionName = "enkrypt";
    this.title = "Enkrypt";
    this.installUrl = "https://www.enkrypt.com/#overview";
    this.noExtensionMessage = "You can use any Polkadot compatible wallet but we recommend using Talisman";
    this.logo = {
      src: logo,
      alt: "Enkrypt Logo"
    };
  }
};
var supportedWallets = [
  new TalismanWallet(),
  new SubWallet(),
  new PolkadotjsWallet(),
  new EnkryptWallet()
];

// src/utils/helpers/getExpiredItem.ts
var FIVE_SECONDS = 5e3;
function getExpiredItem(items, expirationPeriod) {
  if (expirationPeriod === 0)
    return [];
  const timeFromCreation = (creationTime) => Date.now() - creationTime;
  return items.filter(
    (item) => timeFromCreation(item.createdAt) >= (expirationPeriod || FIVE_SECONDS)
  );
}

// src/utils/helpers/pseudoRandomId.ts
var import_nanoid = require("nanoid");
var pseudoRandomId = (t = 21) => (0, import_nanoid.nanoid)(t);

// src/notifications/hooks/useNotifications.ts
var import_react4 = require("react");
var useNotifications = () => {
  const { addNotification, notifications, removeNotification, config } = (0, import_react4.useContext)(NotificationsContext);
  const chainNotifications = (0, import_react4.useMemo)(() => {
    return notifications ?? [];
  }, [notifications]);
  useInterval(() => {
    if (config?.expiration === 0)
      return;
    const expiredNotifications = getExpiredItem(
      chainNotifications,
      config?.expiration
    );
    for (const notification of expiredNotifications) {
      removeNotification(notification.id);
    }
  }, config?.checkInterval || HALF_A_SECOND);
  return {
    notifications: chainNotifications,
    addNotification,
    removeNotification
  };
};

// src/notifications/hooks/useTxNotifications.ts
var import_react5 = require("react");
function useTxNotifications(tx, chain) {
  const { addNotification } = useNotifications();
  (0, import_react5.useEffect)(() => {
    if (["Ready", "None"].includes(tx.status))
      return;
    addNotification({
      type: tx.status,
      message: tx.status,
      result: tx.result,
      chain
    });
  }, [tx.status]);
}

// src/react/hooks/internal/useIsMounted.ts
var import_react6 = __toESM(require("react"));
function useIsMounted() {
  const isMounted = import_react6.default.useRef(false);
  import_react6.default.useEffect(() => {
    isMounted.current = true;
    return () => {
      isMounted.current = false;
    };
  }, []);
  return import_react6.default.useCallback(() => isMounted.current, []);
}

// src/notifications/reducer.ts
function notificationReducer(state, action) {
  const chainState = state ?? [];
  switch (action.type) {
    case "ADD_NOTIFICATION":
      return [...state, action.notification];
    case "REMOVE_NOTIFICATION": {
      return chainState.filter((n) => n.id !== action.notificationId);
    }
  }
}

// src/notifications/provider.tsx
var import_react7 = __toESM(require("react"));
var NotificationsProvider = ({ children, config }) => {
  const [notifications, dispatch] = (0, import_react7.useReducer)(
    notificationReducer,
    DEFAULT_NOTIFICATIONS
  );
  const isMounted = useIsMounted();
  const addNotification = (0, import_react7.useCallback)(
    (notification) => {
      if (isMounted()) {
        dispatch({
          type: "ADD_NOTIFICATION",
          notification: {
            ...notification,
            id: pseudoRandomId(),
            createdAt: Date.now()
          }
        });
      }
    },
    [dispatch]
  );
  const removeNotification = (0, import_react7.useCallback)(
    (notificationId) => {
      if (isMounted()) {
        dispatch({
          type: "REMOVE_NOTIFICATION",
          notificationId
        });
      }
    },
    [dispatch]
  );
  return /* @__PURE__ */ import_react7.default.createElement(
    NotificationsContext.Provider,
    {
      value: { addNotification, notifications, removeNotification, config },
      children
    }
  );
};

// src/notifications/utils/toNotificationLevel.ts
var toNotificationLevel = (type) => {
  switch (type) {
    case "None":
      return "info";
    case "DryRun":
      return "info";
    case "PendingSignature":
      return "info";
    case "Future":
      return "info";
    case "Ready":
      return "info";
    case "Broadcast":
      return "info";
    case "InBlock":
      return "success";
    case "Retracted":
      return "warning";
    case "FinalityTimeout":
      return "error";
    case "Finalized":
      return "success";
    case "Usurped":
      return "error";
    case "Dropped":
      return "error";
    case "Invalid":
      return "warning";
    case "Errored":
      return "error";
    case "WalletConnected":
      return "info";
    case "WalletDisconnected":
      return "info";
    default:
      return "info";
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  DEFAULT_NOTIFICATIONS,
  NotificationsContext,
  NotificationsProvider,
  toNotificationLevel,
  useNotifications,
  useTxNotifications
});
