// useink/dist/utils.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __commonJS = (cb, mod) => function __require() {
  return mod || (0, cb[__getOwnPropNames(cb)[0]])((mod = { exports: {} }).exports, mod), mod.exports;
};
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// ../../node_modules/.pnpm/bn.js@5.2.1/node_modules/bn.js/lib/bn.js
var require_bn = __commonJS({
  "../../node_modules/.pnpm/bn.js@5.2.1/node_modules/bn.js/lib/bn.js"(exports, module2) {
    (function(module3, exports2) {
      "use strict";
      function assert2(val, msg) {
        if (!val)
          throw new Error(msg || "Assertion failed");
      }
      function inherits(ctor, superCtor) {
        ctor.super_ = superCtor;
        var TempCtor = function() {
        };
        TempCtor.prototype = superCtor.prototype;
        ctor.prototype = new TempCtor();
        ctor.prototype.constructor = ctor;
      }
      function BN2(number, base, endian) {
        if (BN2.isBN(number)) {
          return number;
        }
        this.negative = 0;
        this.words = null;
        this.length = 0;
        this.red = null;
        if (number !== null) {
          if (base === "le" || base === "be") {
            endian = base;
            base = 10;
          }
          this._init(number || 0, base || 10, endian || "be");
        }
      }
      if (typeof module3 === "object") {
        module3.exports = BN2;
      } else {
        exports2.BN = BN2;
      }
      BN2.BN = BN2;
      BN2.wordSize = 26;
      var Buffer2;
      try {
        if (typeof window !== "undefined" && typeof window.Buffer !== "undefined") {
          Buffer2 = window.Buffer;
        } else {
          Buffer2 = require("buffer").Buffer;
        }
      } catch (e) {
      }
      BN2.isBN = function isBN(num) {
        if (num instanceof BN2) {
          return true;
        }
        return num !== null && typeof num === "object" && num.constructor.wordSize === BN2.wordSize && Array.isArray(num.words);
      };
      BN2.max = function max(left, right) {
        if (left.cmp(right) > 0)
          return left;
        return right;
      };
      BN2.min = function min(left, right) {
        if (left.cmp(right) < 0)
          return left;
        return right;
      };
      BN2.prototype._init = function init(number, base, endian) {
        if (typeof number === "number") {
          return this._initNumber(number, base, endian);
        }
        if (typeof number === "object") {
          return this._initArray(number, base, endian);
        }
        if (base === "hex") {
          base = 16;
        }
        assert2(base === (base | 0) && base >= 2 && base <= 36);
        number = number.toString().replace(/\s+/g, "");
        var start = 0;
        if (number[0] === "-") {
          start++;
          this.negative = 1;
        }
        if (start < number.length) {
          if (base === 16) {
            this._parseHex(number, start, endian);
          } else {
            this._parseBase(number, base, start);
            if (endian === "le") {
              this._initArray(this.toArray(), base, endian);
            }
          }
        }
      };
      BN2.prototype._initNumber = function _initNumber(number, base, endian) {
        if (number < 0) {
          this.negative = 1;
          number = -number;
        }
        if (number < 67108864) {
          this.words = [number & 67108863];
          this.length = 1;
        } else if (number < 4503599627370496) {
          this.words = [
            number & 67108863,
            number / 67108864 & 67108863
          ];
          this.length = 2;
        } else {
          assert2(number < 9007199254740992);
          this.words = [
            number & 67108863,
            number / 67108864 & 67108863,
            1
          ];
          this.length = 3;
        }
        if (endian !== "le")
          return;
        this._initArray(this.toArray(), base, endian);
      };
      BN2.prototype._initArray = function _initArray(number, base, endian) {
        assert2(typeof number.length === "number");
        if (number.length <= 0) {
          this.words = [0];
          this.length = 1;
          return this;
        }
        this.length = Math.ceil(number.length / 3);
        this.words = new Array(this.length);
        for (var i = 0; i < this.length; i++) {
          this.words[i] = 0;
        }
        var j, w;
        var off = 0;
        if (endian === "be") {
          for (i = number.length - 1, j = 0; i >= 0; i -= 3) {
            w = number[i] | number[i - 1] << 8 | number[i - 2] << 16;
            this.words[j] |= w << off & 67108863;
            this.words[j + 1] = w >>> 26 - off & 67108863;
            off += 24;
            if (off >= 26) {
              off -= 26;
              j++;
            }
          }
        } else if (endian === "le") {
          for (i = 0, j = 0; i < number.length; i += 3) {
            w = number[i] | number[i + 1] << 8 | number[i + 2] << 16;
            this.words[j] |= w << off & 67108863;
            this.words[j + 1] = w >>> 26 - off & 67108863;
            off += 24;
            if (off >= 26) {
              off -= 26;
              j++;
            }
          }
        }
        return this._strip();
      };
      function parseHex4Bits(string, index) {
        var c = string.charCodeAt(index);
        if (c >= 48 && c <= 57) {
          return c - 48;
        } else if (c >= 65 && c <= 70) {
          return c - 55;
        } else if (c >= 97 && c <= 102) {
          return c - 87;
        } else {
          assert2(false, "Invalid character in " + string);
        }
      }
      function parseHexByte(string, lowerBound, index) {
        var r = parseHex4Bits(string, index);
        if (index - 1 >= lowerBound) {
          r |= parseHex4Bits(string, index - 1) << 4;
        }
        return r;
      }
      BN2.prototype._parseHex = function _parseHex(number, start, endian) {
        this.length = Math.ceil((number.length - start) / 6);
        this.words = new Array(this.length);
        for (var i = 0; i < this.length; i++) {
          this.words[i] = 0;
        }
        var off = 0;
        var j = 0;
        var w;
        if (endian === "be") {
          for (i = number.length - 1; i >= start; i -= 2) {
            w = parseHexByte(number, start, i) << off;
            this.words[j] |= w & 67108863;
            if (off >= 18) {
              off -= 18;
              j += 1;
              this.words[j] |= w >>> 26;
            } else {
              off += 8;
            }
          }
        } else {
          var parseLength = number.length - start;
          for (i = parseLength % 2 === 0 ? start + 1 : start; i < number.length; i += 2) {
            w = parseHexByte(number, start, i) << off;
            this.words[j] |= w & 67108863;
            if (off >= 18) {
              off -= 18;
              j += 1;
              this.words[j] |= w >>> 26;
            } else {
              off += 8;
            }
          }
        }
        this._strip();
      };
      function parseBase(str, start, end, mul) {
        var r = 0;
        var b = 0;
        var len = Math.min(str.length, end);
        for (var i = start; i < len; i++) {
          var c = str.charCodeAt(i) - 48;
          r *= mul;
          if (c >= 49) {
            b = c - 49 + 10;
          } else if (c >= 17) {
            b = c - 17 + 10;
          } else {
            b = c;
          }
          assert2(c >= 0 && b < mul, "Invalid character");
          r += b;
        }
        return r;
      }
      BN2.prototype._parseBase = function _parseBase(number, base, start) {
        this.words = [0];
        this.length = 1;
        for (var limbLen = 0, limbPow = 1; limbPow <= 67108863; limbPow *= base) {
          limbLen++;
        }
        limbLen--;
        limbPow = limbPow / base | 0;
        var total = number.length - start;
        var mod = total % limbLen;
        var end = Math.min(total, total - mod) + start;
        var word = 0;
        for (var i = start; i < end; i += limbLen) {
          word = parseBase(number, i, i + limbLen, base);
          this.imuln(limbPow);
          if (this.words[0] + word < 67108864) {
            this.words[0] += word;
          } else {
            this._iaddn(word);
          }
        }
        if (mod !== 0) {
          var pow = 1;
          word = parseBase(number, i, number.length, base);
          for (i = 0; i < mod; i++) {
            pow *= base;
          }
          this.imuln(pow);
          if (this.words[0] + word < 67108864) {
            this.words[0] += word;
          } else {
            this._iaddn(word);
          }
        }
        this._strip();
      };
      BN2.prototype.copy = function copy(dest) {
        dest.words = new Array(this.length);
        for (var i = 0; i < this.length; i++) {
          dest.words[i] = this.words[i];
        }
        dest.length = this.length;
        dest.negative = this.negative;
        dest.red = this.red;
      };
      function move(dest, src) {
        dest.words = src.words;
        dest.length = src.length;
        dest.negative = src.negative;
        dest.red = src.red;
      }
      BN2.prototype._move = function _move(dest) {
        move(dest, this);
      };
      BN2.prototype.clone = function clone() {
        var r = new BN2(null);
        this.copy(r);
        return r;
      };
      BN2.prototype._expand = function _expand(size) {
        while (this.length < size) {
          this.words[this.length++] = 0;
        }
        return this;
      };
      BN2.prototype._strip = function strip() {
        while (this.length > 1 && this.words[this.length - 1] === 0) {
          this.length--;
        }
        return this._normSign();
      };
      BN2.prototype._normSign = function _normSign() {
        if (this.length === 1 && this.words[0] === 0) {
          this.negative = 0;
        }
        return this;
      };
      if (typeof Symbol !== "undefined" && typeof Symbol.for === "function") {
        try {
          BN2.prototype[Symbol.for("nodejs.util.inspect.custom")] = inspect;
        } catch (e) {
          BN2.prototype.inspect = inspect;
        }
      } else {
        BN2.prototype.inspect = inspect;
      }
      function inspect() {
        return (this.red ? "<BN-R: " : "<BN: ") + this.toString(16) + ">";
      }
      var zeros = [
        "",
        "0",
        "00",
        "000",
        "0000",
        "00000",
        "000000",
        "0000000",
        "00000000",
        "000000000",
        "0000000000",
        "00000000000",
        "000000000000",
        "0000000000000",
        "00000000000000",
        "000000000000000",
        "0000000000000000",
        "00000000000000000",
        "000000000000000000",
        "0000000000000000000",
        "00000000000000000000",
        "000000000000000000000",
        "0000000000000000000000",
        "00000000000000000000000",
        "000000000000000000000000",
        "0000000000000000000000000"
      ];
      var groupSizes = [
        0,
        0,
        25,
        16,
        12,
        11,
        10,
        9,
        8,
        8,
        7,
        7,
        7,
        7,
        6,
        6,
        6,
        6,
        6,
        6,
        6,
        5,
        5,
        5,
        5,
        5,
        5,
        5,
        5,
        5,
        5,
        5,
        5,
        5,
        5,
        5,
        5
      ];
      var groupBases = [
        0,
        0,
        33554432,
        43046721,
        16777216,
        48828125,
        60466176,
        40353607,
        16777216,
        43046721,
        1e7,
        19487171,
        35831808,
        62748517,
        7529536,
        11390625,
        16777216,
        24137569,
        34012224,
        47045881,
        64e6,
        4084101,
        5153632,
        6436343,
        7962624,
        9765625,
        11881376,
        14348907,
        17210368,
        20511149,
        243e5,
        28629151,
        33554432,
        39135393,
        45435424,
        52521875,
        60466176
      ];
      BN2.prototype.toString = function toString(base, padding) {
        base = base || 10;
        padding = padding | 0 || 1;
        var out;
        if (base === 16 || base === "hex") {
          out = "";
          var off = 0;
          var carry = 0;
          for (var i = 0; i < this.length; i++) {
            var w = this.words[i];
            var word = ((w << off | carry) & 16777215).toString(16);
            carry = w >>> 24 - off & 16777215;
            off += 2;
            if (off >= 26) {
              off -= 26;
              i--;
            }
            if (carry !== 0 || i !== this.length - 1) {
              out = zeros[6 - word.length] + word + out;
            } else {
              out = word + out;
            }
          }
          if (carry !== 0) {
            out = carry.toString(16) + out;
          }
          while (out.length % padding !== 0) {
            out = "0" + out;
          }
          if (this.negative !== 0) {
            out = "-" + out;
          }
          return out;
        }
        if (base === (base | 0) && base >= 2 && base <= 36) {
          var groupSize = groupSizes[base];
          var groupBase = groupBases[base];
          out = "";
          var c = this.clone();
          c.negative = 0;
          while (!c.isZero()) {
            var r = c.modrn(groupBase).toString(base);
            c = c.idivn(groupBase);
            if (!c.isZero()) {
              out = zeros[groupSize - r.length] + r + out;
            } else {
              out = r + out;
            }
          }
          if (this.isZero()) {
            out = "0" + out;
          }
          while (out.length % padding !== 0) {
            out = "0" + out;
          }
          if (this.negative !== 0) {
            out = "-" + out;
          }
          return out;
        }
        assert2(false, "Base should be between 2 and 36");
      };
      BN2.prototype.toNumber = function toNumber() {
        var ret = this.words[0];
        if (this.length === 2) {
          ret += this.words[1] * 67108864;
        } else if (this.length === 3 && this.words[2] === 1) {
          ret += 4503599627370496 + this.words[1] * 67108864;
        } else if (this.length > 2) {
          assert2(false, "Number can only safely store up to 53 bits");
        }
        return this.negative !== 0 ? -ret : ret;
      };
      BN2.prototype.toJSON = function toJSON() {
        return this.toString(16, 2);
      };
      if (Buffer2) {
        BN2.prototype.toBuffer = function toBuffer(endian, length) {
          return this.toArrayLike(Buffer2, endian, length);
        };
      }
      BN2.prototype.toArray = function toArray(endian, length) {
        return this.toArrayLike(Array, endian, length);
      };
      var allocate = function allocate2(ArrayType, size) {
        if (ArrayType.allocUnsafe) {
          return ArrayType.allocUnsafe(size);
        }
        return new ArrayType(size);
      };
      BN2.prototype.toArrayLike = function toArrayLike(ArrayType, endian, length) {
        this._strip();
        var byteLength = this.byteLength();
        var reqLength = length || Math.max(1, byteLength);
        assert2(byteLength <= reqLength, "byte array longer than desired length");
        assert2(reqLength > 0, "Requested array length <= 0");
        var res = allocate(ArrayType, reqLength);
        var postfix = endian === "le" ? "LE" : "BE";
        this["_toArrayLike" + postfix](res, byteLength);
        return res;
      };
      BN2.prototype._toArrayLikeLE = function _toArrayLikeLE(res, byteLength) {
        var position = 0;
        var carry = 0;
        for (var i = 0, shift = 0; i < this.length; i++) {
          var word = this.words[i] << shift | carry;
          res[position++] = word & 255;
          if (position < res.length) {
            res[position++] = word >> 8 & 255;
          }
          if (position < res.length) {
            res[position++] = word >> 16 & 255;
          }
          if (shift === 6) {
            if (position < res.length) {
              res[position++] = word >> 24 & 255;
            }
            carry = 0;
            shift = 0;
          } else {
            carry = word >>> 24;
            shift += 2;
          }
        }
        if (position < res.length) {
          res[position++] = carry;
          while (position < res.length) {
            res[position++] = 0;
          }
        }
      };
      BN2.prototype._toArrayLikeBE = function _toArrayLikeBE(res, byteLength) {
        var position = res.length - 1;
        var carry = 0;
        for (var i = 0, shift = 0; i < this.length; i++) {
          var word = this.words[i] << shift | carry;
          res[position--] = word & 255;
          if (position >= 0) {
            res[position--] = word >> 8 & 255;
          }
          if (position >= 0) {
            res[position--] = word >> 16 & 255;
          }
          if (shift === 6) {
            if (position >= 0) {
              res[position--] = word >> 24 & 255;
            }
            carry = 0;
            shift = 0;
          } else {
            carry = word >>> 24;
            shift += 2;
          }
        }
        if (position >= 0) {
          res[position--] = carry;
          while (position >= 0) {
            res[position--] = 0;
          }
        }
      };
      if (Math.clz32) {
        BN2.prototype._countBits = function _countBits(w) {
          return 32 - Math.clz32(w);
        };
      } else {
        BN2.prototype._countBits = function _countBits(w) {
          var t = w;
          var r = 0;
          if (t >= 4096) {
            r += 13;
            t >>>= 13;
          }
          if (t >= 64) {
            r += 7;
            t >>>= 7;
          }
          if (t >= 8) {
            r += 4;
            t >>>= 4;
          }
          if (t >= 2) {
            r += 2;
            t >>>= 2;
          }
          return r + t;
        };
      }
      BN2.prototype._zeroBits = function _zeroBits(w) {
        if (w === 0)
          return 26;
        var t = w;
        var r = 0;
        if ((t & 8191) === 0) {
          r += 13;
          t >>>= 13;
        }
        if ((t & 127) === 0) {
          r += 7;
          t >>>= 7;
        }
        if ((t & 15) === 0) {
          r += 4;
          t >>>= 4;
        }
        if ((t & 3) === 0) {
          r += 2;
          t >>>= 2;
        }
        if ((t & 1) === 0) {
          r++;
        }
        return r;
      };
      BN2.prototype.bitLength = function bitLength() {
        var w = this.words[this.length - 1];
        var hi = this._countBits(w);
        return (this.length - 1) * 26 + hi;
      };
      function toBitArray(num) {
        var w = new Array(num.bitLength());
        for (var bit = 0; bit < w.length; bit++) {
          var off = bit / 26 | 0;
          var wbit = bit % 26;
          w[bit] = num.words[off] >>> wbit & 1;
        }
        return w;
      }
      BN2.prototype.zeroBits = function zeroBits() {
        if (this.isZero())
          return 0;
        var r = 0;
        for (var i = 0; i < this.length; i++) {
          var b = this._zeroBits(this.words[i]);
          r += b;
          if (b !== 26)
            break;
        }
        return r;
      };
      BN2.prototype.byteLength = function byteLength() {
        return Math.ceil(this.bitLength() / 8);
      };
      BN2.prototype.toTwos = function toTwos(width) {
        if (this.negative !== 0) {
          return this.abs().inotn(width).iaddn(1);
        }
        return this.clone();
      };
      BN2.prototype.fromTwos = function fromTwos(width) {
        if (this.testn(width - 1)) {
          return this.notn(width).iaddn(1).ineg();
        }
        return this.clone();
      };
      BN2.prototype.isNeg = function isNeg() {
        return this.negative !== 0;
      };
      BN2.prototype.neg = function neg() {
        return this.clone().ineg();
      };
      BN2.prototype.ineg = function ineg() {
        if (!this.isZero()) {
          this.negative ^= 1;
        }
        return this;
      };
      BN2.prototype.iuor = function iuor(num) {
        while (this.length < num.length) {
          this.words[this.length++] = 0;
        }
        for (var i = 0; i < num.length; i++) {
          this.words[i] = this.words[i] | num.words[i];
        }
        return this._strip();
      };
      BN2.prototype.ior = function ior(num) {
        assert2((this.negative | num.negative) === 0);
        return this.iuor(num);
      };
      BN2.prototype.or = function or(num) {
        if (this.length > num.length)
          return this.clone().ior(num);
        return num.clone().ior(this);
      };
      BN2.prototype.uor = function uor(num) {
        if (this.length > num.length)
          return this.clone().iuor(num);
        return num.clone().iuor(this);
      };
      BN2.prototype.iuand = function iuand(num) {
        var b;
        if (this.length > num.length) {
          b = num;
        } else {
          b = this;
        }
        for (var i = 0; i < b.length; i++) {
          this.words[i] = this.words[i] & num.words[i];
        }
        this.length = b.length;
        return this._strip();
      };
      BN2.prototype.iand = function iand(num) {
        assert2((this.negative | num.negative) === 0);
        return this.iuand(num);
      };
      BN2.prototype.and = function and(num) {
        if (this.length > num.length)
          return this.clone().iand(num);
        return num.clone().iand(this);
      };
      BN2.prototype.uand = function uand(num) {
        if (this.length > num.length)
          return this.clone().iuand(num);
        return num.clone().iuand(this);
      };
      BN2.prototype.iuxor = function iuxor(num) {
        var a;
        var b;
        if (this.length > num.length) {
          a = this;
          b = num;
        } else {
          a = num;
          b = this;
        }
        for (var i = 0; i < b.length; i++) {
          this.words[i] = a.words[i] ^ b.words[i];
        }
        if (this !== a) {
          for (; i < a.length; i++) {
            this.words[i] = a.words[i];
          }
        }
        this.length = a.length;
        return this._strip();
      };
      BN2.prototype.ixor = function ixor(num) {
        assert2((this.negative | num.negative) === 0);
        return this.iuxor(num);
      };
      BN2.prototype.xor = function xor(num) {
        if (this.length > num.length)
          return this.clone().ixor(num);
        return num.clone().ixor(this);
      };
      BN2.prototype.uxor = function uxor(num) {
        if (this.length > num.length)
          return this.clone().iuxor(num);
        return num.clone().iuxor(this);
      };
      BN2.prototype.inotn = function inotn(width) {
        assert2(typeof width === "number" && width >= 0);
        var bytesNeeded = Math.ceil(width / 26) | 0;
        var bitsLeft = width % 26;
        this._expand(bytesNeeded);
        if (bitsLeft > 0) {
          bytesNeeded--;
        }
        for (var i = 0; i < bytesNeeded; i++) {
          this.words[i] = ~this.words[i] & 67108863;
        }
        if (bitsLeft > 0) {
          this.words[i] = ~this.words[i] & 67108863 >> 26 - bitsLeft;
        }
        return this._strip();
      };
      BN2.prototype.notn = function notn(width) {
        return this.clone().inotn(width);
      };
      BN2.prototype.setn = function setn(bit, val) {
        assert2(typeof bit === "number" && bit >= 0);
        var off = bit / 26 | 0;
        var wbit = bit % 26;
        this._expand(off + 1);
        if (val) {
          this.words[off] = this.words[off] | 1 << wbit;
        } else {
          this.words[off] = this.words[off] & ~(1 << wbit);
        }
        return this._strip();
      };
      BN2.prototype.iadd = function iadd(num) {
        var r;
        if (this.negative !== 0 && num.negative === 0) {
          this.negative = 0;
          r = this.isub(num);
          this.negative ^= 1;
          return this._normSign();
        } else if (this.negative === 0 && num.negative !== 0) {
          num.negative = 0;
          r = this.isub(num);
          num.negative = 1;
          return r._normSign();
        }
        var a, b;
        if (this.length > num.length) {
          a = this;
          b = num;
        } else {
          a = num;
          b = this;
        }
        var carry = 0;
        for (var i = 0; i < b.length; i++) {
          r = (a.words[i] | 0) + (b.words[i] | 0) + carry;
          this.words[i] = r & 67108863;
          carry = r >>> 26;
        }
        for (; carry !== 0 && i < a.length; i++) {
          r = (a.words[i] | 0) + carry;
          this.words[i] = r & 67108863;
          carry = r >>> 26;
        }
        this.length = a.length;
        if (carry !== 0) {
          this.words[this.length] = carry;
          this.length++;
        } else if (a !== this) {
          for (; i < a.length; i++) {
            this.words[i] = a.words[i];
          }
        }
        return this;
      };
      BN2.prototype.add = function add2(num) {
        var res;
        if (num.negative !== 0 && this.negative === 0) {
          num.negative = 0;
          res = this.sub(num);
          num.negative ^= 1;
          return res;
        } else if (num.negative === 0 && this.negative !== 0) {
          this.negative = 0;
          res = num.sub(this);
          this.negative = 1;
          return res;
        }
        if (this.length > num.length)
          return this.clone().iadd(num);
        return num.clone().iadd(this);
      };
      BN2.prototype.isub = function isub(num) {
        if (num.negative !== 0) {
          num.negative = 0;
          var r = this.iadd(num);
          num.negative = 1;
          return r._normSign();
        } else if (this.negative !== 0) {
          this.negative = 0;
          this.iadd(num);
          this.negative = 1;
          return this._normSign();
        }
        var cmp = this.cmp(num);
        if (cmp === 0) {
          this.negative = 0;
          this.length = 1;
          this.words[0] = 0;
          return this;
        }
        var a, b;
        if (cmp > 0) {
          a = this;
          b = num;
        } else {
          a = num;
          b = this;
        }
        var carry = 0;
        for (var i = 0; i < b.length; i++) {
          r = (a.words[i] | 0) - (b.words[i] | 0) + carry;
          carry = r >> 26;
          this.words[i] = r & 67108863;
        }
        for (; carry !== 0 && i < a.length; i++) {
          r = (a.words[i] | 0) + carry;
          carry = r >> 26;
          this.words[i] = r & 67108863;
        }
        if (carry === 0 && i < a.length && a !== this) {
          for (; i < a.length; i++) {
            this.words[i] = a.words[i];
          }
        }
        this.length = Math.max(this.length, i);
        if (a !== this) {
          this.negative = 1;
        }
        return this._strip();
      };
      BN2.prototype.sub = function sub(num) {
        return this.clone().isub(num);
      };
      function smallMulTo(self2, num, out) {
        out.negative = num.negative ^ self2.negative;
        var len = self2.length + num.length | 0;
        out.length = len;
        len = len - 1 | 0;
        var a = self2.words[0] | 0;
        var b = num.words[0] | 0;
        var r = a * b;
        var lo = r & 67108863;
        var carry = r / 67108864 | 0;
        out.words[0] = lo;
        for (var k = 1; k < len; k++) {
          var ncarry = carry >>> 26;
          var rword = carry & 67108863;
          var maxJ = Math.min(k, num.length - 1);
          for (var j = Math.max(0, k - self2.length + 1); j <= maxJ; j++) {
            var i = k - j | 0;
            a = self2.words[i] | 0;
            b = num.words[j] | 0;
            r = a * b + rword;
            ncarry += r / 67108864 | 0;
            rword = r & 67108863;
          }
          out.words[k] = rword | 0;
          carry = ncarry | 0;
        }
        if (carry !== 0) {
          out.words[k] = carry | 0;
        } else {
          out.length--;
        }
        return out._strip();
      }
      var comb10MulTo = function comb10MulTo2(self2, num, out) {
        var a = self2.words;
        var b = num.words;
        var o = out.words;
        var c = 0;
        var lo;
        var mid;
        var hi;
        var a0 = a[0] | 0;
        var al0 = a0 & 8191;
        var ah0 = a0 >>> 13;
        var a1 = a[1] | 0;
        var al1 = a1 & 8191;
        var ah1 = a1 >>> 13;
        var a2 = a[2] | 0;
        var al2 = a2 & 8191;
        var ah2 = a2 >>> 13;
        var a3 = a[3] | 0;
        var al3 = a3 & 8191;
        var ah3 = a3 >>> 13;
        var a4 = a[4] | 0;
        var al4 = a4 & 8191;
        var ah4 = a4 >>> 13;
        var a5 = a[5] | 0;
        var al5 = a5 & 8191;
        var ah5 = a5 >>> 13;
        var a6 = a[6] | 0;
        var al6 = a6 & 8191;
        var ah6 = a6 >>> 13;
        var a7 = a[7] | 0;
        var al7 = a7 & 8191;
        var ah7 = a7 >>> 13;
        var a8 = a[8] | 0;
        var al8 = a8 & 8191;
        var ah8 = a8 >>> 13;
        var a9 = a[9] | 0;
        var al9 = a9 & 8191;
        var ah9 = a9 >>> 13;
        var b0 = b[0] | 0;
        var bl0 = b0 & 8191;
        var bh0 = b0 >>> 13;
        var b1 = b[1] | 0;
        var bl1 = b1 & 8191;
        var bh1 = b1 >>> 13;
        var b2 = b[2] | 0;
        var bl2 = b2 & 8191;
        var bh2 = b2 >>> 13;
        var b3 = b[3] | 0;
        var bl3 = b3 & 8191;
        var bh3 = b3 >>> 13;
        var b4 = b[4] | 0;
        var bl4 = b4 & 8191;
        var bh4 = b4 >>> 13;
        var b5 = b[5] | 0;
        var bl5 = b5 & 8191;
        var bh5 = b5 >>> 13;
        var b6 = b[6] | 0;
        var bl6 = b6 & 8191;
        var bh6 = b6 >>> 13;
        var b7 = b[7] | 0;
        var bl7 = b7 & 8191;
        var bh7 = b7 >>> 13;
        var b8 = b[8] | 0;
        var bl8 = b8 & 8191;
        var bh8 = b8 >>> 13;
        var b9 = b[9] | 0;
        var bl9 = b9 & 8191;
        var bh9 = b9 >>> 13;
        out.negative = self2.negative ^ num.negative;
        out.length = 19;
        lo = Math.imul(al0, bl0);
        mid = Math.imul(al0, bh0);
        mid = mid + Math.imul(ah0, bl0) | 0;
        hi = Math.imul(ah0, bh0);
        var w0 = (c + lo | 0) + ((mid & 8191) << 13) | 0;
        c = (hi + (mid >>> 13) | 0) + (w0 >>> 26) | 0;
        w0 &= 67108863;
        lo = Math.imul(al1, bl0);
        mid = Math.imul(al1, bh0);
        mid = mid + Math.imul(ah1, bl0) | 0;
        hi = Math.imul(ah1, bh0);
        lo = lo + Math.imul(al0, bl1) | 0;
        mid = mid + Math.imul(al0, bh1) | 0;
        mid = mid + Math.imul(ah0, bl1) | 0;
        hi = hi + Math.imul(ah0, bh1) | 0;
        var w1 = (c + lo | 0) + ((mid & 8191) << 13) | 0;
        c = (hi + (mid >>> 13) | 0) + (w1 >>> 26) | 0;
        w1 &= 67108863;
        lo = Math.imul(al2, bl0);
        mid = Math.imul(al2, bh0);
        mid = mid + Math.imul(ah2, bl0) | 0;
        hi = Math.imul(ah2, bh0);
        lo = lo + Math.imul(al1, bl1) | 0;
        mid = mid + Math.imul(al1, bh1) | 0;
        mid = mid + Math.imul(ah1, bl1) | 0;
        hi = hi + Math.imul(ah1, bh1) | 0;
        lo = lo + Math.imul(al0, bl2) | 0;
        mid = mid + Math.imul(al0, bh2) | 0;
        mid = mid + Math.imul(ah0, bl2) | 0;
        hi = hi + Math.imul(ah0, bh2) | 0;
        var w2 = (c + lo | 0) + ((mid & 8191) << 13) | 0;
        c = (hi + (mid >>> 13) | 0) + (w2 >>> 26) | 0;
        w2 &= 67108863;
        lo = Math.imul(al3, bl0);
        mid = Math.imul(al3, bh0);
        mid = mid + Math.imul(ah3, bl0) | 0;
        hi = Math.imul(ah3, bh0);
        lo = lo + Math.imul(al2, bl1) | 0;
        mid = mid + Math.imul(al2, bh1) | 0;
        mid = mid + Math.imul(ah2, bl1) | 0;
        hi = hi + Math.imul(ah2, bh1) | 0;
        lo = lo + Math.imul(al1, bl2) | 0;
        mid = mid + Math.imul(al1, bh2) | 0;
        mid = mid + Math.imul(ah1, bl2) | 0;
        hi = hi + Math.imul(ah1, bh2) | 0;
        lo = lo + Math.imul(al0, bl3) | 0;
        mid = mid + Math.imul(al0, bh3) | 0;
        mid = mid + Math.imul(ah0, bl3) | 0;
        hi = hi + Math.imul(ah0, bh3) | 0;
        var w3 = (c + lo | 0) + ((mid & 8191) << 13) | 0;
        c = (hi + (mid >>> 13) | 0) + (w3 >>> 26) | 0;
        w3 &= 67108863;
        lo = Math.imul(al4, bl0);
        mid = Math.imul(al4, bh0);
        mid = mid + Math.imul(ah4, bl0) | 0;
        hi = Math.imul(ah4, bh0);
        lo = lo + Math.imul(al3, bl1) | 0;
        mid = mid + Math.imul(al3, bh1) | 0;
        mid = mid + Math.imul(ah3, bl1) | 0;
        hi = hi + Math.imul(ah3, bh1) | 0;
        lo = lo + Math.imul(al2, bl2) | 0;
        mid = mid + Math.imul(al2, bh2) | 0;
        mid = mid + Math.imul(ah2, bl2) | 0;
        hi = hi + Math.imul(ah2, bh2) | 0;
        lo = lo + Math.imul(al1, bl3) | 0;
        mid = mid + Math.imul(al1, bh3) | 0;
        mid = mid + Math.imul(ah1, bl3) | 0;
        hi = hi + Math.imul(ah1, bh3) | 0;
        lo = lo + Math.imul(al0, bl4) | 0;
        mid = mid + Math.imul(al0, bh4) | 0;
        mid = mid + Math.imul(ah0, bl4) | 0;
        hi = hi + Math.imul(ah0, bh4) | 0;
        var w4 = (c + lo | 0) + ((mid & 8191) << 13) | 0;
        c = (hi + (mid >>> 13) | 0) + (w4 >>> 26) | 0;
        w4 &= 67108863;
        lo = Math.imul(al5, bl0);
        mid = Math.imul(al5, bh0);
        mid = mid + Math.imul(ah5, bl0) | 0;
        hi = Math.imul(ah5, bh0);
        lo = lo + Math.imul(al4, bl1) | 0;
        mid = mid + Math.imul(al4, bh1) | 0;
        mid = mid + Math.imul(ah4, bl1) | 0;
        hi = hi + Math.imul(ah4, bh1) | 0;
        lo = lo + Math.imul(al3, bl2) | 0;
        mid = mid + Math.imul(al3, bh2) | 0;
        mid = mid + Math.imul(ah3, bl2) | 0;
        hi = hi + Math.imul(ah3, bh2) | 0;
        lo = lo + Math.imul(al2, bl3) | 0;
        mid = mid + Math.imul(al2, bh3) | 0;
        mid = mid + Math.imul(ah2, bl3) | 0;
        hi = hi + Math.imul(ah2, bh3) | 0;
        lo = lo + Math.imul(al1, bl4) | 0;
        mid = mid + Math.imul(al1, bh4) | 0;
        mid = mid + Math.imul(ah1, bl4) | 0;
        hi = hi + Math.imul(ah1, bh4) | 0;
        lo = lo + Math.imul(al0, bl5) | 0;
        mid = mid + Math.imul(al0, bh5) | 0;
        mid = mid + Math.imul(ah0, bl5) | 0;
        hi = hi + Math.imul(ah0, bh5) | 0;
        var w5 = (c + lo | 0) + ((mid & 8191) << 13) | 0;
        c = (hi + (mid >>> 13) | 0) + (w5 >>> 26) | 0;
        w5 &= 67108863;
        lo = Math.imul(al6, bl0);
        mid = Math.imul(al6, bh0);
        mid = mid + Math.imul(ah6, bl0) | 0;
        hi = Math.imul(ah6, bh0);
        lo = lo + Math.imul(al5, bl1) | 0;
        mid = mid + Math.imul(al5, bh1) | 0;
        mid = mid + Math.imul(ah5, bl1) | 0;
        hi = hi + Math.imul(ah5, bh1) | 0;
        lo = lo + Math.imul(al4, bl2) | 0;
        mid = mid + Math.imul(al4, bh2) | 0;
        mid = mid + Math.imul(ah4, bl2) | 0;
        hi = hi + Math.imul(ah4, bh2) | 0;
        lo = lo + Math.imul(al3, bl3) | 0;
        mid = mid + Math.imul(al3, bh3) | 0;
        mid = mid + Math.imul(ah3, bl3) | 0;
        hi = hi + Math.imul(ah3, bh3) | 0;
        lo = lo + Math.imul(al2, bl4) | 0;
        mid = mid + Math.imul(al2, bh4) | 0;
        mid = mid + Math.imul(ah2, bl4) | 0;
        hi = hi + Math.imul(ah2, bh4) | 0;
        lo = lo + Math.imul(al1, bl5) | 0;
        mid = mid + Math.imul(al1, bh5) | 0;
        mid = mid + Math.imul(ah1, bl5) | 0;
        hi = hi + Math.imul(ah1, bh5) | 0;
        lo = lo + Math.imul(al0, bl6) | 0;
        mid = mid + Math.imul(al0, bh6) | 0;
        mid = mid + Math.imul(ah0, bl6) | 0;
        hi = hi + Math.imul(ah0, bh6) | 0;
        var w6 = (c + lo | 0) + ((mid & 8191) << 13) | 0;
        c = (hi + (mid >>> 13) | 0) + (w6 >>> 26) | 0;
        w6 &= 67108863;
        lo = Math.imul(al7, bl0);
        mid = Math.imul(al7, bh0);
        mid = mid + Math.imul(ah7, bl0) | 0;
        hi = Math.imul(ah7, bh0);
        lo = lo + Math.imul(al6, bl1) | 0;
        mid = mid + Math.imul(al6, bh1) | 0;
        mid = mid + Math.imul(ah6, bl1) | 0;
        hi = hi + Math.imul(ah6, bh1) | 0;
        lo = lo + Math.imul(al5, bl2) | 0;
        mid = mid + Math.imul(al5, bh2) | 0;
        mid = mid + Math.imul(ah5, bl2) | 0;
        hi = hi + Math.imul(ah5, bh2) | 0;
        lo = lo + Math.imul(al4, bl3) | 0;
        mid = mid + Math.imul(al4, bh3) | 0;
        mid = mid + Math.imul(ah4, bl3) | 0;
        hi = hi + Math.imul(ah4, bh3) | 0;
        lo = lo + Math.imul(al3, bl4) | 0;
        mid = mid + Math.imul(al3, bh4) | 0;
        mid = mid + Math.imul(ah3, bl4) | 0;
        hi = hi + Math.imul(ah3, bh4) | 0;
        lo = lo + Math.imul(al2, bl5) | 0;
        mid = mid + Math.imul(al2, bh5) | 0;
        mid = mid + Math.imul(ah2, bl5) | 0;
        hi = hi + Math.imul(ah2, bh5) | 0;
        lo = lo + Math.imul(al1, bl6) | 0;
        mid = mid + Math.imul(al1, bh6) | 0;
        mid = mid + Math.imul(ah1, bl6) | 0;
        hi = hi + Math.imul(ah1, bh6) | 0;
        lo = lo + Math.imul(al0, bl7) | 0;
        mid = mid + Math.imul(al0, bh7) | 0;
        mid = mid + Math.imul(ah0, bl7) | 0;
        hi = hi + Math.imul(ah0, bh7) | 0;
        var w7 = (c + lo | 0) + ((mid & 8191) << 13) | 0;
        c = (hi + (mid >>> 13) | 0) + (w7 >>> 26) | 0;
        w7 &= 67108863;
        lo = Math.imul(al8, bl0);
        mid = Math.imul(al8, bh0);
        mid = mid + Math.imul(ah8, bl0) | 0;
        hi = Math.imul(ah8, bh0);
        lo = lo + Math.imul(al7, bl1) | 0;
        mid = mid + Math.imul(al7, bh1) | 0;
        mid = mid + Math.imul(ah7, bl1) | 0;
        hi = hi + Math.imul(ah7, bh1) | 0;
        lo = lo + Math.imul(al6, bl2) | 0;
        mid = mid + Math.imul(al6, bh2) | 0;
        mid = mid + Math.imul(ah6, bl2) | 0;
        hi = hi + Math.imul(ah6, bh2) | 0;
        lo = lo + Math.imul(al5, bl3) | 0;
        mid = mid + Math.imul(al5, bh3) | 0;
        mid = mid + Math.imul(ah5, bl3) | 0;
        hi = hi + Math.imul(ah5, bh3) | 0;
        lo = lo + Math.imul(al4, bl4) | 0;
        mid = mid + Math.imul(al4, bh4) | 0;
        mid = mid + Math.imul(ah4, bl4) | 0;
        hi = hi + Math.imul(ah4, bh4) | 0;
        lo = lo + Math.imul(al3, bl5) | 0;
        mid = mid + Math.imul(al3, bh5) | 0;
        mid = mid + Math.imul(ah3, bl5) | 0;
        hi = hi + Math.imul(ah3, bh5) | 0;
        lo = lo + Math.imul(al2, bl6) | 0;
        mid = mid + Math.imul(al2, bh6) | 0;
        mid = mid + Math.imul(ah2, bl6) | 0;
        hi = hi + Math.imul(ah2, bh6) | 0;
        lo = lo + Math.imul(al1, bl7) | 0;
        mid = mid + Math.imul(al1, bh7) | 0;
        mid = mid + Math.imul(ah1, bl7) | 0;
        hi = hi + Math.imul(ah1, bh7) | 0;
        lo = lo + Math.imul(al0, bl8) | 0;
        mid = mid + Math.imul(al0, bh8) | 0;
        mid = mid + Math.imul(ah0, bl8) | 0;
        hi = hi + Math.imul(ah0, bh8) | 0;
        var w8 = (c + lo | 0) + ((mid & 8191) << 13) | 0;
        c = (hi + (mid >>> 13) | 0) + (w8 >>> 26) | 0;
        w8 &= 67108863;
        lo = Math.imul(al9, bl0);
        mid = Math.imul(al9, bh0);
        mid = mid + Math.imul(ah9, bl0) | 0;
        hi = Math.imul(ah9, bh0);
        lo = lo + Math.imul(al8, bl1) | 0;
        mid = mid + Math.imul(al8, bh1) | 0;
        mid = mid + Math.imul(ah8, bl1) | 0;
        hi = hi + Math.imul(ah8, bh1) | 0;
        lo = lo + Math.imul(al7, bl2) | 0;
        mid = mid + Math.imul(al7, bh2) | 0;
        mid = mid + Math.imul(ah7, bl2) | 0;
        hi = hi + Math.imul(ah7, bh2) | 0;
        lo = lo + Math.imul(al6, bl3) | 0;
        mid = mid + Math.imul(al6, bh3) | 0;
        mid = mid + Math.imul(ah6, bl3) | 0;
        hi = hi + Math.imul(ah6, bh3) | 0;
        lo = lo + Math.imul(al5, bl4) | 0;
        mid = mid + Math.imul(al5, bh4) | 0;
        mid = mid + Math.imul(ah5, bl4) | 0;
        hi = hi + Math.imul(ah5, bh4) | 0;
        lo = lo + Math.imul(al4, bl5) | 0;
        mid = mid + Math.imul(al4, bh5) | 0;
        mid = mid + Math.imul(ah4, bl5) | 0;
        hi = hi + Math.imul(ah4, bh5) | 0;
        lo = lo + Math.imul(al3, bl6) | 0;
        mid = mid + Math.imul(al3, bh6) | 0;
        mid = mid + Math.imul(ah3, bl6) | 0;
        hi = hi + Math.imul(ah3, bh6) | 0;
        lo = lo + Math.imul(al2, bl7) | 0;
        mid = mid + Math.imul(al2, bh7) | 0;
        mid = mid + Math.imul(ah2, bl7) | 0;
        hi = hi + Math.imul(ah2, bh7) | 0;
        lo = lo + Math.imul(al1, bl8) | 0;
        mid = mid + Math.imul(al1, bh8) | 0;
        mid = mid + Math.imul(ah1, bl8) | 0;
        hi = hi + Math.imul(ah1, bh8) | 0;
        lo = lo + Math.imul(al0, bl9) | 0;
        mid = mid + Math.imul(al0, bh9) | 0;
        mid = mid + Math.imul(ah0, bl9) | 0;
        hi = hi + Math.imul(ah0, bh9) | 0;
        var w9 = (c + lo | 0) + ((mid & 8191) << 13) | 0;
        c = (hi + (mid >>> 13) | 0) + (w9 >>> 26) | 0;
        w9 &= 67108863;
        lo = Math.imul(al9, bl1);
        mid = Math.imul(al9, bh1);
        mid = mid + Math.imul(ah9, bl1) | 0;
        hi = Math.imul(ah9, bh1);
        lo = lo + Math.imul(al8, bl2) | 0;
        mid = mid + Math.imul(al8, bh2) | 0;
        mid = mid + Math.imul(ah8, bl2) | 0;
        hi = hi + Math.imul(ah8, bh2) | 0;
        lo = lo + Math.imul(al7, bl3) | 0;
        mid = mid + Math.imul(al7, bh3) | 0;
        mid = mid + Math.imul(ah7, bl3) | 0;
        hi = hi + Math.imul(ah7, bh3) | 0;
        lo = lo + Math.imul(al6, bl4) | 0;
        mid = mid + Math.imul(al6, bh4) | 0;
        mid = mid + Math.imul(ah6, bl4) | 0;
        hi = hi + Math.imul(ah6, bh4) | 0;
        lo = lo + Math.imul(al5, bl5) | 0;
        mid = mid + Math.imul(al5, bh5) | 0;
        mid = mid + Math.imul(ah5, bl5) | 0;
        hi = hi + Math.imul(ah5, bh5) | 0;
        lo = lo + Math.imul(al4, bl6) | 0;
        mid = mid + Math.imul(al4, bh6) | 0;
        mid = mid + Math.imul(ah4, bl6) | 0;
        hi = hi + Math.imul(ah4, bh6) | 0;
        lo = lo + Math.imul(al3, bl7) | 0;
        mid = mid + Math.imul(al3, bh7) | 0;
        mid = mid + Math.imul(ah3, bl7) | 0;
        hi = hi + Math.imul(ah3, bh7) | 0;
        lo = lo + Math.imul(al2, bl8) | 0;
        mid = mid + Math.imul(al2, bh8) | 0;
        mid = mid + Math.imul(ah2, bl8) | 0;
        hi = hi + Math.imul(ah2, bh8) | 0;
        lo = lo + Math.imul(al1, bl9) | 0;
        mid = mid + Math.imul(al1, bh9) | 0;
        mid = mid + Math.imul(ah1, bl9) | 0;
        hi = hi + Math.imul(ah1, bh9) | 0;
        var w10 = (c + lo | 0) + ((mid & 8191) << 13) | 0;
        c = (hi + (mid >>> 13) | 0) + (w10 >>> 26) | 0;
        w10 &= 67108863;
        lo = Math.imul(al9, bl2);
        mid = Math.imul(al9, bh2);
        mid = mid + Math.imul(ah9, bl2) | 0;
        hi = Math.imul(ah9, bh2);
        lo = lo + Math.imul(al8, bl3) | 0;
        mid = mid + Math.imul(al8, bh3) | 0;
        mid = mid + Math.imul(ah8, bl3) | 0;
        hi = hi + Math.imul(ah8, bh3) | 0;
        lo = lo + Math.imul(al7, bl4) | 0;
        mid = mid + Math.imul(al7, bh4) | 0;
        mid = mid + Math.imul(ah7, bl4) | 0;
        hi = hi + Math.imul(ah7, bh4) | 0;
        lo = lo + Math.imul(al6, bl5) | 0;
        mid = mid + Math.imul(al6, bh5) | 0;
        mid = mid + Math.imul(ah6, bl5) | 0;
        hi = hi + Math.imul(ah6, bh5) | 0;
        lo = lo + Math.imul(al5, bl6) | 0;
        mid = mid + Math.imul(al5, bh6) | 0;
        mid = mid + Math.imul(ah5, bl6) | 0;
        hi = hi + Math.imul(ah5, bh6) | 0;
        lo = lo + Math.imul(al4, bl7) | 0;
        mid = mid + Math.imul(al4, bh7) | 0;
        mid = mid + Math.imul(ah4, bl7) | 0;
        hi = hi + Math.imul(ah4, bh7) | 0;
        lo = lo + Math.imul(al3, bl8) | 0;
        mid = mid + Math.imul(al3, bh8) | 0;
        mid = mid + Math.imul(ah3, bl8) | 0;
        hi = hi + Math.imul(ah3, bh8) | 0;
        lo = lo + Math.imul(al2, bl9) | 0;
        mid = mid + Math.imul(al2, bh9) | 0;
        mid = mid + Math.imul(ah2, bl9) | 0;
        hi = hi + Math.imul(ah2, bh9) | 0;
        var w11 = (c + lo | 0) + ((mid & 8191) << 13) | 0;
        c = (hi + (mid >>> 13) | 0) + (w11 >>> 26) | 0;
        w11 &= 67108863;
        lo = Math.imul(al9, bl3);
        mid = Math.imul(al9, bh3);
        mid = mid + Math.imul(ah9, bl3) | 0;
        hi = Math.imul(ah9, bh3);
        lo = lo + Math.imul(al8, bl4) | 0;
        mid = mid + Math.imul(al8, bh4) | 0;
        mid = mid + Math.imul(ah8, bl4) | 0;
        hi = hi + Math.imul(ah8, bh4) | 0;
        lo = lo + Math.imul(al7, bl5) | 0;
        mid = mid + Math.imul(al7, bh5) | 0;
        mid = mid + Math.imul(ah7, bl5) | 0;
        hi = hi + Math.imul(ah7, bh5) | 0;
        lo = lo + Math.imul(al6, bl6) | 0;
        mid = mid + Math.imul(al6, bh6) | 0;
        mid = mid + Math.imul(ah6, bl6) | 0;
        hi = hi + Math.imul(ah6, bh6) | 0;
        lo = lo + Math.imul(al5, bl7) | 0;
        mid = mid + Math.imul(al5, bh7) | 0;
        mid = mid + Math.imul(ah5, bl7) | 0;
        hi = hi + Math.imul(ah5, bh7) | 0;
        lo = lo + Math.imul(al4, bl8) | 0;
        mid = mid + Math.imul(al4, bh8) | 0;
        mid = mid + Math.imul(ah4, bl8) | 0;
        hi = hi + Math.imul(ah4, bh8) | 0;
        lo = lo + Math.imul(al3, bl9) | 0;
        mid = mid + Math.imul(al3, bh9) | 0;
        mid = mid + Math.imul(ah3, bl9) | 0;
        hi = hi + Math.imul(ah3, bh9) | 0;
        var w12 = (c + lo | 0) + ((mid & 8191) << 13) | 0;
        c = (hi + (mid >>> 13) | 0) + (w12 >>> 26) | 0;
        w12 &= 67108863;
        lo = Math.imul(al9, bl4);
        mid = Math.imul(al9, bh4);
        mid = mid + Math.imul(ah9, bl4) | 0;
        hi = Math.imul(ah9, bh4);
        lo = lo + Math.imul(al8, bl5) | 0;
        mid = mid + Math.imul(al8, bh5) | 0;
        mid = mid + Math.imul(ah8, bl5) | 0;
        hi = hi + Math.imul(ah8, bh5) | 0;
        lo = lo + Math.imul(al7, bl6) | 0;
        mid = mid + Math.imul(al7, bh6) | 0;
        mid = mid + Math.imul(ah7, bl6) | 0;
        hi = hi + Math.imul(ah7, bh6) | 0;
        lo = lo + Math.imul(al6, bl7) | 0;
        mid = mid + Math.imul(al6, bh7) | 0;
        mid = mid + Math.imul(ah6, bl7) | 0;
        hi = hi + Math.imul(ah6, bh7) | 0;
        lo = lo + Math.imul(al5, bl8) | 0;
        mid = mid + Math.imul(al5, bh8) | 0;
        mid = mid + Math.imul(ah5, bl8) | 0;
        hi = hi + Math.imul(ah5, bh8) | 0;
        lo = lo + Math.imul(al4, bl9) | 0;
        mid = mid + Math.imul(al4, bh9) | 0;
        mid = mid + Math.imul(ah4, bl9) | 0;
        hi = hi + Math.imul(ah4, bh9) | 0;
        var w13 = (c + lo | 0) + ((mid & 8191) << 13) | 0;
        c = (hi + (mid >>> 13) | 0) + (w13 >>> 26) | 0;
        w13 &= 67108863;
        lo = Math.imul(al9, bl5);
        mid = Math.imul(al9, bh5);
        mid = mid + Math.imul(ah9, bl5) | 0;
        hi = Math.imul(ah9, bh5);
        lo = lo + Math.imul(al8, bl6) | 0;
        mid = mid + Math.imul(al8, bh6) | 0;
        mid = mid + Math.imul(ah8, bl6) | 0;
        hi = hi + Math.imul(ah8, bh6) | 0;
        lo = lo + Math.imul(al7, bl7) | 0;
        mid = mid + Math.imul(al7, bh7) | 0;
        mid = mid + Math.imul(ah7, bl7) | 0;
        hi = hi + Math.imul(ah7, bh7) | 0;
        lo = lo + Math.imul(al6, bl8) | 0;
        mid = mid + Math.imul(al6, bh8) | 0;
        mid = mid + Math.imul(ah6, bl8) | 0;
        hi = hi + Math.imul(ah6, bh8) | 0;
        lo = lo + Math.imul(al5, bl9) | 0;
        mid = mid + Math.imul(al5, bh9) | 0;
        mid = mid + Math.imul(ah5, bl9) | 0;
        hi = hi + Math.imul(ah5, bh9) | 0;
        var w14 = (c + lo | 0) + ((mid & 8191) << 13) | 0;
        c = (hi + (mid >>> 13) | 0) + (w14 >>> 26) | 0;
        w14 &= 67108863;
        lo = Math.imul(al9, bl6);
        mid = Math.imul(al9, bh6);
        mid = mid + Math.imul(ah9, bl6) | 0;
        hi = Math.imul(ah9, bh6);
        lo = lo + Math.imul(al8, bl7) | 0;
        mid = mid + Math.imul(al8, bh7) | 0;
        mid = mid + Math.imul(ah8, bl7) | 0;
        hi = hi + Math.imul(ah8, bh7) | 0;
        lo = lo + Math.imul(al7, bl8) | 0;
        mid = mid + Math.imul(al7, bh8) | 0;
        mid = mid + Math.imul(ah7, bl8) | 0;
        hi = hi + Math.imul(ah7, bh8) | 0;
        lo = lo + Math.imul(al6, bl9) | 0;
        mid = mid + Math.imul(al6, bh9) | 0;
        mid = mid + Math.imul(ah6, bl9) | 0;
        hi = hi + Math.imul(ah6, bh9) | 0;
        var w15 = (c + lo | 0) + ((mid & 8191) << 13) | 0;
        c = (hi + (mid >>> 13) | 0) + (w15 >>> 26) | 0;
        w15 &= 67108863;
        lo = Math.imul(al9, bl7);
        mid = Math.imul(al9, bh7);
        mid = mid + Math.imul(ah9, bl7) | 0;
        hi = Math.imul(ah9, bh7);
        lo = lo + Math.imul(al8, bl8) | 0;
        mid = mid + Math.imul(al8, bh8) | 0;
        mid = mid + Math.imul(ah8, bl8) | 0;
        hi = hi + Math.imul(ah8, bh8) | 0;
        lo = lo + Math.imul(al7, bl9) | 0;
        mid = mid + Math.imul(al7, bh9) | 0;
        mid = mid + Math.imul(ah7, bl9) | 0;
        hi = hi + Math.imul(ah7, bh9) | 0;
        var w16 = (c + lo | 0) + ((mid & 8191) << 13) | 0;
        c = (hi + (mid >>> 13) | 0) + (w16 >>> 26) | 0;
        w16 &= 67108863;
        lo = Math.imul(al9, bl8);
        mid = Math.imul(al9, bh8);
        mid = mid + Math.imul(ah9, bl8) | 0;
        hi = Math.imul(ah9, bh8);
        lo = lo + Math.imul(al8, bl9) | 0;
        mid = mid + Math.imul(al8, bh9) | 0;
        mid = mid + Math.imul(ah8, bl9) | 0;
        hi = hi + Math.imul(ah8, bh9) | 0;
        var w17 = (c + lo | 0) + ((mid & 8191) << 13) | 0;
        c = (hi + (mid >>> 13) | 0) + (w17 >>> 26) | 0;
        w17 &= 67108863;
        lo = Math.imul(al9, bl9);
        mid = Math.imul(al9, bh9);
        mid = mid + Math.imul(ah9, bl9) | 0;
        hi = Math.imul(ah9, bh9);
        var w18 = (c + lo | 0) + ((mid & 8191) << 13) | 0;
        c = (hi + (mid >>> 13) | 0) + (w18 >>> 26) | 0;
        w18 &= 67108863;
        o[0] = w0;
        o[1] = w1;
        o[2] = w2;
        o[3] = w3;
        o[4] = w4;
        o[5] = w5;
        o[6] = w6;
        o[7] = w7;
        o[8] = w8;
        o[9] = w9;
        o[10] = w10;
        o[11] = w11;
        o[12] = w12;
        o[13] = w13;
        o[14] = w14;
        o[15] = w15;
        o[16] = w16;
        o[17] = w17;
        o[18] = w18;
        if (c !== 0) {
          o[19] = c;
          out.length++;
        }
        return out;
      };
      if (!Math.imul) {
        comb10MulTo = smallMulTo;
      }
      function bigMulTo(self2, num, out) {
        out.negative = num.negative ^ self2.negative;
        out.length = self2.length + num.length;
        var carry = 0;
        var hncarry = 0;
        for (var k = 0; k < out.length - 1; k++) {
          var ncarry = hncarry;
          hncarry = 0;
          var rword = carry & 67108863;
          var maxJ = Math.min(k, num.length - 1);
          for (var j = Math.max(0, k - self2.length + 1); j <= maxJ; j++) {
            var i = k - j;
            var a = self2.words[i] | 0;
            var b = num.words[j] | 0;
            var r = a * b;
            var lo = r & 67108863;
            ncarry = ncarry + (r / 67108864 | 0) | 0;
            lo = lo + rword | 0;
            rword = lo & 67108863;
            ncarry = ncarry + (lo >>> 26) | 0;
            hncarry += ncarry >>> 26;
            ncarry &= 67108863;
          }
          out.words[k] = rword;
          carry = ncarry;
          ncarry = hncarry;
        }
        if (carry !== 0) {
          out.words[k] = carry;
        } else {
          out.length--;
        }
        return out._strip();
      }
      function jumboMulTo(self2, num, out) {
        return bigMulTo(self2, num, out);
      }
      BN2.prototype.mulTo = function mulTo(num, out) {
        var res;
        var len = this.length + num.length;
        if (this.length === 10 && num.length === 10) {
          res = comb10MulTo(this, num, out);
        } else if (len < 63) {
          res = smallMulTo(this, num, out);
        } else if (len < 1024) {
          res = bigMulTo(this, num, out);
        } else {
          res = jumboMulTo(this, num, out);
        }
        return res;
      };
      function FFTM(x, y) {
        this.x = x;
        this.y = y;
      }
      FFTM.prototype.makeRBT = function makeRBT(N) {
        var t = new Array(N);
        var l = BN2.prototype._countBits(N) - 1;
        for (var i = 0; i < N; i++) {
          t[i] = this.revBin(i, l, N);
        }
        return t;
      };
      FFTM.prototype.revBin = function revBin(x, l, N) {
        if (x === 0 || x === N - 1)
          return x;
        var rb = 0;
        for (var i = 0; i < l; i++) {
          rb |= (x & 1) << l - i - 1;
          x >>= 1;
        }
        return rb;
      };
      FFTM.prototype.permute = function permute(rbt, rws, iws, rtws, itws, N) {
        for (var i = 0; i < N; i++) {
          rtws[i] = rws[rbt[i]];
          itws[i] = iws[rbt[i]];
        }
      };
      FFTM.prototype.transform = function transform(rws, iws, rtws, itws, N, rbt) {
        this.permute(rbt, rws, iws, rtws, itws, N);
        for (var s = 1; s < N; s <<= 1) {
          var l = s << 1;
          var rtwdf = Math.cos(2 * Math.PI / l);
          var itwdf = Math.sin(2 * Math.PI / l);
          for (var p = 0; p < N; p += l) {
            var rtwdf_ = rtwdf;
            var itwdf_ = itwdf;
            for (var j = 0; j < s; j++) {
              var re = rtws[p + j];
              var ie = itws[p + j];
              var ro = rtws[p + j + s];
              var io = itws[p + j + s];
              var rx = rtwdf_ * ro - itwdf_ * io;
              io = rtwdf_ * io + itwdf_ * ro;
              ro = rx;
              rtws[p + j] = re + ro;
              itws[p + j] = ie + io;
              rtws[p + j + s] = re - ro;
              itws[p + j + s] = ie - io;
              if (j !== l) {
                rx = rtwdf * rtwdf_ - itwdf * itwdf_;
                itwdf_ = rtwdf * itwdf_ + itwdf * rtwdf_;
                rtwdf_ = rx;
              }
            }
          }
        }
      };
      FFTM.prototype.guessLen13b = function guessLen13b(n, m) {
        var N = Math.max(m, n) | 1;
        var odd = N & 1;
        var i = 0;
        for (N = N / 2 | 0; N; N = N >>> 1) {
          i++;
        }
        return 1 << i + 1 + odd;
      };
      FFTM.prototype.conjugate = function conjugate(rws, iws, N) {
        if (N <= 1)
          return;
        for (var i = 0; i < N / 2; i++) {
          var t = rws[i];
          rws[i] = rws[N - i - 1];
          rws[N - i - 1] = t;
          t = iws[i];
          iws[i] = -iws[N - i - 1];
          iws[N - i - 1] = -t;
        }
      };
      FFTM.prototype.normalize13b = function normalize13b(ws, N) {
        var carry = 0;
        for (var i = 0; i < N / 2; i++) {
          var w = Math.round(ws[2 * i + 1] / N) * 8192 + Math.round(ws[2 * i] / N) + carry;
          ws[i] = w & 67108863;
          if (w < 67108864) {
            carry = 0;
          } else {
            carry = w / 67108864 | 0;
          }
        }
        return ws;
      };
      FFTM.prototype.convert13b = function convert13b(ws, len, rws, N) {
        var carry = 0;
        for (var i = 0; i < len; i++) {
          carry = carry + (ws[i] | 0);
          rws[2 * i] = carry & 8191;
          carry = carry >>> 13;
          rws[2 * i + 1] = carry & 8191;
          carry = carry >>> 13;
        }
        for (i = 2 * len; i < N; ++i) {
          rws[i] = 0;
        }
        assert2(carry === 0);
        assert2((carry & ~8191) === 0);
      };
      FFTM.prototype.stub = function stub(N) {
        var ph = new Array(N);
        for (var i = 0; i < N; i++) {
          ph[i] = 0;
        }
        return ph;
      };
      FFTM.prototype.mulp = function mulp(x, y, out) {
        var N = 2 * this.guessLen13b(x.length, y.length);
        var rbt = this.makeRBT(N);
        var _ = this.stub(N);
        var rws = new Array(N);
        var rwst = new Array(N);
        var iwst = new Array(N);
        var nrws = new Array(N);
        var nrwst = new Array(N);
        var niwst = new Array(N);
        var rmws = out.words;
        rmws.length = N;
        this.convert13b(x.words, x.length, rws, N);
        this.convert13b(y.words, y.length, nrws, N);
        this.transform(rws, _, rwst, iwst, N, rbt);
        this.transform(nrws, _, nrwst, niwst, N, rbt);
        for (var i = 0; i < N; i++) {
          var rx = rwst[i] * nrwst[i] - iwst[i] * niwst[i];
          iwst[i] = rwst[i] * niwst[i] + iwst[i] * nrwst[i];
          rwst[i] = rx;
        }
        this.conjugate(rwst, iwst, N);
        this.transform(rwst, iwst, rmws, _, N, rbt);
        this.conjugate(rmws, _, N);
        this.normalize13b(rmws, N);
        out.negative = x.negative ^ y.negative;
        out.length = x.length + y.length;
        return out._strip();
      };
      BN2.prototype.mul = function mul(num) {
        var out = new BN2(null);
        out.words = new Array(this.length + num.length);
        return this.mulTo(num, out);
      };
      BN2.prototype.mulf = function mulf(num) {
        var out = new BN2(null);
        out.words = new Array(this.length + num.length);
        return jumboMulTo(this, num, out);
      };
      BN2.prototype.imul = function imul(num) {
        return this.clone().mulTo(num, this);
      };
      BN2.prototype.imuln = function imuln(num) {
        var isNegNum = num < 0;
        if (isNegNum)
          num = -num;
        assert2(typeof num === "number");
        assert2(num < 67108864);
        var carry = 0;
        for (var i = 0; i < this.length; i++) {
          var w = (this.words[i] | 0) * num;
          var lo = (w & 67108863) + (carry & 67108863);
          carry >>= 26;
          carry += w / 67108864 | 0;
          carry += lo >>> 26;
          this.words[i] = lo & 67108863;
        }
        if (carry !== 0) {
          this.words[i] = carry;
          this.length++;
        }
        return isNegNum ? this.ineg() : this;
      };
      BN2.prototype.muln = function muln(num) {
        return this.clone().imuln(num);
      };
      BN2.prototype.sqr = function sqr() {
        return this.mul(this);
      };
      BN2.prototype.isqr = function isqr() {
        return this.imul(this.clone());
      };
      BN2.prototype.pow = function pow(num) {
        var w = toBitArray(num);
        if (w.length === 0)
          return new BN2(1);
        var res = this;
        for (var i = 0; i < w.length; i++, res = res.sqr()) {
          if (w[i] !== 0)
            break;
        }
        if (++i < w.length) {
          for (var q = res.sqr(); i < w.length; i++, q = q.sqr()) {
            if (w[i] === 0)
              continue;
            res = res.mul(q);
          }
        }
        return res;
      };
      BN2.prototype.iushln = function iushln(bits) {
        assert2(typeof bits === "number" && bits >= 0);
        var r = bits % 26;
        var s = (bits - r) / 26;
        var carryMask = 67108863 >>> 26 - r << 26 - r;
        var i;
        if (r !== 0) {
          var carry = 0;
          for (i = 0; i < this.length; i++) {
            var newCarry = this.words[i] & carryMask;
            var c = (this.words[i] | 0) - newCarry << r;
            this.words[i] = c | carry;
            carry = newCarry >>> 26 - r;
          }
          if (carry) {
            this.words[i] = carry;
            this.length++;
          }
        }
        if (s !== 0) {
          for (i = this.length - 1; i >= 0; i--) {
            this.words[i + s] = this.words[i];
          }
          for (i = 0; i < s; i++) {
            this.words[i] = 0;
          }
          this.length += s;
        }
        return this._strip();
      };
      BN2.prototype.ishln = function ishln(bits) {
        assert2(this.negative === 0);
        return this.iushln(bits);
      };
      BN2.prototype.iushrn = function iushrn(bits, hint, extended) {
        assert2(typeof bits === "number" && bits >= 0);
        var h;
        if (hint) {
          h = (hint - hint % 26) / 26;
        } else {
          h = 0;
        }
        var r = bits % 26;
        var s = Math.min((bits - r) / 26, this.length);
        var mask = 67108863 ^ 67108863 >>> r << r;
        var maskedWords = extended;
        h -= s;
        h = Math.max(0, h);
        if (maskedWords) {
          for (var i = 0; i < s; i++) {
            maskedWords.words[i] = this.words[i];
          }
          maskedWords.length = s;
        }
        if (s === 0) {
        } else if (this.length > s) {
          this.length -= s;
          for (i = 0; i < this.length; i++) {
            this.words[i] = this.words[i + s];
          }
        } else {
          this.words[0] = 0;
          this.length = 1;
        }
        var carry = 0;
        for (i = this.length - 1; i >= 0 && (carry !== 0 || i >= h); i--) {
          var word = this.words[i] | 0;
          this.words[i] = carry << 26 - r | word >>> r;
          carry = word & mask;
        }
        if (maskedWords && carry !== 0) {
          maskedWords.words[maskedWords.length++] = carry;
        }
        if (this.length === 0) {
          this.words[0] = 0;
          this.length = 1;
        }
        return this._strip();
      };
      BN2.prototype.ishrn = function ishrn(bits, hint, extended) {
        assert2(this.negative === 0);
        return this.iushrn(bits, hint, extended);
      };
      BN2.prototype.shln = function shln(bits) {
        return this.clone().ishln(bits);
      };
      BN2.prototype.ushln = function ushln(bits) {
        return this.clone().iushln(bits);
      };
      BN2.prototype.shrn = function shrn(bits) {
        return this.clone().ishrn(bits);
      };
      BN2.prototype.ushrn = function ushrn(bits) {
        return this.clone().iushrn(bits);
      };
      BN2.prototype.testn = function testn(bit) {
        assert2(typeof bit === "number" && bit >= 0);
        var r = bit % 26;
        var s = (bit - r) / 26;
        var q = 1 << r;
        if (this.length <= s)
          return false;
        var w = this.words[s];
        return !!(w & q);
      };
      BN2.prototype.imaskn = function imaskn(bits) {
        assert2(typeof bits === "number" && bits >= 0);
        var r = bits % 26;
        var s = (bits - r) / 26;
        assert2(this.negative === 0, "imaskn works only with positive numbers");
        if (this.length <= s) {
          return this;
        }
        if (r !== 0) {
          s++;
        }
        this.length = Math.min(s, this.length);
        if (r !== 0) {
          var mask = 67108863 ^ 67108863 >>> r << r;
          this.words[this.length - 1] &= mask;
        }
        return this._strip();
      };
      BN2.prototype.maskn = function maskn(bits) {
        return this.clone().imaskn(bits);
      };
      BN2.prototype.iaddn = function iaddn(num) {
        assert2(typeof num === "number");
        assert2(num < 67108864);
        if (num < 0)
          return this.isubn(-num);
        if (this.negative !== 0) {
          if (this.length === 1 && (this.words[0] | 0) <= num) {
            this.words[0] = num - (this.words[0] | 0);
            this.negative = 0;
            return this;
          }
          this.negative = 0;
          this.isubn(num);
          this.negative = 1;
          return this;
        }
        return this._iaddn(num);
      };
      BN2.prototype._iaddn = function _iaddn(num) {
        this.words[0] += num;
        for (var i = 0; i < this.length && this.words[i] >= 67108864; i++) {
          this.words[i] -= 67108864;
          if (i === this.length - 1) {
            this.words[i + 1] = 1;
          } else {
            this.words[i + 1]++;
          }
        }
        this.length = Math.max(this.length, i + 1);
        return this;
      };
      BN2.prototype.isubn = function isubn(num) {
        assert2(typeof num === "number");
        assert2(num < 67108864);
        if (num < 0)
          return this.iaddn(-num);
        if (this.negative !== 0) {
          this.negative = 0;
          this.iaddn(num);
          this.negative = 1;
          return this;
        }
        this.words[0] -= num;
        if (this.length === 1 && this.words[0] < 0) {
          this.words[0] = -this.words[0];
          this.negative = 1;
        } else {
          for (var i = 0; i < this.length && this.words[i] < 0; i++) {
            this.words[i] += 67108864;
            this.words[i + 1] -= 1;
          }
        }
        return this._strip();
      };
      BN2.prototype.addn = function addn(num) {
        return this.clone().iaddn(num);
      };
      BN2.prototype.subn = function subn(num) {
        return this.clone().isubn(num);
      };
      BN2.prototype.iabs = function iabs() {
        this.negative = 0;
        return this;
      };
      BN2.prototype.abs = function abs() {
        return this.clone().iabs();
      };
      BN2.prototype._ishlnsubmul = function _ishlnsubmul(num, mul, shift) {
        var len = num.length + shift;
        var i;
        this._expand(len);
        var w;
        var carry = 0;
        for (i = 0; i < num.length; i++) {
          w = (this.words[i + shift] | 0) + carry;
          var right = (num.words[i] | 0) * mul;
          w -= right & 67108863;
          carry = (w >> 26) - (right / 67108864 | 0);
          this.words[i + shift] = w & 67108863;
        }
        for (; i < this.length - shift; i++) {
          w = (this.words[i + shift] | 0) + carry;
          carry = w >> 26;
          this.words[i + shift] = w & 67108863;
        }
        if (carry === 0)
          return this._strip();
        assert2(carry === -1);
        carry = 0;
        for (i = 0; i < this.length; i++) {
          w = -(this.words[i] | 0) + carry;
          carry = w >> 26;
          this.words[i] = w & 67108863;
        }
        this.negative = 1;
        return this._strip();
      };
      BN2.prototype._wordDiv = function _wordDiv(num, mode) {
        var shift = this.length - num.length;
        var a = this.clone();
        var b = num;
        var bhi = b.words[b.length - 1] | 0;
        var bhiBits = this._countBits(bhi);
        shift = 26 - bhiBits;
        if (shift !== 0) {
          b = b.ushln(shift);
          a.iushln(shift);
          bhi = b.words[b.length - 1] | 0;
        }
        var m = a.length - b.length;
        var q;
        if (mode !== "mod") {
          q = new BN2(null);
          q.length = m + 1;
          q.words = new Array(q.length);
          for (var i = 0; i < q.length; i++) {
            q.words[i] = 0;
          }
        }
        var diff = a.clone()._ishlnsubmul(b, 1, m);
        if (diff.negative === 0) {
          a = diff;
          if (q) {
            q.words[m] = 1;
          }
        }
        for (var j = m - 1; j >= 0; j--) {
          var qj = (a.words[b.length + j] | 0) * 67108864 + (a.words[b.length + j - 1] | 0);
          qj = Math.min(qj / bhi | 0, 67108863);
          a._ishlnsubmul(b, qj, j);
          while (a.negative !== 0) {
            qj--;
            a.negative = 0;
            a._ishlnsubmul(b, 1, j);
            if (!a.isZero()) {
              a.negative ^= 1;
            }
          }
          if (q) {
            q.words[j] = qj;
          }
        }
        if (q) {
          q._strip();
        }
        a._strip();
        if (mode !== "div" && shift !== 0) {
          a.iushrn(shift);
        }
        return {
          div: q || null,
          mod: a
        };
      };
      BN2.prototype.divmod = function divmod(num, mode, positive) {
        assert2(!num.isZero());
        if (this.isZero()) {
          return {
            div: new BN2(0),
            mod: new BN2(0)
          };
        }
        var div, mod, res;
        if (this.negative !== 0 && num.negative === 0) {
          res = this.neg().divmod(num, mode);
          if (mode !== "mod") {
            div = res.div.neg();
          }
          if (mode !== "div") {
            mod = res.mod.neg();
            if (positive && mod.negative !== 0) {
              mod.iadd(num);
            }
          }
          return {
            div,
            mod
          };
        }
        if (this.negative === 0 && num.negative !== 0) {
          res = this.divmod(num.neg(), mode);
          if (mode !== "mod") {
            div = res.div.neg();
          }
          return {
            div,
            mod: res.mod
          };
        }
        if ((this.negative & num.negative) !== 0) {
          res = this.neg().divmod(num.neg(), mode);
          if (mode !== "div") {
            mod = res.mod.neg();
            if (positive && mod.negative !== 0) {
              mod.isub(num);
            }
          }
          return {
            div: res.div,
            mod
          };
        }
        if (num.length > this.length || this.cmp(num) < 0) {
          return {
            div: new BN2(0),
            mod: this
          };
        }
        if (num.length === 1) {
          if (mode === "div") {
            return {
              div: this.divn(num.words[0]),
              mod: null
            };
          }
          if (mode === "mod") {
            return {
              div: null,
              mod: new BN2(this.modrn(num.words[0]))
            };
          }
          return {
            div: this.divn(num.words[0]),
            mod: new BN2(this.modrn(num.words[0]))
          };
        }
        return this._wordDiv(num, mode);
      };
      BN2.prototype.div = function div(num) {
        return this.divmod(num, "div", false).div;
      };
      BN2.prototype.mod = function mod(num) {
        return this.divmod(num, "mod", false).mod;
      };
      BN2.prototype.umod = function umod(num) {
        return this.divmod(num, "mod", true).mod;
      };
      BN2.prototype.divRound = function divRound(num) {
        var dm = this.divmod(num);
        if (dm.mod.isZero())
          return dm.div;
        var mod = dm.div.negative !== 0 ? dm.mod.isub(num) : dm.mod;
        var half = num.ushrn(1);
        var r2 = num.andln(1);
        var cmp = mod.cmp(half);
        if (cmp < 0 || r2 === 1 && cmp === 0)
          return dm.div;
        return dm.div.negative !== 0 ? dm.div.isubn(1) : dm.div.iaddn(1);
      };
      BN2.prototype.modrn = function modrn(num) {
        var isNegNum = num < 0;
        if (isNegNum)
          num = -num;
        assert2(num <= 67108863);
        var p = (1 << 26) % num;
        var acc = 0;
        for (var i = this.length - 1; i >= 0; i--) {
          acc = (p * acc + (this.words[i] | 0)) % num;
        }
        return isNegNum ? -acc : acc;
      };
      BN2.prototype.modn = function modn(num) {
        return this.modrn(num);
      };
      BN2.prototype.idivn = function idivn(num) {
        var isNegNum = num < 0;
        if (isNegNum)
          num = -num;
        assert2(num <= 67108863);
        var carry = 0;
        for (var i = this.length - 1; i >= 0; i--) {
          var w = (this.words[i] | 0) + carry * 67108864;
          this.words[i] = w / num | 0;
          carry = w % num;
        }
        this._strip();
        return isNegNum ? this.ineg() : this;
      };
      BN2.prototype.divn = function divn(num) {
        return this.clone().idivn(num);
      };
      BN2.prototype.egcd = function egcd(p) {
        assert2(p.negative === 0);
        assert2(!p.isZero());
        var x = this;
        var y = p.clone();
        if (x.negative !== 0) {
          x = x.umod(p);
        } else {
          x = x.clone();
        }
        var A = new BN2(1);
        var B = new BN2(0);
        var C = new BN2(0);
        var D = new BN2(1);
        var g = 0;
        while (x.isEven() && y.isEven()) {
          x.iushrn(1);
          y.iushrn(1);
          ++g;
        }
        var yp = y.clone();
        var xp = x.clone();
        while (!x.isZero()) {
          for (var i = 0, im = 1; (x.words[0] & im) === 0 && i < 26; ++i, im <<= 1)
            ;
          if (i > 0) {
            x.iushrn(i);
            while (i-- > 0) {
              if (A.isOdd() || B.isOdd()) {
                A.iadd(yp);
                B.isub(xp);
              }
              A.iushrn(1);
              B.iushrn(1);
            }
          }
          for (var j = 0, jm = 1; (y.words[0] & jm) === 0 && j < 26; ++j, jm <<= 1)
            ;
          if (j > 0) {
            y.iushrn(j);
            while (j-- > 0) {
              if (C.isOdd() || D.isOdd()) {
                C.iadd(yp);
                D.isub(xp);
              }
              C.iushrn(1);
              D.iushrn(1);
            }
          }
          if (x.cmp(y) >= 0) {
            x.isub(y);
            A.isub(C);
            B.isub(D);
          } else {
            y.isub(x);
            C.isub(A);
            D.isub(B);
          }
        }
        return {
          a: C,
          b: D,
          gcd: y.iushln(g)
        };
      };
      BN2.prototype._invmp = function _invmp(p) {
        assert2(p.negative === 0);
        assert2(!p.isZero());
        var a = this;
        var b = p.clone();
        if (a.negative !== 0) {
          a = a.umod(p);
        } else {
          a = a.clone();
        }
        var x1 = new BN2(1);
        var x2 = new BN2(0);
        var delta = b.clone();
        while (a.cmpn(1) > 0 && b.cmpn(1) > 0) {
          for (var i = 0, im = 1; (a.words[0] & im) === 0 && i < 26; ++i, im <<= 1)
            ;
          if (i > 0) {
            a.iushrn(i);
            while (i-- > 0) {
              if (x1.isOdd()) {
                x1.iadd(delta);
              }
              x1.iushrn(1);
            }
          }
          for (var j = 0, jm = 1; (b.words[0] & jm) === 0 && j < 26; ++j, jm <<= 1)
            ;
          if (j > 0) {
            b.iushrn(j);
            while (j-- > 0) {
              if (x2.isOdd()) {
                x2.iadd(delta);
              }
              x2.iushrn(1);
            }
          }
          if (a.cmp(b) >= 0) {
            a.isub(b);
            x1.isub(x2);
          } else {
            b.isub(a);
            x2.isub(x1);
          }
        }
        var res;
        if (a.cmpn(1) === 0) {
          res = x1;
        } else {
          res = x2;
        }
        if (res.cmpn(0) < 0) {
          res.iadd(p);
        }
        return res;
      };
      BN2.prototype.gcd = function gcd(num) {
        if (this.isZero())
          return num.abs();
        if (num.isZero())
          return this.abs();
        var a = this.clone();
        var b = num.clone();
        a.negative = 0;
        b.negative = 0;
        for (var shift = 0; a.isEven() && b.isEven(); shift++) {
          a.iushrn(1);
          b.iushrn(1);
        }
        do {
          while (a.isEven()) {
            a.iushrn(1);
          }
          while (b.isEven()) {
            b.iushrn(1);
          }
          var r = a.cmp(b);
          if (r < 0) {
            var t = a;
            a = b;
            b = t;
          } else if (r === 0 || b.cmpn(1) === 0) {
            break;
          }
          a.isub(b);
        } while (true);
        return b.iushln(shift);
      };
      BN2.prototype.invm = function invm(num) {
        return this.egcd(num).a.umod(num);
      };
      BN2.prototype.isEven = function isEven() {
        return (this.words[0] & 1) === 0;
      };
      BN2.prototype.isOdd = function isOdd() {
        return (this.words[0] & 1) === 1;
      };
      BN2.prototype.andln = function andln(num) {
        return this.words[0] & num;
      };
      BN2.prototype.bincn = function bincn(bit) {
        assert2(typeof bit === "number");
        var r = bit % 26;
        var s = (bit - r) / 26;
        var q = 1 << r;
        if (this.length <= s) {
          this._expand(s + 1);
          this.words[s] |= q;
          return this;
        }
        var carry = q;
        for (var i = s; carry !== 0 && i < this.length; i++) {
          var w = this.words[i] | 0;
          w += carry;
          carry = w >>> 26;
          w &= 67108863;
          this.words[i] = w;
        }
        if (carry !== 0) {
          this.words[i] = carry;
          this.length++;
        }
        return this;
      };
      BN2.prototype.isZero = function isZero() {
        return this.length === 1 && this.words[0] === 0;
      };
      BN2.prototype.cmpn = function cmpn(num) {
        var negative = num < 0;
        if (this.negative !== 0 && !negative)
          return -1;
        if (this.negative === 0 && negative)
          return 1;
        this._strip();
        var res;
        if (this.length > 1) {
          res = 1;
        } else {
          if (negative) {
            num = -num;
          }
          assert2(num <= 67108863, "Number is too big");
          var w = this.words[0] | 0;
          res = w === num ? 0 : w < num ? -1 : 1;
        }
        if (this.negative !== 0)
          return -res | 0;
        return res;
      };
      BN2.prototype.cmp = function cmp(num) {
        if (this.negative !== 0 && num.negative === 0)
          return -1;
        if (this.negative === 0 && num.negative !== 0)
          return 1;
        var res = this.ucmp(num);
        if (this.negative !== 0)
          return -res | 0;
        return res;
      };
      BN2.prototype.ucmp = function ucmp(num) {
        if (this.length > num.length)
          return 1;
        if (this.length < num.length)
          return -1;
        var res = 0;
        for (var i = this.length - 1; i >= 0; i--) {
          var a = this.words[i] | 0;
          var b = num.words[i] | 0;
          if (a === b)
            continue;
          if (a < b) {
            res = -1;
          } else if (a > b) {
            res = 1;
          }
          break;
        }
        return res;
      };
      BN2.prototype.gtn = function gtn(num) {
        return this.cmpn(num) === 1;
      };
      BN2.prototype.gt = function gt(num) {
        return this.cmp(num) === 1;
      };
      BN2.prototype.gten = function gten(num) {
        return this.cmpn(num) >= 0;
      };
      BN2.prototype.gte = function gte(num) {
        return this.cmp(num) >= 0;
      };
      BN2.prototype.ltn = function ltn(num) {
        return this.cmpn(num) === -1;
      };
      BN2.prototype.lt = function lt(num) {
        return this.cmp(num) === -1;
      };
      BN2.prototype.lten = function lten(num) {
        return this.cmpn(num) <= 0;
      };
      BN2.prototype.lte = function lte(num) {
        return this.cmp(num) <= 0;
      };
      BN2.prototype.eqn = function eqn(num) {
        return this.cmpn(num) === 0;
      };
      BN2.prototype.eq = function eq(num) {
        return this.cmp(num) === 0;
      };
      BN2.red = function red(num) {
        return new Red(num);
      };
      BN2.prototype.toRed = function toRed(ctx) {
        assert2(!this.red, "Already a number in reduction context");
        assert2(this.negative === 0, "red works only with positives");
        return ctx.convertTo(this)._forceRed(ctx);
      };
      BN2.prototype.fromRed = function fromRed() {
        assert2(this.red, "fromRed works only with numbers in reduction context");
        return this.red.convertFrom(this);
      };
      BN2.prototype._forceRed = function _forceRed(ctx) {
        this.red = ctx;
        return this;
      };
      BN2.prototype.forceRed = function forceRed(ctx) {
        assert2(!this.red, "Already a number in reduction context");
        return this._forceRed(ctx);
      };
      BN2.prototype.redAdd = function redAdd(num) {
        assert2(this.red, "redAdd works only with red numbers");
        return this.red.add(this, num);
      };
      BN2.prototype.redIAdd = function redIAdd(num) {
        assert2(this.red, "redIAdd works only with red numbers");
        return this.red.iadd(this, num);
      };
      BN2.prototype.redSub = function redSub(num) {
        assert2(this.red, "redSub works only with red numbers");
        return this.red.sub(this, num);
      };
      BN2.prototype.redISub = function redISub(num) {
        assert2(this.red, "redISub works only with red numbers");
        return this.red.isub(this, num);
      };
      BN2.prototype.redShl = function redShl(num) {
        assert2(this.red, "redShl works only with red numbers");
        return this.red.shl(this, num);
      };
      BN2.prototype.redMul = function redMul(num) {
        assert2(this.red, "redMul works only with red numbers");
        this.red._verify2(this, num);
        return this.red.mul(this, num);
      };
      BN2.prototype.redIMul = function redIMul(num) {
        assert2(this.red, "redMul works only with red numbers");
        this.red._verify2(this, num);
        return this.red.imul(this, num);
      };
      BN2.prototype.redSqr = function redSqr() {
        assert2(this.red, "redSqr works only with red numbers");
        this.red._verify1(this);
        return this.red.sqr(this);
      };
      BN2.prototype.redISqr = function redISqr() {
        assert2(this.red, "redISqr works only with red numbers");
        this.red._verify1(this);
        return this.red.isqr(this);
      };
      BN2.prototype.redSqrt = function redSqrt() {
        assert2(this.red, "redSqrt works only with red numbers");
        this.red._verify1(this);
        return this.red.sqrt(this);
      };
      BN2.prototype.redInvm = function redInvm() {
        assert2(this.red, "redInvm works only with red numbers");
        this.red._verify1(this);
        return this.red.invm(this);
      };
      BN2.prototype.redNeg = function redNeg() {
        assert2(this.red, "redNeg works only with red numbers");
        this.red._verify1(this);
        return this.red.neg(this);
      };
      BN2.prototype.redPow = function redPow(num) {
        assert2(this.red && !num.red, "redPow(normalNum)");
        this.red._verify1(this);
        return this.red.pow(this, num);
      };
      var primes = {
        k256: null,
        p224: null,
        p192: null,
        p25519: null
      };
      function MPrime(name, p) {
        this.name = name;
        this.p = new BN2(p, 16);
        this.n = this.p.bitLength();
        this.k = new BN2(1).iushln(this.n).isub(this.p);
        this.tmp = this._tmp();
      }
      MPrime.prototype._tmp = function _tmp() {
        var tmp = new BN2(null);
        tmp.words = new Array(Math.ceil(this.n / 13));
        return tmp;
      };
      MPrime.prototype.ireduce = function ireduce(num) {
        var r = num;
        var rlen;
        do {
          this.split(r, this.tmp);
          r = this.imulK(r);
          r = r.iadd(this.tmp);
          rlen = r.bitLength();
        } while (rlen > this.n);
        var cmp = rlen < this.n ? -1 : r.ucmp(this.p);
        if (cmp === 0) {
          r.words[0] = 0;
          r.length = 1;
        } else if (cmp > 0) {
          r.isub(this.p);
        } else {
          if (r.strip !== void 0) {
            r.strip();
          } else {
            r._strip();
          }
        }
        return r;
      };
      MPrime.prototype.split = function split(input, out) {
        input.iushrn(this.n, 0, out);
      };
      MPrime.prototype.imulK = function imulK(num) {
        return num.imul(this.k);
      };
      function K256() {
        MPrime.call(
          this,
          "k256",
          "ffffffff ffffffff ffffffff ffffffff ffffffff ffffffff fffffffe fffffc2f"
        );
      }
      inherits(K256, MPrime);
      K256.prototype.split = function split(input, output) {
        var mask = 4194303;
        var outLen = Math.min(input.length, 9);
        for (var i = 0; i < outLen; i++) {
          output.words[i] = input.words[i];
        }
        output.length = outLen;
        if (input.length <= 9) {
          input.words[0] = 0;
          input.length = 1;
          return;
        }
        var prev = input.words[9];
        output.words[output.length++] = prev & mask;
        for (i = 10; i < input.length; i++) {
          var next = input.words[i] | 0;
          input.words[i - 10] = (next & mask) << 4 | prev >>> 22;
          prev = next;
        }
        prev >>>= 22;
        input.words[i - 10] = prev;
        if (prev === 0 && input.length > 10) {
          input.length -= 10;
        } else {
          input.length -= 9;
        }
      };
      K256.prototype.imulK = function imulK(num) {
        num.words[num.length] = 0;
        num.words[num.length + 1] = 0;
        num.length += 2;
        var lo = 0;
        for (var i = 0; i < num.length; i++) {
          var w = num.words[i] | 0;
          lo += w * 977;
          num.words[i] = lo & 67108863;
          lo = w * 64 + (lo / 67108864 | 0);
        }
        if (num.words[num.length - 1] === 0) {
          num.length--;
          if (num.words[num.length - 1] === 0) {
            num.length--;
          }
        }
        return num;
      };
      function P224() {
        MPrime.call(
          this,
          "p224",
          "ffffffff ffffffff ffffffff ffffffff 00000000 00000000 00000001"
        );
      }
      inherits(P224, MPrime);
      function P192() {
        MPrime.call(
          this,
          "p192",
          "ffffffff ffffffff ffffffff fffffffe ffffffff ffffffff"
        );
      }
      inherits(P192, MPrime);
      function P25519() {
        MPrime.call(
          this,
          "25519",
          "7fffffffffffffff ffffffffffffffff ffffffffffffffff ffffffffffffffed"
        );
      }
      inherits(P25519, MPrime);
      P25519.prototype.imulK = function imulK(num) {
        var carry = 0;
        for (var i = 0; i < num.length; i++) {
          var hi = (num.words[i] | 0) * 19 + carry;
          var lo = hi & 67108863;
          hi >>>= 26;
          num.words[i] = lo;
          carry = hi;
        }
        if (carry !== 0) {
          num.words[num.length++] = carry;
        }
        return num;
      };
      BN2._prime = function prime(name) {
        if (primes[name])
          return primes[name];
        var prime2;
        if (name === "k256") {
          prime2 = new K256();
        } else if (name === "p224") {
          prime2 = new P224();
        } else if (name === "p192") {
          prime2 = new P192();
        } else if (name === "p25519") {
          prime2 = new P25519();
        } else {
          throw new Error("Unknown prime " + name);
        }
        primes[name] = prime2;
        return prime2;
      };
      function Red(m) {
        if (typeof m === "string") {
          var prime = BN2._prime(m);
          this.m = prime.p;
          this.prime = prime;
        } else {
          assert2(m.gtn(1), "modulus must be greater than 1");
          this.m = m;
          this.prime = null;
        }
      }
      Red.prototype._verify1 = function _verify1(a) {
        assert2(a.negative === 0, "red works only with positives");
        assert2(a.red, "red works only with red numbers");
      };
      Red.prototype._verify2 = function _verify2(a, b) {
        assert2((a.negative | b.negative) === 0, "red works only with positives");
        assert2(
          a.red && a.red === b.red,
          "red works only with red numbers"
        );
      };
      Red.prototype.imod = function imod(a) {
        if (this.prime)
          return this.prime.ireduce(a)._forceRed(this);
        move(a, a.umod(this.m)._forceRed(this));
        return a;
      };
      Red.prototype.neg = function neg(a) {
        if (a.isZero()) {
          return a.clone();
        }
        return this.m.sub(a)._forceRed(this);
      };
      Red.prototype.add = function add2(a, b) {
        this._verify2(a, b);
        var res = a.add(b);
        if (res.cmp(this.m) >= 0) {
          res.isub(this.m);
        }
        return res._forceRed(this);
      };
      Red.prototype.iadd = function iadd(a, b) {
        this._verify2(a, b);
        var res = a.iadd(b);
        if (res.cmp(this.m) >= 0) {
          res.isub(this.m);
        }
        return res;
      };
      Red.prototype.sub = function sub(a, b) {
        this._verify2(a, b);
        var res = a.sub(b);
        if (res.cmpn(0) < 0) {
          res.iadd(this.m);
        }
        return res._forceRed(this);
      };
      Red.prototype.isub = function isub(a, b) {
        this._verify2(a, b);
        var res = a.isub(b);
        if (res.cmpn(0) < 0) {
          res.iadd(this.m);
        }
        return res;
      };
      Red.prototype.shl = function shl(a, num) {
        this._verify1(a);
        return this.imod(a.ushln(num));
      };
      Red.prototype.imul = function imul(a, b) {
        this._verify2(a, b);
        return this.imod(a.imul(b));
      };
      Red.prototype.mul = function mul(a, b) {
        this._verify2(a, b);
        return this.imod(a.mul(b));
      };
      Red.prototype.isqr = function isqr(a) {
        return this.imul(a, a.clone());
      };
      Red.prototype.sqr = function sqr(a) {
        return this.mul(a, a);
      };
      Red.prototype.sqrt = function sqrt(a) {
        if (a.isZero())
          return a.clone();
        var mod3 = this.m.andln(3);
        assert2(mod3 % 2 === 1);
        if (mod3 === 3) {
          var pow = this.m.add(new BN2(1)).iushrn(2);
          return this.pow(a, pow);
        }
        var q = this.m.subn(1);
        var s = 0;
        while (!q.isZero() && q.andln(1) === 0) {
          s++;
          q.iushrn(1);
        }
        assert2(!q.isZero());
        var one = new BN2(1).toRed(this);
        var nOne = one.redNeg();
        var lpow = this.m.subn(1).iushrn(1);
        var z = this.m.bitLength();
        z = new BN2(2 * z * z).toRed(this);
        while (this.pow(z, lpow).cmp(nOne) !== 0) {
          z.redIAdd(nOne);
        }
        var c = this.pow(z, q);
        var r = this.pow(a, q.addn(1).iushrn(1));
        var t = this.pow(a, q);
        var m = s;
        while (t.cmp(one) !== 0) {
          var tmp = t;
          for (var i = 0; tmp.cmp(one) !== 0; i++) {
            tmp = tmp.redSqr();
          }
          assert2(i < m);
          var b = this.pow(c, new BN2(1).iushln(m - i - 1));
          r = r.redMul(b);
          c = b.redSqr();
          t = t.redMul(c);
          m = i;
        }
        return r;
      };
      Red.prototype.invm = function invm(a) {
        var inv = a._invmp(this.m);
        if (inv.negative !== 0) {
          inv.negative = 0;
          return this.imod(inv).redNeg();
        } else {
          return this.imod(inv);
        }
      };
      Red.prototype.pow = function pow(a, num) {
        if (num.isZero())
          return new BN2(1).toRed(this);
        if (num.cmpn(1) === 0)
          return a.clone();
        var windowSize = 4;
        var wnd = new Array(1 << windowSize);
        wnd[0] = new BN2(1).toRed(this);
        wnd[1] = a;
        for (var i = 2; i < wnd.length; i++) {
          wnd[i] = this.mul(wnd[i - 1], a);
        }
        var res = wnd[0];
        var current = 0;
        var currentLen = 0;
        var start = num.bitLength() % 26;
        if (start === 0) {
          start = 26;
        }
        for (i = num.length - 1; i >= 0; i--) {
          var word = num.words[i];
          for (var j = start - 1; j >= 0; j--) {
            var bit = word >> j & 1;
            if (res !== wnd[0]) {
              res = this.sqr(res);
            }
            if (bit === 0 && current === 0) {
              currentLen = 0;
              continue;
            }
            current <<= 1;
            current |= bit;
            currentLen++;
            if (currentLen !== windowSize && (i !== 0 || j !== 0))
              continue;
            res = this.mul(res, wnd[current]);
            currentLen = 0;
            current = 0;
          }
          start = 26;
        }
        return res;
      };
      Red.prototype.convertTo = function convertTo(num) {
        var r = num.umod(this.m);
        return r === num ? r.clone() : r;
      };
      Red.prototype.convertFrom = function convertFrom(num) {
        var res = num.clone();
        res.red = null;
        return res;
      };
      BN2.mont = function mont(num) {
        return new Mont(num);
      };
      function Mont(m) {
        Red.call(this, m);
        this.shift = this.m.bitLength();
        if (this.shift % 26 !== 0) {
          this.shift += 26 - this.shift % 26;
        }
        this.r = new BN2(1).iushln(this.shift);
        this.r2 = this.imod(this.r.sqr());
        this.rinv = this.r._invmp(this.m);
        this.minv = this.rinv.mul(this.r).isubn(1).div(this.m);
        this.minv = this.minv.umod(this.r);
        this.minv = this.r.sub(this.minv);
      }
      inherits(Mont, Red);
      Mont.prototype.convertTo = function convertTo(num) {
        return this.imod(num.ushln(this.shift));
      };
      Mont.prototype.convertFrom = function convertFrom(num) {
        var r = this.imod(num.mul(this.rinv));
        r.red = null;
        return r;
      };
      Mont.prototype.imul = function imul(a, b) {
        if (a.isZero() || b.isZero()) {
          a.words[0] = 0;
          a.length = 1;
          return a;
        }
        var t = a.imul(b);
        var c = t.maskn(this.shift).mul(this.minv).imaskn(this.shift).mul(this.m);
        var u = t.isub(c).iushrn(this.shift);
        var res = u;
        if (u.cmp(this.m) >= 0) {
          res = u.isub(this.m);
        } else if (u.cmpn(0) < 0) {
          res = u.iadd(this.m);
        }
        return res._forceRed(this);
      };
      Mont.prototype.mul = function mul(a, b) {
        if (a.isZero() || b.isZero())
          return new BN2(0)._forceRed(this);
        var t = a.mul(b);
        var c = t.maskn(this.shift).mul(this.minv).imaskn(this.shift).mul(this.m);
        var u = t.isub(c).iushrn(this.shift);
        var res = u;
        if (u.cmp(this.m) >= 0) {
          res = u.isub(this.m);
        } else if (u.cmpn(0) < 0) {
          res = u.iadd(this.m);
        }
        return res._forceRed(this);
      };
      Mont.prototype.invm = function invm(a) {
        var res = this.imod(a._invmp(this.m).mul(this.r2));
        return res._forceRed(this);
      };
    })(typeof module2 === "undefined" || module2, exports);
  }
});

// src/utils/index.ts
var utils_exports = {};
__export(utils_exports, {
  BN: () => import_bn.default,
  BN_BILLION: () => BN_BILLION,
  BN_EIGHT: () => BN_EIGHT,
  BN_FIVE: () => BN_FIVE,
  BN_FOUR: () => BN_FOUR,
  BN_HUNDRED: () => BN_HUNDRED,
  BN_MAX_INTEGER: () => BN_MAX_INTEGER,
  BN_MILLION: () => BN_MILLION,
  BN_NINE: () => BN_NINE,
  BN_ONE: () => BN_ONE,
  BN_QUINTILL: () => BN_QUINTILL,
  BN_SEVEN: () => BN_SEVEN,
  BN_SIX: () => BN_SIX,
  BN_SQRT_MAX_INTEGER: () => BN_SQRT_MAX_INTEGER,
  BN_TEN: () => BN_TEN,
  BN_THOUSAND: () => BN_THOUSAND,
  BN_THREE: () => BN_THREE,
  BN_TWO: () => BN_TWO,
  BN_ZERO: () => BN_ZERO,
  MetadataError: () => MetadataError,
  NOOP: () => NOOP,
  U8A_WRAP_ETHEREUM: () => U8A_WRAP_ETHEREUM,
  U8A_WRAP_POSTFIX: () => U8A_WRAP_POSTFIX,
  U8A_WRAP_PREFIX: () => U8A_WRAP_PREFIX,
  _0n: () => _0n,
  _1000n: () => _1000n,
  _100n: () => _100n,
  _10n: () => _10n,
  _1Bn: () => _1Bn,
  _1Mn: () => _1Mn,
  _1Qn: () => _1Qn,
  _1n: () => _1n,
  _2n: () => _2n,
  _2pow53n: () => _2pow53n,
  _3n: () => _3n,
  _4n: () => _4n,
  _5n: () => _5n,
  _6n: () => _6n,
  _7n: () => _7n,
  _8n: () => _8n,
  _9n: () => _9n,
  _sqrt2pow53n: () => _sqrt2pow53n,
  arrayChunk: () => arrayChunk,
  arrayFilter: () => arrayFilter,
  arrayFlatten: () => arrayFlatten,
  arrayRange: () => arrayRange,
  arrayShuffle: () => arrayShuffle,
  arrayUnzip: () => arrayUnzip,
  arrayZip: () => arrayZip,
  asContractInstantiatedEvent: () => asContractInstantiatedEvent,
  asExtrinsicFailedEvent: () => asExtrinsicFailedEvent,
  assert: () => assert,
  assertReturn: () => assertReturn,
  assertUnreachable: () => assertUnreachable,
  bnFromHex: () => hexToBn,
  bnMax: () => bnMax,
  bnMin: () => bnMin,
  bnSqrt: () => bnSqrt,
  bnToBalance: () => bnToBalance,
  bnToBn: () => bnToBn,
  bnToHex: () => bnToHex,
  bnToU8a: () => bnToU8a,
  bufferToU8a: () => bufferToU8a,
  calcSi: () => calcSi,
  compactAddLength: () => compactAddLength,
  compactFromU8a: () => compactFromU8a,
  compactFromU8aLim: () => compactFromU8aLim,
  compactStripLength: () => compactStripLength,
  compactToU8a: () => compactToU8a,
  decimalToPlanck: () => decimalToPlanck,
  detectPackage: () => detectPackage,
  encodeSalt: () => encodeSalt,
  extractTime: () => extractTime,
  findSi: () => findSi,
  floatToU8a: () => floatToU8a,
  formatBalance: () => formatBalance,
  formatDate: () => formatDate,
  formatDecimal: () => formatDecimal,
  formatElapsed: () => formatElapsed,
  formatEventName: () => formatEventName,
  formatExtrinsicFailed: () => formatExtrinsicFailed,
  formatFileName: () => formatFileName,
  formatNumber: () => formatNumber,
  getExpiredItem: () => getExpiredItem,
  hasAny: () => hasAny,
  hasBigInt: () => hasBigInt,
  hasBuffer: () => hasBuffer,
  hasCjs: () => hasCjs,
  hasDirname: () => hasDirname,
  hasEsm: () => hasEsm,
  hasProcess: () => hasProcess,
  hasWasm: () => hasWasm,
  hexAddPrefix: () => hexAddPrefix,
  hexFixLength: () => hexFixLength,
  hexHasPrefix: () => hexHasPrefix,
  hexStripPrefix: () => hexStripPrefix,
  hexToBigInt: () => hexToBigInt,
  hexToBn: () => hexToBn,
  hexToNumber: () => hexToNumber,
  hexToString: () => hexToString,
  hexToU8a: () => hexToU8a,
  identity: () => identity,
  isArray: () => isArray,
  isAscii: () => isAscii,
  isBigInt: () => isBigInt,
  isBn: () => isBn,
  isBoolean: () => isBoolean,
  isBroadcast: () => isBroadcast,
  isBuffer: () => isBuffer,
  isChildClass: () => isChildClass,
  isClass: () => isClass,
  isCodec: () => isCodec,
  isCompact: () => isCompact,
  isContractInstantiatedEvent: () => isContractInstantiatedEvent,
  isDropped: () => isDropped,
  isDryRun: () => isDryRun,
  isError: () => isError,
  isErrored: () => isErrored,
  isExtrinsicFailedEvent: () => isExtrinsicFailedEvent,
  isFinalityTimeout: () => isFinalityTimeout,
  isFinalized: () => isFinalized,
  isFunction: () => isFunction,
  isFuture: () => isFuture,
  isHex: () => isHex,
  isInBlock: () => isInBlock,
  isInstanceOf: () => isInstanceOf,
  isInvalid: () => isInvalid,
  isIp: () => isIp,
  isJsonObject: () => isJsonObject,
  isNone: () => isNone,
  isNull: () => isNull,
  isNumber: () => isNumber,
  isObject: () => isObject,
  isObservable: () => isObservable,
  isPendingSignature: () => isPendingSignature,
  isPromise: () => isPromise,
  isReady: () => isReady,
  isRetracted: () => isRetracted,
  isString: () => isString,
  isTestChain: () => isTestChain,
  isToBigInt: () => isToBigInt,
  isToBn: () => isToBn,
  isTxCancelledError: () => isTxCancelledError,
  isU8a: () => isU8a,
  isUndefined: () => isUndefined,
  isUsurped: () => isUsurped,
  isUtf8: () => isUtf8,
  isValidHash: () => isValidHash,
  isWasm: () => isWasm,
  lazyMethod: () => lazyMethod,
  lazyMethods: () => lazyMethods,
  logger: () => logger,
  loggerFormat: () => loggerFormat,
  memoize: () => memoize,
  nMax: () => nMax,
  nMin: () => nMin,
  nSqrt: () => nSqrt,
  nToBigInt: () => nToBigInt,
  nToHex: () => nToHex,
  nToU8a: () => nToU8a,
  nextTick: () => nextTick,
  noop: () => noop,
  numberToHex: () => numberToHex,
  numberToU8a: () => numberToU8a,
  objectClear: () => objectClear,
  objectCopy: () => objectCopy,
  objectEntries: () => objectEntries,
  objectKeys: () => objectKeys,
  objectProperties: () => objectProperties,
  objectProperty: () => objectProperty,
  objectSpread: () => objectSpread,
  objectValues: () => objectValues,
  packageInfo: () => packageInfo,
  pickCallInfo: () => pickCallInfo,
  pickDecoded: () => pickDecoded,
  pickDecodedError: () => pickDecodedError,
  pickError: () => pickError,
  pickResultErr: () => pickResultErr,
  pickResultOk: () => pickResultOk,
  pickTxInfo: () => pickTxInfo,
  planckToDecimal: () => planckToDecimal,
  planckToDecimalFormatted: () => planckToDecimalFormatted,
  promisify: () => promisify,
  pseudoRandomHex: () => pseudoRandomHex,
  pseudoRandomId: () => pseudoRandomId,
  pseudoRandomU8a: () => pseudoRandomU8a,
  shouldDisable: () => shouldDisable,
  shouldDisableStrict: () => shouldDisableStrict,
  stringCamelCase: () => stringCamelCase,
  stringLowerFirst: () => stringLowerFirst,
  stringNumberToBN: () => stringNumberToBN,
  stringPascalCase: () => stringPascalCase,
  stringShorten: () => stringShorten,
  stringToHex: () => stringToHex,
  stringToU8a: () => stringToU8a,
  stringUpperFirst: () => stringUpperFirst,
  stringify: () => stringify,
  toBasicMetadata: () => toBasicMetadata,
  toJSType: () => toJSType,
  toMessageParams: () => toMessageParams,
  u8aCmp: () => u8aCmp,
  u8aConcat: () => u8aConcat,
  u8aConcatStrict: () => u8aConcatStrict,
  u8aEmpty: () => u8aEmpty,
  u8aEq: () => u8aEq,
  u8aFixLength: () => u8aFixLength,
  u8aIsWrapped: () => u8aIsWrapped,
  u8aSorted: () => u8aSorted,
  u8aToBigInt: () => u8aToBigInt,
  u8aToBn: () => u8aToBn,
  u8aToBuffer: () => u8aToBuffer,
  u8aToFloat: () => u8aToFloat,
  u8aToHex: () => u8aToHex,
  u8aToNumber: () => u8aToNumber,
  u8aToString: () => u8aToString,
  u8aToU8a: () => u8aToU8a,
  u8aUnwrapBytes: () => u8aUnwrapBytes,
  u8aWrapBytes: () => u8aWrapBytes,
  unixMilliToDate: () => unixMilliToDate,
  validateMetadata: () => validateMetadata
});
module.exports = __toCommonJS(utils_exports);

// ../../node_modules/.pnpm/@polkadot+x-textdecoder@12.3.2/node_modules/@polkadot/x-textdecoder/node.js
var import_node_util = __toESM(require("util"), 1);

// ../../node_modules/.pnpm/@polkadot+x-global@12.3.2/node_modules/@polkadot/x-global/index.js
function evaluateThis(fn) {
  return fn("return this");
}
var xglobal = typeof globalThis !== "undefined" ? globalThis : typeof global !== "undefined" ? global : typeof self !== "undefined" ? self : typeof window !== "undefined" ? window : evaluateThis(Function);
function extractGlobal(name, fallback) {
  return typeof xglobal[name] === "undefined" ? fallback : xglobal[name];
}

// ../../node_modules/.pnpm/@polkadot+x-textdecoder@12.3.2/node_modules/@polkadot/x-textdecoder/node.js
var TextDecoder = /* @__PURE__ */ extractGlobal("TextDecoder", import_node_util.default.TextDecoder);

// ../../node_modules/.pnpm/@polkadot+x-textencoder@12.3.2/node_modules/@polkadot/x-textencoder/node.js
var import_node_util2 = __toESM(require("util"), 1);
var Fallback = class {
  constructor() {
    this.__internal__encoder = new import_node_util2.default.TextEncoder();
  }
  // For a Jest 26.0.1 environment, Buffer !== Uint8Array
  encode(value) {
    return Uint8Array.from(this.__internal__encoder.encode(value));
  }
};
var TextEncoder2 = /* @__PURE__ */ extractGlobal("TextEncoder", Fallback);

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/packageInfo.js
var import_meta = {};
var packageInfo = { name: "@polkadot/util", path: import_meta && import_meta.url ? new URL(import_meta.url).pathname.substring(0, new URL(import_meta.url).pathname.lastIndexOf("/") + 1) : "auto", type: "esm", version: "12.3.2" };

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/function.js
function isFunction(value) {
  return typeof value === "function";
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/versionDetect.js
var DEDUPE = "Either remove and explicitly install matching versions or dedupe using your package manager.\nThe following conflicting packages were found:";
function getEntry(name) {
  const _global = xglobal;
  if (!_global.__polkadotjs) {
    _global.__polkadotjs = {};
  }
  if (!_global.__polkadotjs[name]) {
    _global.__polkadotjs[name] = [];
  }
  return _global.__polkadotjs[name];
}
function formatDisplay(all, fmt) {
  let max = 0;
  for (let i = 0, count = all.length; i < count; i++) {
    max = Math.max(max, all[i].version.length);
  }
  return all.map((d) => `	${fmt(d.version.padEnd(max), d).join("	")}`).join("\n");
}
function formatInfo(version, { name }) {
  return [
    version,
    name
  ];
}
function formatVersion(version, { path, type }) {
  let extracted;
  if (path && path.length >= 5) {
    const nmIndex = path.indexOf("node_modules");
    extracted = nmIndex === -1 ? path : path.substring(nmIndex);
  } else {
    extracted = "<unknown>";
  }
  return [
    `${`${type || ""}`.padStart(3)} ${version}`,
    extracted
  ];
}
function getPath(infoPath, pathOrFn) {
  if (infoPath) {
    return infoPath;
  } else if (isFunction(pathOrFn)) {
    try {
      return pathOrFn() || "";
    } catch {
      return "";
    }
  }
  return pathOrFn || "";
}
function warn(pre, all, fmt) {
  console.warn(`${pre}
${DEDUPE}
${formatDisplay(all, fmt)}`);
}
function detectPackage({ name, path, type, version }, pathOrFn, deps = []) {
  if (!name.startsWith("@polkadot")) {
    throw new Error(`Invalid package descriptor ${name}`);
  }
  const entry = getEntry(name);
  entry.push({ path: getPath(path, pathOrFn), type, version });
  if (entry.length !== 1) {
    warn(`${name} has multiple versions, ensure that there is only one installed.`, entry, formatVersion);
  } else {
    const mismatches = deps.filter((d) => d && d.version !== version);
    if (mismatches.length) {
      warn(`${name} requires direct dependencies exactly matching version ${version}.`, mismatches, formatInfo);
    }
  }
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/array/chunk.js
function arrayChunk(array, chunkSize) {
  const outputSize = Math.ceil(array.length / chunkSize);
  if (outputSize === 1) {
    return [array];
  }
  const output = Array(outputSize);
  for (let i = 0; i < outputSize; i++) {
    const offset = i * chunkSize;
    output[i] = array.slice(offset, offset + chunkSize);
  }
  return output;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/array/filter.js
function arrayFilter(array, allowNulls = true) {
  return array.filter((v) => v !== void 0 && (allowNulls || v !== null));
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/array/flatten.js
function arrayFlatten(arrays) {
  const num = arrays.length;
  if (num === 0) {
    return [];
  } else if (num === 1) {
    return arrays[0];
  }
  let size = 0;
  for (let i2 = 0; i2 < num; i2++) {
    size += arrays[i2].length;
  }
  const output = new Array(size);
  let i = -1;
  for (let j = 0; j < num; j++) {
    const a = arrays[j];
    for (let e = 0, count = a.length; e < count; e++) {
      output[++i] = a[e];
    }
  }
  return output;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/array/range.js
function arrayRange(size, startAt = 0) {
  if (size <= 0) {
    throw new Error("Expected non-zero, positive number as a range size");
  }
  const result = new Array(size);
  for (let i = 0; i < size; i++) {
    result[i] = i + startAt;
  }
  return result;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/array/shuffle.js
function arrayShuffle(input) {
  const result = input.slice();
  let curr = result.length;
  if (curr === 1) {
    return result;
  }
  while (curr !== 0) {
    const rand = ~~(Math.random() * curr);
    curr--;
    [result[curr], result[rand]] = [result[rand], result[curr]];
  }
  return result;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/array/unzip.js
function arrayUnzip(entries) {
  const count = entries.length;
  const keys = new Array(count);
  const values = new Array(count);
  for (let i = 0; i < count; i++) {
    [keys[i], values[i]] = entries[i];
  }
  return [keys, values];
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/array/zip.js
function arrayZip(keys, values) {
  const count = keys.length;
  const result = new Array(count);
  for (let i = 0; i < count; i++) {
    result[i] = [keys[i], values[i]];
  }
  return result;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/assert.js
function assert(condition, message) {
  if (!condition) {
    throw new Error(isFunction(message) ? message() : message);
  }
}
function assertReturn(value, message) {
  assert(value !== void 0 && value !== null, message);
  return value;
}
function assertUnreachable(x) {
  throw new Error(`This codepath should be unreachable. Unhandled input: ${x}`);
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/bi/helpers.js
function createCmp(cmp) {
  return (...items) => {
    const count = items.length;
    if (count === 0) {
      throw new Error("Must provide one or more arguments");
    }
    let result = items[0];
    for (let i = 1; i < count; i++) {
      if (cmp(items[i], result)) {
        result = items[i];
      }
    }
    return result;
  };
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/bi/min.js
var nMax = /* @__PURE__ */ createCmp((a, b) => a > b);
var nMin = /* @__PURE__ */ createCmp((a, b) => a < b);

// ../../node_modules/.pnpm/@polkadot+x-bigint@12.3.2/node_modules/@polkadot/x-bigint/index.js
function invalidFallback() {
  return Number.NaN;
}
var BigInt2 = /* @__PURE__ */ extractGlobal("BigInt", invalidFallback);

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/bi/consts.js
var _0n = /* @__PURE__ */ BigInt2(0);
var _1n = /* @__PURE__ */ BigInt2(1);
var _2n = /* @__PURE__ */ BigInt2(2);
var _3n = /* @__PURE__ */ BigInt2(3);
var _4n = /* @__PURE__ */ BigInt2(4);
var _5n = /* @__PURE__ */ BigInt2(5);
var _6n = /* @__PURE__ */ BigInt2(6);
var _7n = /* @__PURE__ */ BigInt2(7);
var _8n = /* @__PURE__ */ BigInt2(8);
var _9n = /* @__PURE__ */ BigInt2(9);
var _10n = /* @__PURE__ */ BigInt2(10);
var _100n = /* @__PURE__ */ BigInt2(100);
var _1000n = /* @__PURE__ */ BigInt2(1e3);
var _1Mn = /* @__PURE__ */ BigInt2(1e6);
var _1Bn = /* @__PURE__ */ BigInt2(1e9);
var _1Qn = _1Bn * _1Bn;
var _2pow53n = /* @__PURE__ */ BigInt2(Number.MAX_SAFE_INTEGER);
var _sqrt2pow53n = /* @__PURE__ */ BigInt2(94906265);

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/u8a/toBigInt.js
var U8_MAX = BigInt2(256);
var U16_MAX = BigInt2(256 * 256);
var U64_MAX = BigInt2("0x10000000000000000");
function u8aToBigInt(value, { isLe = true, isNegative = false } = {}) {
  const u8a = isLe ? value : value.slice().reverse();
  const count = u8a.length;
  if (isNegative) {
    switch (count) {
      case 0:
        return BigInt2(0);
      case 1:
        return BigInt2((u8a[0] ^ 255) * -1 - 1);
      case 2:
        return BigInt2((u8a[0] + (u8a[1] << 8) ^ 65535) * -1 - 1);
      case 4:
        return BigInt2((u8a[0] + (u8a[1] << 8) + (u8a[2] << 16) + u8a[3] * 16777216 ^ 4294967295) * -1 - 1);
    }
    const dvI2 = new DataView(u8a.buffer, u8a.byteOffset);
    if (count === 8) {
      return dvI2.getBigInt64(0, true);
    }
    let result = BigInt2(0);
    const mod = count % 2;
    for (let i = count - 2; i >= mod; i -= 2) {
      result = result * U16_MAX + BigInt2(dvI2.getUint16(i, true) ^ 65535);
    }
    if (mod) {
      result = result * U8_MAX + BigInt2(u8a[0] ^ 255);
    }
    return result * -_1n - _1n;
  }
  switch (count) {
    case 0:
      return BigInt2(0);
    case 1:
      return BigInt2(u8a[0]);
    case 2:
      return BigInt2(u8a[0] + (u8a[1] << 8));
    case 4:
      return BigInt2(u8a[0] + (u8a[1] << 8) + (u8a[2] << 16) + u8a[3] * 16777216);
  }
  const dvI = new DataView(u8a.buffer, u8a.byteOffset);
  switch (count) {
    case 8:
      return dvI.getBigUint64(0, true);
    case 16:
      return dvI.getBigUint64(8, true) * U64_MAX + dvI.getBigUint64(0, true);
    default: {
      let result = BigInt2(0);
      const mod = count % 2;
      for (let i = count - 2; i >= mod; i -= 2) {
        result = result * U16_MAX + BigInt2(dvI.getUint16(i, true));
      }
      if (mod) {
        result = result * U8_MAX + BigInt2(u8a[0]);
      }
      return result;
    }
  }
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/hex/toU8a.js
var CHR = "0123456789abcdef";
var U8 = new Uint8Array(256);
var U16 = new Uint8Array(256 * 256);
for (let i = 0, count = CHR.length; i < count; i++) {
  U8[CHR[i].charCodeAt(0) | 0] = i | 0;
  if (i > 9) {
    U8[CHR[i].toUpperCase().charCodeAt(0) | 0] = i | 0;
  }
}
for (let i = 0; i < 256; i++) {
  const s = i << 8;
  for (let j = 0; j < 256; j++) {
    U16[s | j] = U8[i] << 4 | U8[j];
  }
}
function hexToU8a(value, bitLength = -1) {
  if (!value) {
    return new Uint8Array();
  }
  let s = value.startsWith("0x") ? 2 : 0;
  const decLength = Math.ceil((value.length - s) / 2);
  const endLength = Math.ceil(bitLength === -1 ? decLength : bitLength / 8);
  const result = new Uint8Array(endLength);
  const offset = endLength > decLength ? endLength - decLength : 0;
  for (let i = offset; i < endLength; i++, s += 2) {
    result[i] = U16[value.charCodeAt(s) << 8 | value.charCodeAt(s + 1)];
  }
  return result;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/hex/toBigInt.js
function hexToBigInt(value, { isLe = false, isNegative = false } = {}) {
  return !value || value === "0x" ? BigInt2(0) : u8aToBigInt(hexToU8a(value), { isLe, isNegative });
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/bn/bn.js
var import_bn = __toESM(require_bn(), 1);

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/bn.js
function isBn(value) {
  return import_bn.default.isBN(value);
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/hex.js
var REGEX_HEX_PREFIXED = /^0x[\da-fA-F]+$/;
var REGEX_HEX_NOPREFIX = /^[\da-fA-F]+$/;
function isHex(value, bitLength = -1, ignoreLength) {
  return typeof value === "string" && (value === "0x" || REGEX_HEX_PREFIXED.test(value)) && (bitLength === -1 ? ignoreLength || value.length % 2 === 0 : value.length === 2 + Math.ceil(bitLength / 4));
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/object.js
function isObject(value) {
  return !!value && typeof value === "object";
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/helpers.js
function isOn(...fns) {
  return (value) => (isObject(value) || isFunction(value)) && fns.every((f) => isFunction(value[f]));
}
function isOnFunction(...fns) {
  return (value) => isFunction(value) && fns.every((f) => isFunction(value[f]));
}
function isOnObject(...fns) {
  return (value) => isObject(value) && fns.every((f) => isFunction(value[f]));
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/toBigInt.js
var isToBigInt = /* @__PURE__ */ isOn("toBigInt");

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/toBn.js
var isToBn = /* @__PURE__ */ isOn("toBn");

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/bi/toBigInt.js
function nToBigInt(value) {
  return typeof value === "bigint" ? value : !value ? BigInt2(0) : isHex(value) ? hexToBigInt(value.toString()) : isBn(value) ? BigInt2(value.toString()) : isToBigInt(value) ? value.toBigInt() : isToBn(value) ? BigInt2(value.toBn().toString()) : BigInt2(value);
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/bi/sqrt.js
function nSqrt(value) {
  const n = nToBigInt(value);
  if (n < _0n) {
    throw new Error("square root of negative numbers is not supported");
  }
  if (n <= _2pow53n) {
    return BigInt2(~~Math.sqrt(Number(n)));
  }
  let x0 = _sqrt2pow53n;
  while (true) {
    const x1 = n / x0 + x0 >> _1n;
    if (x0 === x1 || x0 === x1 - _1n) {
      return x0;
    }
    x0 = x1;
  }
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/has.js
var hasBigInt = typeof BigInt2 === "function" && typeof BigInt2.asIntN === "function";
var hasCjs = typeof require === "function" && typeof module !== "undefined";
var hasDirname = typeof __dirname !== "undefined";
var hasEsm = !hasCjs;
var hasWasm = typeof WebAssembly !== "undefined";
var hasBuffer = typeof xglobal.Buffer === "function" && typeof xglobal.Buffer.isBuffer === "function";
var hasProcess = typeof xglobal.process === "object";

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/buffer.js
function isBuffer(value) {
  return hasBuffer && !!value && isFunction(value.readDoubleLE) && xglobal.Buffer.isBuffer(value);
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/u8a.js
function isU8a(value) {
  return (value && value.constructor) === Uint8Array || value instanceof Uint8Array;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/string/toU8a.js
var encoder = new TextEncoder2();
function stringToU8a(value) {
  return value ? encoder.encode(value.toString()) : new Uint8Array();
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/u8a/toU8a.js
function u8aToU8a(value) {
  return isU8a(value) ? isBuffer(value) ? new Uint8Array(value) : value : isHex(value) ? hexToU8a(value) : Array.isArray(value) ? new Uint8Array(value) : stringToU8a(value);
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/u8a/cmp.js
function u8aCmp(a, b) {
  const u8aa = u8aToU8a(a);
  const u8ab = u8aToU8a(b);
  let i = 0;
  while (true) {
    const overA = i >= u8aa.length;
    const overB = i >= u8ab.length;
    if (overA && overB) {
      return 0;
    } else if (overA) {
      return -1;
    } else if (overB) {
      return 1;
    } else if (u8aa[i] !== u8ab[i]) {
      return u8aa[i] > u8ab[i] ? 1 : -1;
    }
    i++;
  }
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/u8a/concat.js
function u8aConcat(...list) {
  const count = list.length;
  const u8as = new Array(count);
  let length = 0;
  for (let i = 0; i < count; i++) {
    u8as[i] = u8aToU8a(list[i]);
    length += u8as[i].length;
  }
  return u8aConcatStrict(u8as, length);
}
function u8aConcatStrict(u8as, length = 0) {
  const count = u8as.length;
  let offset = 0;
  if (!length) {
    for (let i = 0; i < count; i++) {
      length += u8as[i].length;
    }
  }
  const result = new Uint8Array(length);
  for (let i = 0; i < count; i++) {
    result.set(u8as[i], offset);
    offset += u8as[i].length;
  }
  return result;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/u8a/empty.js
function u8aEmpty(value) {
  const len = value.length | 0;
  for (let i = 0; i < len; i++) {
    if (value[i] | 0) {
      return false;
    }
  }
  return true;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/u8a/eq.js
function u8aEq(a, b) {
  const u8aa = u8aToU8a(a);
  const u8ab = u8aToU8a(b);
  if (u8aa.length === u8ab.length) {
    const dvA = new DataView(u8aa.buffer, u8aa.byteOffset);
    const dvB = new DataView(u8ab.buffer, u8ab.byteOffset);
    const mod = u8aa.length % 4 | 0;
    const length = u8aa.length - mod | 0;
    for (let i = 0; i < length; i += 4) {
      if (dvA.getUint32(i) !== dvB.getUint32(i)) {
        return false;
      }
    }
    for (let i = length, count = u8aa.length; i < count; i++) {
      if (u8aa[i] !== u8ab[i]) {
        return false;
      }
    }
    return true;
  }
  return false;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/u8a/fixLength.js
function u8aFixLength(value, bitLength = -1, atStart = false) {
  const byteLength = Math.ceil(bitLength / 8);
  if (bitLength === -1 || value.length === byteLength) {
    return value;
  } else if (value.length > byteLength) {
    return value.subarray(0, byteLength);
  }
  const result = new Uint8Array(byteLength);
  result.set(value, atStart ? 0 : byteLength - value.length);
  return result;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/u8a/sorted.js
function u8aSorted(u8as) {
  return u8as.sort(u8aCmp);
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/u8a/toBn.js
function u8aToBn(value, { isLe = true, isNegative = false } = {}) {
  const count = value.length;
  if (isNegative) {
    if (isLe) {
      switch (count) {
        case 0:
          return new import_bn.default(0);
        case 1:
          return new import_bn.default((value[0] ^ 255) * -1 - 1);
        case 2:
          return new import_bn.default((value[0] + (value[1] << 8) ^ 65535) * -1 - 1);
        case 3:
          return new import_bn.default((value[0] + (value[1] << 8) + (value[2] << 16) ^ 16777215) * -1 - 1);
        case 4:
          return new import_bn.default((value[0] + (value[1] << 8) + (value[2] << 16) + value[3] * 16777216 ^ 4294967295) * -1 - 1);
        case 5:
          return new import_bn.default(((value[0] + (value[1] << 8) + (value[2] << 16) + value[3] * 16777216 ^ 4294967295) + (value[4] ^ 255) * 4294967296) * -1 - 1);
        case 6:
          return new import_bn.default(((value[0] + (value[1] << 8) + (value[2] << 16) + value[3] * 16777216 ^ 4294967295) + (value[4] + (value[5] << 8) ^ 65535) * 4294967296) * -1 - 1);
        default:
          return new import_bn.default(value, "le").fromTwos(value.length * 8);
      }
    }
    if (count === 0) {
      return new import_bn.default(0);
    } else if (count > 6) {
      return new import_bn.default(value, "be").fromTwos(value.length * 8);
    }
    let result2 = 0;
    for (let i = 0; i < count; i++) {
      result2 = result2 * 256 + (value[i] ^ 255);
    }
    return new import_bn.default(result2 * -1 - 1);
  }
  if (isLe) {
    switch (count) {
      case 0:
        return new import_bn.default(0);
      case 1:
        return new import_bn.default(value[0]);
      case 2:
        return new import_bn.default(value[0] + (value[1] << 8));
      case 3:
        return new import_bn.default(value[0] + (value[1] << 8) + (value[2] << 16));
      case 4:
        return new import_bn.default(value[0] + (value[1] << 8) + (value[2] << 16) + value[3] * 16777216);
      case 5:
        return new import_bn.default(value[0] + (value[1] << 8) + (value[2] << 16) + (value[3] + (value[4] << 8)) * 16777216);
      case 6:
        return new import_bn.default(value[0] + (value[1] << 8) + (value[2] << 16) + (value[3] + (value[4] << 8) + (value[5] << 16)) * 16777216);
      default:
        return new import_bn.default(value, "le");
    }
  }
  if (count === 0) {
    return new import_bn.default(0);
  } else if (count > 6) {
    return new import_bn.default(value, "be");
  }
  let result = 0;
  for (let i = 0; i < count; i++) {
    result = result * 256 + value[i];
  }
  return new import_bn.default(result);
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/u8a/toBuffer.js
function u8aToBuffer(value) {
  return Buffer.from(value || []);
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/u8a/toFloat.js
function u8aToFloat(value, { bitLength = 32, isLe = true } = {}) {
  if (bitLength !== 32 && bitLength !== 64) {
    throw new Error("Invalid bitLength provided, expected 32 or 64");
  } else if (value.length < bitLength / 8) {
    throw new Error(`Invalid input buffer provided, expected at least ${bitLength / 8} bytes, found ${value.length}`);
  }
  const dv = new DataView(value.buffer, value.byteOffset);
  return bitLength === 32 ? dv.getFloat32(0, isLe) : dv.getFloat64(0, isLe);
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/u8a/toHex.js
var U82 = new Array(256);
var U162 = new Array(256 * 256);
for (let n = 0; n < 256; n++) {
  U82[n] = n.toString(16).padStart(2, "0");
}
for (let i = 0; i < 256; i++) {
  const s = i << 8;
  for (let j = 0; j < 256; j++) {
    U162[s | j] = U82[i] + U82[j];
  }
}
function hex(value, result) {
  const mod = value.length % 2 | 0;
  const length = value.length - mod | 0;
  for (let i = 0; i < length; i += 2) {
    result += U162[value[i] << 8 | value[i + 1]];
  }
  if (mod) {
    result += U82[value[length] | 0];
  }
  return result;
}
function u8aToHex(value, bitLength = -1, isPrefixed = true) {
  const empty = isPrefixed ? "0x" : "";
  if (!value || !value.length) {
    return empty;
  } else if (bitLength > 0) {
    const length = Math.ceil(bitLength / 8);
    if (value.length > length) {
      return `${hex(value.subarray(0, length / 2), empty)}\u2026${hex(value.subarray(value.length - length / 2), "")}`;
    }
  }
  return hex(value, empty);
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/u8a/toNumber.js
function u8aToNumber(value, { isNegative = false } = {}) {
  const count = value.length;
  if (isNegative) {
    switch (count) {
      case 0:
        return 0;
      case 1:
        return (value[0] ^ 255) * -1 - 1;
      case 2:
        return (value[0] + (value[1] << 8) ^ 65535) * -1 - 1;
      case 3:
        return (value[0] + (value[1] << 8) + (value[2] << 16) ^ 16777215) * -1 - 1;
      case 4:
        return (value[0] + (value[1] << 8) + (value[2] << 16) + value[3] * 16777216 ^ 4294967295) * -1 - 1;
      case 5:
        return ((value[0] + (value[1] << 8) + (value[2] << 16) + value[3] * 16777216 ^ 4294967295) + (value[4] ^ 255) * 4294967296) * -1 - 1;
      case 6:
        return ((value[0] + (value[1] << 8) + (value[2] << 16) + value[3] * 16777216 ^ 4294967295) + (value[4] + (value[5] << 8) ^ 65535) * 4294967296) * -1 - 1;
      default:
        throw new Error("Value more than 48-bits cannot be reliably converted");
    }
  }
  switch (count) {
    case 0:
      return 0;
    case 1:
      return value[0];
    case 2:
      return value[0] + (value[1] << 8);
    case 3:
      return value[0] + (value[1] << 8) + (value[2] << 16);
    case 4:
      return value[0] + (value[1] << 8) + (value[2] << 16) + value[3] * 16777216;
    case 5:
      return value[0] + (value[1] << 8) + (value[2] << 16) + (value[3] + (value[4] << 8)) * 16777216;
    case 6:
      return value[0] + (value[1] << 8) + (value[2] << 16) + (value[3] + (value[4] << 8) + (value[5] << 16)) * 16777216;
    default:
      throw new Error("Value more than 48-bits cannot be reliably converted");
  }
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/u8a/toString.js
var decoder = new TextDecoder("utf-8");
function u8aToString(value) {
  return value ? decoder.decode(value) : "";
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/u8a/wrap.js
var U8A_WRAP_ETHEREUM = /* @__PURE__ */ u8aToU8a("Ethereum Signed Message:\n");
var U8A_WRAP_PREFIX = /* @__PURE__ */ u8aToU8a("<Bytes>");
var U8A_WRAP_POSTFIX = /* @__PURE__ */ u8aToU8a("</Bytes>");
var WRAP_LEN = U8A_WRAP_PREFIX.length + U8A_WRAP_POSTFIX.length;
function u8aIsWrapped(u8a, withEthereum) {
  return u8a.length >= WRAP_LEN && u8aEq(u8a.subarray(0, U8A_WRAP_PREFIX.length), U8A_WRAP_PREFIX) && u8aEq(u8a.slice(-U8A_WRAP_POSTFIX.length), U8A_WRAP_POSTFIX) || withEthereum && u8a.length >= U8A_WRAP_ETHEREUM.length && u8aEq(u8a.subarray(0, U8A_WRAP_ETHEREUM.length), U8A_WRAP_ETHEREUM);
}
function u8aUnwrapBytes(bytes) {
  const u8a = u8aToU8a(bytes);
  return u8aIsWrapped(u8a, false) ? u8a.subarray(U8A_WRAP_PREFIX.length, u8a.length - U8A_WRAP_POSTFIX.length) : u8a;
}
function u8aWrapBytes(bytes) {
  const u8a = u8aToU8a(bytes);
  return u8aIsWrapped(u8a, true) ? u8a : u8aConcatStrict([U8A_WRAP_PREFIX, u8a, U8A_WRAP_POSTFIX]);
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/bi/toU8a.js
var DIV = BigInt2(256);
var NEG_MASK = BigInt2(255);
function toU8a(value, isLe, isNegative) {
  const arr = [];
  if (isNegative) {
    value = (value + _1n) * -_1n;
  }
  while (value !== _0n) {
    const mod = value % DIV;
    const val = Number(isNegative ? mod ^ NEG_MASK : mod);
    if (isLe) {
      arr.push(val);
    } else {
      arr.unshift(val);
    }
    value = (value - mod) / DIV;
  }
  return Uint8Array.from(arr);
}
function nToU8a(value, { bitLength = -1, isLe = true, isNegative = false } = {}) {
  const valueBi = nToBigInt(value);
  if (valueBi === _0n) {
    return bitLength === -1 ? new Uint8Array(1) : new Uint8Array(Math.ceil((bitLength || 0) / 8));
  }
  const u8a = toU8a(valueBi, isLe, isNegative);
  if (bitLength === -1) {
    return u8a;
  }
  const byteLength = Math.ceil((bitLength || 0) / 8);
  const output = new Uint8Array(byteLength);
  if (isNegative) {
    output.fill(255);
  }
  output.set(u8a, isLe ? 0 : byteLength - u8a.length);
  return output;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/bi/toHex.js
function nToHex(value, { bitLength = -1, isLe = false, isNegative = false } = {}) {
  return u8aToHex(nToU8a(value || 0, { bitLength, isLe, isNegative }));
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/hex/stripPrefix.js
function hexStripPrefix(value) {
  if (!value || value === "0x") {
    return "";
  } else if (REGEX_HEX_PREFIXED.test(value)) {
    return value.substring(2);
  } else if (REGEX_HEX_NOPREFIX.test(value)) {
    return value;
  }
  throw new Error(`Expected hex value to convert, found '${value}'`);
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/hex/toBn.js
function hexToBn(value, { isLe = false, isNegative = false } = {}) {
  if (!value || value === "0x") {
    return new import_bn.default(0);
  }
  const stripped = hexStripPrefix(value);
  const bn = new import_bn.default(stripped, 16, isLe ? "le" : "be");
  return isNegative ? bn.fromTwos(stripped.length * 4) : bn;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/bn/min.js
var bnMax = /* @__PURE__ */ createCmp((a, b) => a.gt(b));
var bnMin = /* @__PURE__ */ createCmp((a, b) => a.lt(b));

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/bn/consts.js
var BN_ZERO = /* @__PURE__ */ new import_bn.default(0);
var BN_ONE = /* @__PURE__ */ new import_bn.default(1);
var BN_TWO = /* @__PURE__ */ new import_bn.default(2);
var BN_THREE = /* @__PURE__ */ new import_bn.default(3);
var BN_FOUR = /* @__PURE__ */ new import_bn.default(4);
var BN_FIVE = /* @__PURE__ */ new import_bn.default(5);
var BN_SIX = /* @__PURE__ */ new import_bn.default(6);
var BN_SEVEN = /* @__PURE__ */ new import_bn.default(7);
var BN_EIGHT = /* @__PURE__ */ new import_bn.default(8);
var BN_NINE = /* @__PURE__ */ new import_bn.default(9);
var BN_TEN = /* @__PURE__ */ new import_bn.default(10);
var BN_HUNDRED = /* @__PURE__ */ new import_bn.default(100);
var BN_THOUSAND = /* @__PURE__ */ new import_bn.default(1e3);
var BN_MILLION = /* @__PURE__ */ new import_bn.default(1e6);
var BN_BILLION = /* @__PURE__ */ new import_bn.default(1e9);
var BN_QUINTILL = BN_BILLION.mul(BN_BILLION);
var BN_MAX_INTEGER = /* @__PURE__ */ new import_bn.default(Number.MAX_SAFE_INTEGER);
var BN_SQRT_MAX_INTEGER = /* @__PURE__ */ new import_bn.default(94906265);

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/bigInt.js
function isBigInt(value) {
  return typeof value === "bigint";
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/bn/toBn.js
function bnToBn(value) {
  return value ? import_bn.default.isBN(value) ? value : isHex(value) ? hexToBn(value.toString()) : isBigInt(value) ? new import_bn.default(value.toString()) : isToBn(value) ? value.toBn() : isToBigInt(value) ? new import_bn.default(value.toBigInt().toString()) : new import_bn.default(value) : new import_bn.default(0);
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/bn/sqrt.js
function bnSqrt(value) {
  const n = bnToBn(value);
  if (n.isNeg()) {
    throw new Error("square root of negative numbers is not supported");
  }
  if (n.lte(BN_MAX_INTEGER)) {
    return new import_bn.default(~~Math.sqrt(n.toNumber()));
  }
  let x0 = BN_SQRT_MAX_INTEGER.clone();
  while (true) {
    const x1 = n.div(x0).iadd(x0).ishrn(1);
    if (x0.eq(x1) || x0.eq(x1.sub(BN_ONE))) {
      return x0;
    }
    x0 = x1;
  }
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/bn/toU8a.js
var DEFAULT_OPTS = { bitLength: -1, isLe: true, isNegative: false };
function bnToU8a(value, { bitLength = -1, isLe = true, isNegative = false } = DEFAULT_OPTS) {
  const valueBn = bnToBn(value);
  const byteLength = bitLength === -1 ? Math.ceil(valueBn.bitLength() / 8) : Math.ceil((bitLength || 0) / 8);
  if (!value) {
    return bitLength === -1 ? new Uint8Array(1) : new Uint8Array(byteLength);
  }
  const output = new Uint8Array(byteLength);
  const bn = isNegative ? valueBn.toTwos(byteLength * 8) : valueBn;
  output.set(bn.toArray(isLe ? "le" : "be", byteLength), 0);
  return output;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/bn/toHex.js
function bnToHex(value, { bitLength = -1, isLe = false, isNegative = false } = {}) {
  return u8aToHex(bnToU8a(value, { bitLength, isLe, isNegative }));
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/buffer/toU8a.js
function bufferToU8a(buffer) {
  return new Uint8Array(buffer || []);
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/compact/toU8a.js
var MAX_U8 = BN_TWO.pow(new import_bn.default(8 - 2)).isub(BN_ONE);
var MAX_U16 = BN_TWO.pow(new import_bn.default(16 - 2)).isub(BN_ONE);
var MAX_U32 = BN_TWO.pow(new import_bn.default(32 - 2)).isub(BN_ONE);
var BL_16 = { bitLength: 16 };
var BL_32 = { bitLength: 32 };
function compactToU8a(value) {
  const bn = bnToBn(value);
  if (bn.lte(MAX_U8)) {
    return new Uint8Array([bn.toNumber() << 2]);
  } else if (bn.lte(MAX_U16)) {
    return bnToU8a(bn.shln(2).iadd(BN_ONE), BL_16);
  } else if (bn.lte(MAX_U32)) {
    return bnToU8a(bn.shln(2).iadd(BN_TWO), BL_32);
  }
  const u8a = bnToU8a(bn);
  let length = u8a.length;
  while (u8a[length - 1] === 0) {
    length--;
  }
  if (length < 4) {
    throw new Error("Invalid length, previous checks match anything less than 2^30");
  }
  return u8aConcatStrict([
    // subtract 4 as minimum (also catered for in decoding)
    new Uint8Array([(length - 4 << 2) + 3]),
    u8a.subarray(0, length)
  ]);
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/compact/addLength.js
function compactAddLength(input) {
  return u8aConcatStrict([
    compactToU8a(input.length),
    input
  ]);
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/compact/fromU8a.js
function compactFromU8a(input) {
  const u8a = u8aToU8a(input);
  switch (u8a[0] & 3) {
    case 0:
      return [1, new import_bn.default(u8a[0] >>> 2)];
    case 1:
      return [2, new import_bn.default(u8a[0] + (u8a[1] << 8) >>> 2)];
    case 2:
      return [4, new import_bn.default(u8a[0] + (u8a[1] << 8) + (u8a[2] << 16) + u8a[3] * 16777216 >>> 2)];
    default: {
      const offset = (u8a[0] >>> 2) + 5;
      switch (offset) {
        case 5:
          return [5, new import_bn.default(u8a[1] + (u8a[2] << 8) + (u8a[3] << 16) + u8a[4] * 16777216)];
        case 6:
          return [6, new import_bn.default(u8a[1] + (u8a[2] << 8) + (u8a[3] << 16) + (u8a[4] + (u8a[5] << 8)) * 16777216)];
        case 7:
          return [7, new import_bn.default(u8a[1] + (u8a[2] << 8) + (u8a[3] << 16) + (u8a[4] + (u8a[5] << 8) + (u8a[6] << 16)) * 16777216)];
        default:
          return [offset, u8aToBn(u8a.subarray(1, offset))];
      }
    }
  }
}
function compactFromU8aLim(u8a) {
  switch (u8a[0] & 3) {
    case 0:
      return [1, u8a[0] >>> 2];
    case 1:
      return [2, u8a[0] + (u8a[1] << 8) >>> 2];
    case 2:
      return [4, u8a[0] + (u8a[1] << 8) + (u8a[2] << 16) + u8a[3] * 16777216 >>> 2];
    default: {
      switch ((u8a[0] >>> 2) + 5) {
        case 5:
          return [5, u8a[1] + (u8a[2] << 8) + (u8a[3] << 16) + u8a[4] * 16777216];
        case 6:
          return [6, u8a[1] + (u8a[2] << 8) + (u8a[3] << 16) + (u8a[4] + (u8a[5] << 8)) * 16777216];
        case 7:
          return [7, u8a[1] + (u8a[2] << 8) + (u8a[3] << 16) + (u8a[4] + (u8a[5] << 8) + (u8a[6] << 16)) * 16777216];
        default:
          throw new Error("Compact input is > Number.MAX_SAFE_INTEGER");
      }
    }
  }
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/compact/stripLength.js
function compactStripLength(input) {
  const [offset, length] = compactFromU8a(input);
  const total = offset + length.toNumber();
  return [
    total,
    input.subarray(offset, total)
  ];
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/extractTime.js
var MIN_MS = 60 * 1e3;
var HR_MS = MIN_MS * 60;
var DAY_MS = HR_MS * 24;
var ZERO = { days: 0, hours: 0, milliseconds: 0, minutes: 0, seconds: 0 };
function add(a, b) {
  return {
    days: (a.days || 0) + b.days,
    hours: (a.hours || 0) + b.hours,
    milliseconds: (a.milliseconds || 0) + b.milliseconds,
    minutes: (a.minutes || 0) + b.minutes,
    seconds: (a.seconds || 0) + b.seconds
  };
}
function extractSecs(ms) {
  const s = ms / 1e3;
  if (s < 60) {
    const seconds = ~~s;
    return add({ seconds }, extractTime(ms - seconds * 1e3));
  }
  const m = s / 60;
  if (m < 60) {
    const minutes = ~~m;
    return add({ minutes }, extractTime(ms - minutes * MIN_MS));
  }
  const h = m / 60;
  if (h < 24) {
    const hours = ~~h;
    return add({ hours }, extractTime(ms - hours * HR_MS));
  }
  const days = ~~(h / 24);
  return add({ days }, extractTime(ms - days * DAY_MS));
}
function extractTime(milliseconds) {
  return !milliseconds ? ZERO : milliseconds < 1e3 ? add({ milliseconds }, ZERO) : extractSecs(milliseconds);
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/float/toU8a.js
function floatToU8a(value = 0, { bitLength = 32, isLe = true } = {}) {
  if (bitLength !== 32 && bitLength !== 64) {
    throw new Error("Invalid bitLength provided, expected 32 or 64");
  }
  const result = new Uint8Array(bitLength / 8);
  const dv = new DataView(result.buffer, result.byteOffset);
  if (bitLength === 32) {
    dv.setFloat32(0, Number(value), isLe);
  } else {
    dv.setFloat64(0, Number(value), isLe);
  }
  return result;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/boolean.js
function isBoolean(value) {
  return typeof value === "boolean";
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/format/formatDecimal.js
var NUMBER_REGEX = new RegExp("(\\d+?)(?=(\\d{3})+(?!\\d)|$)", "g");
function formatDecimal(value, separator = ",") {
  const isNegative = value[0].startsWith("-");
  const matched = isNegative ? value.substring(1).match(NUMBER_REGEX) : value.match(NUMBER_REGEX);
  return matched ? `${isNegative ? "-" : ""}${matched.join(separator)}` : value;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/format/getSeparator.js
function getSeparator(locale) {
  return {
    decimal: 0.1.toLocaleString(locale).substring(1, 2),
    thousand: 1e3.toLocaleString(locale).substring(1, 2)
  };
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/format/si.js
var SI_MID = 8;
var SI = [
  { power: -24, text: "yocto", value: "y" },
  { power: -21, text: "zepto", value: "z" },
  { power: -18, text: "atto", value: "a" },
  { power: -15, text: "femto", value: "f" },
  { power: -12, text: "pico", value: "p" },
  { power: -9, text: "nano", value: "n" },
  { power: -6, text: "micro", value: "\xB5" },
  { power: -3, text: "milli", value: "m" },
  { power: 0, text: "Unit", value: "-" },
  { power: 3, text: "Kilo", value: "k" },
  { power: 6, text: "Mill", value: "M" },
  { power: 9, text: "Bill", value: "B" },
  { power: 12, text: "Tril", value: "T" },
  { power: 15, text: "Peta", value: "P" },
  { power: 18, text: "Exa", value: "E" },
  { power: 21, text: "Zeta", value: "Z" },
  { power: 24, text: "Yotta", value: "Y" }
];
function findSi(type) {
  for (let i = 0, count = SI.length; i < count; i++) {
    if (SI[i].value === type) {
      return SI[i];
    }
  }
  return SI[SI_MID];
}
function calcSi(text, decimals, forceUnit) {
  if (forceUnit) {
    return findSi(forceUnit);
  }
  const siDefIndex = SI_MID - 1 + Math.ceil((text.length - decimals) / 3);
  return SI[siDefIndex] || SI[siDefIndex < 0 ? 0 : SI.length - 1];
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/format/formatBalance.js
var DEFAULT_DECIMALS = 0;
var DEFAULT_UNIT = SI[SI_MID].text;
var defaultDecimals = DEFAULT_DECIMALS;
var defaultUnit = DEFAULT_UNIT;
function _formatBalance(input, { decimals = defaultDecimals, forceUnit, locale = "en", withAll = false, withSi = true, withSiFull = false, withUnit = true, withZero = true } = {}) {
  let text = bnToBn(input).toString();
  if (text.length === 0 || text === "0") {
    return "0";
  }
  let sign = "";
  if (text[0].startsWith("-")) {
    sign = "-";
    text = text.substring(1);
  }
  const si = calcSi(text, decimals, forceUnit);
  const mid = text.length - (decimals + si.power);
  const pre = mid <= 0 ? "0" : text.substring(0, mid);
  let post = text.padStart(mid < 0 ? decimals : 1, "0").substring(mid < 0 ? 0 : mid).padEnd(withAll ? Math.max(decimals, 4) : 4, "0").substring(0, withAll ? Math.max(4, decimals + si.power) : 4);
  if (!withZero) {
    let end = post.length - 1;
    do {
      if (post[end] === "0") {
        end--;
      }
    } while (post[end] === "0");
    post = post.substring(0, end + 1);
  }
  const unit = isBoolean(withUnit) ? SI[SI_MID].text : withUnit;
  const units = withSi || withSiFull ? si.value === "-" ? withUnit ? ` ${unit}` : "" : ` ${withSiFull ? `${si.text}${withUnit ? " " : ""}` : si.value}${withUnit ? unit : ""}` : "";
  const { decimal, thousand } = getSeparator(locale);
  return `${sign}${formatDecimal(pre, thousand)}${post && `${decimal}${post}`}${units}`;
}
var formatBalance = _formatBalance;
formatBalance.calcSi = (text, decimals = defaultDecimals) => calcSi(text, decimals);
formatBalance.findSi = findSi;
formatBalance.getDefaults = () => {
  return {
    decimals: defaultDecimals,
    unit: defaultUnit
  };
};
formatBalance.getOptions = (decimals = defaultDecimals) => {
  return SI.filter(({ power }) => power < 0 ? decimals + power >= 0 : true);
};
formatBalance.setDefaults = ({ decimals, unit }) => {
  defaultDecimals = (Array.isArray(decimals) ? decimals[0] : decimals) ?? defaultDecimals;
  defaultUnit = (Array.isArray(unit) ? unit[0] : unit) ?? defaultUnit;
  SI[SI_MID].text = defaultUnit;
};

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/format/formatDate.js
function zeroPad(value) {
  return value.toString().padStart(2, "0");
}
function formatDate(date) {
  const year = date.getFullYear().toString();
  const month = zeroPad(date.getMonth() + 1);
  const day = zeroPad(date.getDate());
  const hour = zeroPad(date.getHours());
  const minute = zeroPad(date.getMinutes());
  const second = zeroPad(date.getSeconds());
  return `${year}-${month}-${day} ${hour}:${minute}:${second}`;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/format/formatElapsed.js
function formatValue(elapsed) {
  if (elapsed < 15) {
    return `${elapsed.toFixed(1)}s`;
  } else if (elapsed < 60) {
    return `${elapsed | 0}s`;
  } else if (elapsed < 3600) {
    return `${elapsed / 60 | 0}m`;
  }
  return `${elapsed / 3600 | 0}h`;
}
function formatElapsed(now, value) {
  const tsNow = now && now.getTime() || 0;
  const tsValue = value instanceof Date ? value.getTime() : bnToBn(value).toNumber();
  return tsNow && tsValue ? formatValue(Math.max(Math.abs(tsNow - tsValue), 0) / 1e3) : "0.0s";
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/format/formatNumber.js
function formatNumber(value, { locale = "en" } = {}) {
  const { thousand } = getSeparator(locale);
  return formatDecimal(bnToBn(value).toString(), thousand);
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/hex/hasPrefix.js
function hexHasPrefix(value) {
  return !!value && isHex(value, -1);
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/hex/addPrefix.js
function hexAddPrefix(value) {
  return value && hexHasPrefix(value) ? value : `0x${value && value.length % 2 === 1 ? "0" : ""}${value || ""}`;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/hex/fixLength.js
function hexFixLength(value, bitLength = -1, withPadding = false) {
  const strLength = Math.ceil(bitLength / 4);
  const hexLength = strLength + 2;
  return hexAddPrefix(bitLength === -1 || value.length === hexLength || !withPadding && value.length < hexLength ? hexStripPrefix(value) : value.length > hexLength ? hexStripPrefix(value).slice(-1 * strLength) : `${"0".repeat(strLength)}${hexStripPrefix(value)}`.slice(-1 * strLength));
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/hex/toNumber.js
function hexToNumber(value) {
  return value ? hexToBn(value).toNumber() : NaN;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/hex/toString.js
function hexToString(_value) {
  return u8aToString(hexToU8a(_value));
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/array.js
function isArray(value) {
  return Array.isArray(value);
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/string.js
function isString(value) {
  return typeof value === "string" || value instanceof String;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/ascii.js
function isAsciiStr(str) {
  for (let i = 0, count = str.length; i < count; i++) {
    const b = str.charCodeAt(i);
    if (b < 32 || b > 126) {
      return false;
    }
  }
  return true;
}
function isAsciiBytes(u8a) {
  for (let i = 0, count = u8a.length; i < count; i++) {
    const b = u8a[i] | 0;
    if (b < 32 || b > 126) {
      return false;
    }
  }
  return true;
}
function isAscii(value) {
  return isString(value) ? isHex(value) ? isAsciiBytes(u8aToU8a(value)) : isAsciiStr(value) : value ? isAsciiBytes(value) : false;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/class.js
var isClass = /* @__PURE__ */ isOnFunction("isPrototypeOf", "hasOwnProperty");

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/childClass.js
function isChildClass(Parent, Child) {
  return isClass(Child) && isClass(Parent) ? Parent === Child || Parent.isPrototypeOf(Child) : false;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/codec.js
var checkCodec = /* @__PURE__ */ isOnObject("toHex", "toHuman", "toU8a");
var checkRegistry = /* @__PURE__ */ isOnObject("get");
function isCodec(value) {
  return checkCodec(value) && checkRegistry(value.registry);
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/compact.js
var isCompact = /* @__PURE__ */ isOnObject("toBigInt", "toBn", "toNumber", "unwrap");

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/error.js
function isError(value) {
  return (value && value.constructor) === Error || value instanceof Error;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/instanceOf.js
function isInstanceOf(value, Clazz) {
  return (value && value.constructor) === Clazz || value instanceof Clazz;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/ip.js
var v4 = "(?:25[0-5]|2[0-4]\\d|1\\d\\d|[1-9]\\d|\\d)(?:\\.(?:25[0-5]|2[0-4]\\d|1\\d\\d|[1-9]\\d|\\d)){3}";
var v6s = "[a-fA-F\\d]{1,4}";
var v6 = `
(?:
(?:${v6s}:){7}(?:${v6s}|:)|                                    // 1:2:3:4:5:6:7::  1:2:3:4:5:6:7:8
(?:${v6s}:){6}(?:${v4}|:${v6s}|:)|                             // 1:2:3:4:5:6::    1:2:3:4:5:6::8   1:2:3:4:5:6::8  1:2:3:4:5:6::1.2.3.4
(?:${v6s}:){5}(?::${v4}|(?::${v6s}){1,2}|:)|                   // 1:2:3:4:5::      1:2:3:4:5::7:8   1:2:3:4:5::8    1:2:3:4:5::7:1.2.3.4
(?:${v6s}:){4}(?:(?::${v6s}){0,1}:${v4}|(?::${v6s}){1,3}|:)| // 1:2:3:4::        1:2:3:4::6:7:8   1:2:3:4::8      1:2:3:4::6:7:1.2.3.4
(?:${v6s}:){3}(?:(?::${v6s}){0,2}:${v4}|(?::${v6s}){1,4}|:)| // 1:2:3::          1:2:3::5:6:7:8   1:2:3::8        1:2:3::5:6:7:1.2.3.4
(?:${v6s}:){2}(?:(?::${v6s}){0,3}:${v4}|(?::${v6s}){1,5}|:)| // 1:2::            1:2::4:5:6:7:8   1:2::8          1:2::4:5:6:7:1.2.3.4
(?:${v6s}:){1}(?:(?::${v6s}){0,4}:${v4}|(?::${v6s}){1,6}|:)| // 1::              1::3:4:5:6:7:8   1::8            1::3:4:5:6:7:1.2.3.4
(?::(?:(?::${v6s}){0,5}:${v4}|(?::${v6s}){1,7}|:))             // ::2:3:4:5:6:7:8  ::2:3:4:5:6:7:8  ::8             ::1.2.3.4
)(?:%[0-9a-zA-Z]{1,})?                                             // %eth0            %1
`.replace(/\s*\/\/.*$/gm, "").replace(/\n/g, "").trim();
var v46Exact = new RegExp(`(?:^${v4}$)|(?:^${v6}$)`);
var v4exact = new RegExp(`^${v4}$`);
var v6exact = new RegExp(`^${v6}$`);
function isIp(value, type) {
  switch (type) {
    case "v4":
      return v4exact.test(value);
    case "v6":
      return v6exact.test(value);
    default:
      return v46Exact.test(value);
  }
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/stringify.js
function replacer(_, v) {
  return isBigInt(v) ? v.toString() : v;
}
function stringify(value, space) {
  return JSON.stringify(value, replacer, space);
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/jsonObject.js
function isJsonObject(value) {
  const str = typeof value !== "string" ? stringify(value) : value;
  try {
    const obj = JSON.parse(str);
    return typeof obj === "object" && obj !== null;
  } catch {
    return false;
  }
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/null.js
function isNull(value) {
  return value === null;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/number.js
function isNumber(value) {
  return typeof value === "number";
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/observable.js
var isObservable = /* @__PURE__ */ isOn("next");

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/promise.js
var isPromise = /* @__PURE__ */ isOnObject("catch", "then");

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/testChain.js
var REGEX_DEV = /(Development|Local Testnet)$/;
function isTestChain(chain) {
  if (!chain) {
    return false;
  }
  return !!REGEX_DEV.test(chain.toString());
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/undefined.js
function isUndefined(value) {
  return value === void 0;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/utf8.js
function isUtf8(value) {
  if (!value) {
    return isString(value);
  }
  const u8a = u8aToU8a(value);
  const len = u8a.length;
  let i = 0;
  while (i < len) {
    if (u8a[i] <= 127) {
      i += 1;
    } else if (u8a[i] >= 194 && u8a[i] <= 223) {
      if (i + 1 < len) {
        if (u8a[i + 1] < 128 || u8a[i + 1] > 191) {
          return false;
        }
      } else {
        return false;
      }
      i += 2;
    } else if (u8a[i] === 224) {
      if (i + 2 < len) {
        if (u8a[i + 1] < 160 || u8a[i + 1] > 191) {
          return false;
        }
        if (u8a[i + 2] < 128 || u8a[i + 2] > 191) {
          return false;
        }
      } else {
        return false;
      }
      i += 3;
    } else if (u8a[i] >= 225 && u8a[i] <= 236) {
      if (i + 2 < len) {
        if (u8a[i + 1] < 128 || u8a[i + 1] > 191) {
          return false;
        }
        if (u8a[i + 2] < 128 || u8a[i + 2] > 191) {
          return false;
        }
      } else {
        return false;
      }
      i += 3;
    } else if (u8a[i] === 237) {
      if (i + 2 < len) {
        if (u8a[i + 1] < 128 || u8a[i + 1] > 159) {
          return false;
        }
        if (u8a[i + 2] < 128 || u8a[i + 2] > 191) {
          return false;
        }
      } else {
        return false;
      }
      i += 3;
    } else if (u8a[i] >= 238 && u8a[i] <= 239) {
      if (i + 2 < len) {
        if (u8a[i + 1] < 128 || u8a[i + 1] > 191) {
          return false;
        }
        if (u8a[i + 2] < 128 || u8a[i + 2] > 191) {
          return false;
        }
      } else {
        return false;
      }
      i += 3;
    } else if (u8a[i] === 240) {
      if (i + 3 < len) {
        if (u8a[i + 1] < 144 || u8a[i + 1] > 191) {
          return false;
        }
        if (u8a[i + 2] < 128 || u8a[i + 2] > 191) {
          return false;
        }
        if (u8a[i + 3] < 128 || u8a[i + 3] > 191) {
          return false;
        }
      } else {
        return false;
      }
      i += 4;
    } else if (u8a[i] >= 241 && u8a[i] <= 243) {
      if (i + 3 < len) {
        if (u8a[i + 1] < 128 || u8a[i + 1] > 191) {
          return false;
        }
        if (u8a[i + 2] < 128 || u8a[i + 2] > 191) {
          return false;
        }
        if (u8a[i + 3] < 128 || u8a[i + 3] > 191) {
          return false;
        }
      } else {
        return false;
      }
      i += 4;
    } else if (u8a[i] === 244) {
      if (i + 3 < len) {
        if (u8a[i + 1] < 128 || u8a[i + 1] > 143) {
          return false;
        }
        if (u8a[i + 2] < 128 || u8a[i + 2] > 191) {
          return false;
        }
        if (u8a[i + 3] < 128 || u8a[i + 3] > 191) {
          return false;
        }
      } else {
        return false;
      }
      i += 4;
    } else {
      return false;
    }
  }
  return true;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/is/wasm.js
var WASM_MAGIC = new Uint8Array([0, 97, 115, 109]);
function isWasm(value) {
  return isU8a(value) && u8aEq(value.subarray(0, 4), WASM_MAGIC);
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/lazy.js
function lazyMethod(result, item, creator, getName, index = 0) {
  const name = getName ? getName(item, index) : item.toString();
  let value;
  Object.defineProperty(result, name, {
    // This allows for re-configuration with the embedded defineProperty below
    // and ensures that on tested browsers and Node, it _will_ be redefined
    // and thus short-circuited for future access
    configurable: true,
    enumerable: true,
    // Use a function here, we don't want to capture the outer this, i.e.
    // don't use arrow functions in this context since we have a this inside
    get: function() {
      if (value === void 0) {
        value = creator(item, index, this);
        try {
          Object.defineProperty(this, name, { value });
        } catch {
        }
      }
      return value;
    }
  });
}
function lazyMethods(result, items, creator, getName) {
  for (let i = 0, count = items.length; i < count; i++) {
    lazyMethod(result, items[i], creator, getName, i);
  }
  return result;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/noop.js
function identity(value) {
  return value;
}
function noop() {
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/logger.js
var logTo = {
  debug: "log",
  error: "error",
  log: "log",
  warn: "warn"
};
function formatOther(value) {
  if (value && isObject(value) && value.constructor === Object) {
    const result = {};
    for (const [k, v] of Object.entries(value)) {
      result[k] = loggerFormat(v);
    }
    return result;
  }
  return value;
}
function loggerFormat(value) {
  if (Array.isArray(value)) {
    return value.map(loggerFormat);
  } else if (isBn(value)) {
    return value.toString();
  } else if (isU8a(value) || isBuffer(value)) {
    return u8aToHex(u8aToU8a(value));
  }
  return formatOther(value);
}
function formatWithLength(maxLength) {
  return (v) => {
    if (maxLength <= 0) {
      return v;
    }
    const r = `${v}`;
    return r.length < maxLength ? v : `${r.substring(0, maxLength)} ...`;
  };
}
function apply(log, type, values, maxSize = -1) {
  if (values.length === 1 && isFunction(values[0])) {
    const fnResult = values[0]();
    return apply(log, type, Array.isArray(fnResult) ? fnResult : [fnResult], maxSize);
  }
  console[logTo[log]](formatDate(/* @__PURE__ */ new Date()), type, ...values.map(loggerFormat).map(formatWithLength(maxSize)));
}
function isDebugOn(e, type) {
  return !!e && (e === "*" || type === e || e.endsWith("*") && type.startsWith(e.slice(0, -1)));
}
function isDebugOff(e, type) {
  return !!e && (e.startsWith("-") && (type === e.slice(1) || e.endsWith("*") && type.startsWith(e.slice(1, -1))));
}
function getDebugFlag(env, type) {
  let flag = false;
  for (const e of env) {
    if (isDebugOn(e, type)) {
      flag = true;
    } else if (isDebugOff(e, type)) {
      flag = false;
    }
  }
  return flag;
}
function parseEnv(type) {
  const env = (hasProcess ? xglobal.process : {}).env || {};
  const maxSize = parseInt(env["DEBUG_MAX"] || "-1", 10);
  return [
    getDebugFlag((env["DEBUG"] || "").toLowerCase().split(","), type),
    isNaN(maxSize) ? -1 : maxSize
  ];
}
function logger(origin) {
  const type = `${origin.toUpperCase()}:`.padStart(16);
  const [isDebug, maxSize] = parseEnv(origin.toLowerCase());
  return {
    debug: isDebug ? (...values) => apply("debug", type, values, maxSize) : noop,
    error: (...values) => apply("error", type, values),
    log: (...values) => apply("log", type, values),
    noop,
    warn: (...values) => apply("warn", type, values)
  };
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/memoize.js
function defaultGetId() {
  return "none";
}
function memoize(fn, { getInstanceId = defaultGetId } = {}) {
  const cache = {};
  const memoized = (...args) => {
    const stringParams = stringify(args);
    const instanceId = getInstanceId();
    if (!cache[instanceId]) {
      cache[instanceId] = {};
    }
    if (cache[instanceId][stringParams] === void 0) {
      cache[instanceId][stringParams] = fn(...args);
    }
    return cache[instanceId][stringParams];
  };
  memoized.unmemoize = (...args) => {
    const stringParams = stringify(args);
    const instanceId = getInstanceId();
    if (cache[instanceId] && cache[instanceId][stringParams] !== void 0) {
      delete cache[instanceId][stringParams];
    }
  };
  return memoized;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/nextTick.js
function nextTick(onExec, onError) {
  setTimeout(() => {
    Promise.resolve().then(() => {
      onExec();
    }).catch((error) => {
      if (onError) {
        onError(error);
      } else {
        console.error(error);
      }
    });
  }, 0);
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/number/toHex.js
function numberToHex(value, bitLength = -1) {
  const hex2 = (!value || Number.isNaN(value) ? 0 : value).toString(16);
  return hexFixLength(hex2.length % 2 ? `0${hex2}` : hex2, bitLength, true);
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/number/toU8a.js
function numberToU8a(value, bitLength = -1) {
  return hexToU8a(numberToHex(value, bitLength));
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/object/clear.js
function objectClear(value) {
  const keys = Object.keys(value);
  for (let i = 0, count = keys.length; i < count; i++) {
    delete value[keys[i]];
  }
  return value;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/object/spread.js
function objectSpread(dest, ...sources) {
  for (let i = 0, count = sources.length; i < count; i++) {
    const src = sources[i];
    if (src) {
      if (typeof src.entries === "function") {
        for (const [key, value] of src.entries()) {
          dest[key] = value;
        }
      } else {
        Object.assign(dest, src);
      }
    }
  }
  return dest;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/object/copy.js
function objectCopy(source) {
  return objectSpread({}, source);
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/object/entries.js
function objectEntries(obj) {
  return Object.entries(obj);
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/object/keys.js
function objectKeys(value) {
  return Object.keys(value);
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/object/property.js
function objectProperty(that, key, getter, getName, index = 0) {
  const name = getName ? getName(key, index) : key;
  if (!(name in that)) {
    Object.defineProperty(that, name, {
      enumerable: true,
      // Unlike in lazy, we always call into the upper function, i.e. this method
      // does not cache old values (it is expected to be used for dynamic values)
      get: function() {
        return getter(key, index, this);
      }
    });
  }
}
function objectProperties(that, keys, getter, getName) {
  for (let i = 0, count = keys.length; i < count; i++) {
    objectProperty(that, keys[i], getter, getName, i);
  }
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/object/values.js
function objectValues(obj) {
  return Object.values(obj);
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/promisify.js
function promisify(self2, fn, ...params) {
  return new Promise((resolve, reject) => {
    fn.apply(self2, params.concat((error, result) => {
      if (error) {
        reject(error);
      } else {
        resolve(result);
      }
    }));
  });
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/string/camelCase.js
var CC_TO_UP = new Array(256);
var CC_TO_LO = new Array(256);
for (let i = 0, count = CC_TO_UP.length; i < count; i++) {
  CC_TO_LO[i] = String.fromCharCode(i).toLowerCase();
  CC_TO_UP[i] = String.fromCharCode(i).toUpperCase();
}
function formatAllCaps(w) {
  return w.slice(0, w.length - 1).toLowerCase() + CC_TO_UP[w.charCodeAt(w.length - 1)];
}
function converter(format) {
  return (value) => {
    const parts = value.replace(/[-_., ]+/g, " ").trim().split(" ");
    let result = "";
    for (let i = 0, count = parts.length; i < count; i++) {
      const w = parts[i];
      result += format(/^[\dA-Z]+$/.test(w) ? w.toLowerCase() : w.replace(/^[\dA-Z]{2,}[^a-z]/, formatAllCaps), i);
    }
    return result;
  };
}
var stringCamelCase = /* @__PURE__ */ converter((w, i) => (i ? CC_TO_UP[w.charCodeAt(0)] : CC_TO_LO[w.charCodeAt(0)]) + w.slice(1));
var stringPascalCase = /* @__PURE__ */ converter((w) => CC_TO_UP[w.charCodeAt(0)] + w.slice(1));

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/string/lowerFirst.js
function converter2(map) {
  return (value) => value ? map[value.charCodeAt(0)] + value.slice(1) : "";
}
var stringLowerFirst = /* @__PURE__ */ converter2(CC_TO_LO);
var stringUpperFirst = /* @__PURE__ */ converter2(CC_TO_UP);

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/string/shorten.js
function stringShorten(value, prefixLength = 6) {
  return value.length <= 2 + 2 * prefixLength ? value.toString() : `${value.substring(0, prefixLength)}\u2026${value.slice(-prefixLength)}`;
}

// ../../node_modules/.pnpm/@polkadot+util@12.3.2/node_modules/@polkadot/util/string/toHex.js
function stringToHex(value) {
  return u8aToHex(stringToU8a(value));
}

// src/utils/contracts/toBasicMetadata.ts
var toBasicMetadata = async (file) => {
  return await new Promise((resolve, reject) => {
    const reader = new FileReader();
    reader.onload = ({ target }) => {
      if (target?.result) {
        const name = file.name;
        const data = new Uint8Array(target.result);
        const size = data.length;
        resolve({ data, name, size });
      }
      reject("Target result not found in file");
    };
    reader.readAsArrayBuffer(file);
  });
};

// src/utils/contracts/toJSType.ts
var toJSType = (abiParam) => {
  const map = {
    AccountIndex: "BN",
    Balance: "BN",
    BalanceOf: "BN",
    Compact: "BN",
    Gas: "BN",
    Index: "BN",
    Nonce: "BN",
    ParaId: "BN",
    PropIndex: "BN",
    ProposalIndex: "BN",
    ReferendumIndex: "BN",
    i64: "BN",
    i128: "BN",
    u32: "BN",
    u64: "BN",
    u128: "BN",
    VoteIndex: "BN",
    Moment: "BN",
    VoteThreshold: "BN",
    Vote: "BN",
    BlockNumber: "number",
    u8: "number",
    u16: "number",
    i8: "number",
    i16: "number",
    i32: "number",
    bool: "boolean",
    String: "string",
    Text: "string",
    Hash: "string",
    H256: "string",
    H512: "string",
    H160: "string",
    BlockHash: "string",
    CodeHash: "string",
    AccountId: "string",
    AccountIdOf: "string",
    Address: "string",
    Call: "string",
    CandidateReceipt: "string",
    Digest: "string",
    Header: "string",
    KeyValue: "string",
    LookupSource: "string",
    MisbehaviorReport: "string",
    Proposal: "string",
    Signature: "string",
    SessionKey: "string",
    StorageKey: "string",
    ValidatorId: "string",
    Raw: "string",
    Keys: "string",
    Null: "null",
    Bytes: "Bytes",
    Extrinsic: "Bytes"
  };
  return [abiParam.type.type, map[abiParam.type.type]];
};

// src/utils/contracts/toMessageParams.ts
var toMessageParams = (api, abiParams, userParams) => {
  return abiParams.map(({ name, type: { type } }) => {
    const value = userParams ? userParams[name] : null;
    if (type === "Balance")
      return api.registry.createType("Balance", value);
    return value || null;
  });
};

// src/utils/contracts/txUtils.ts
var isNone = (tx) => tx.status === "None";
var isDryRun = (tx) => tx.status === "DryRun";
var isPendingSignature = (tx) => tx.status === "PendingSignature";
var isErrored = (tx) => tx.status === "Errored";
var isFuture = (tx) => tx.status === "Future";
var isReady = (tx) => tx.status === "Ready";
var isBroadcast = (tx) => tx.status === "Broadcast";
var isRetracted = (tx) => tx.status === "Retracted";
var isFinalityTimeout = (tx) => tx.status === "FinalityTimeout";
var isInBlock = (tx) => tx.status === "InBlock";
var isFinalized = (tx) => tx.status === "Finalized";
var isUsurped = (tx) => tx.status === "Usurped";
var isDropped = (tx) => tx.status === "Dropped";
var isInvalid = (tx) => tx.status === "Invalid";
var hasAny = (tx, ...statuses) => statuses.includes(tx.status);
var shouldDisable = (tx) => hasAny(tx, "DryRun", "PendingSignature", "Broadcast");
var shouldDisableStrict = (tx) => shouldDisable(tx) || isInBlock(tx);

// src/utils/contracts/validateMetadata.ts
var MetadataError = /* @__PURE__ */ ((MetadataError2) => {
  MetadataError2["InvalidFile"] = "Invalid file.";
  MetadataError2["EmptyWasm"] = "Wasm field not found.";
  MetadataError2["InvalidWasm"] = "Invalid Wasm field.";
  return MetadataError2;
})(MetadataError || {});
var validateMetadata = (metadata, { requireWasm }) => {
  if (!metadata)
    return { error: "Invalid file." /* InvalidFile */ };
  const wasm = metadata.info.source.wasm;
  const isWasmEmpty = wasm.isEmpty;
  const isWasmInvalid = !WebAssembly.validate(wasm);
  if (requireWasm && isWasmEmpty) {
    return { error: "Wasm field not found." /* EmptyWasm */ };
  }
  if (requireWasm && isWasmInvalid) {
    return { error: "Invalid Wasm field." /* InvalidWasm */ };
  }
  return {};
};

// src/utils/events/formatEventName.ts
var formatEventName = ({ event }) => `${event.section}:${event.method}`;

// src/utils/events/contracts/ContractInstantiated.ts
var isContractInstantiatedEvent = ({ event }) => event.section === "contracts" && event.method === "Instantiated";
var asContractInstantiatedEvent = (eventRecord) => {
  if (!isContractInstantiatedEvent(eventRecord))
    return;
  const deployer = eventRecord.event.data[0]?.toString();
  const contractAddress = eventRecord.event.data[1]?.toString();
  if (!deployer || !contractAddress)
    return;
  return { deployer, contractAddress, name: formatEventName(eventRecord) };
};

// src/core/contracts/getRegistryError.ts
var getRegistryError = (error, { contract: { api } }) => {
  if (!error?.isModule)
    return;
  return api?.registry.findMetaError(error.asModule);
};

// src/core/contracts/decodeError.ts
var formatErrorMessage = (registryError) => `${registryError.section}.${registryError.method}: ${registryError.docs}`;
var decodeError = (dispatchError, chainContract, moduleMessages, defaultMessage) => {
  if (!chainContract)
    return void 0;
  const registryError = getRegistryError(dispatchError, chainContract);
  if (!registryError)
    return void 0;
  return moduleMessages?.[registryError.method] || defaultMessage || formatErrorMessage(registryError);
};

// ../../node_modules/.pnpm/@talismn+connect-wallets@1.2.3_@polkadot+api@10.9.1_@polkadot+extension-inject@0.46.5/node_modules/@talismn/connect-wallets/dist/esm/index.js
var import_react = require("react");
function __awaiter(thisArg, _arguments, P, generator) {
  function adopt(value) {
    return value instanceof P ? value : new P(function(resolve) {
      resolve(value);
    });
  }
  return new (P || (P = Promise))(function(resolve, reject) {
    function fulfilled(value) {
      try {
        step(generator.next(value));
      } catch (e) {
        reject(e);
      }
    }
    function rejected(value) {
      try {
        step(generator["throw"](value));
      } catch (e) {
        reject(e);
      }
    }
    function step(result) {
      result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected);
    }
    step((generator = generator.apply(thisArg, _arguments || [])).next());
  });
}
var BaseWalletError = class extends Error {
  constructor(message, wallet) {
    super(message);
    this.name = "WalletError";
    Object.setPrototypeOf(this, new.target.prototype);
    if (Error.captureStackTrace) {
      Error.captureStackTrace(this, BaseWalletError);
    }
    this.wallet = wallet;
  }
};
var AuthError = class extends BaseWalletError {
  constructor() {
    super(...arguments);
    this.name = "AuthError";
  }
};
var NotInstalledError = class extends BaseWalletError {
  constructor() {
    super(...arguments);
    this.name = "NotInstalledError";
  }
};
var BaseDotsamaWallet = class {
  constructor() {
    this.extensionName = "";
    this.title = "";
    this.installUrl = "";
    this.logo = {
      src: "",
      alt: ""
    };
    this.transformError = (err) => {
      if (err.message.includes("pending authorization request")) {
        return new AuthError(err.message, this);
      }
      return err;
    };
    this.enable = (dappName) => __awaiter(this, void 0, void 0, function* () {
      if (!dappName) {
        throw new Error("MissingParamsError: Dapp name is required.");
      }
      if (!this.installed) {
        throw new NotInstalledError(`Refresh the browser if ${this.title} is already installed.`, this);
      }
      try {
        const injectedExtension = this.rawExtension;
        const rawExtension = yield injectedExtension === null || injectedExtension === void 0 ? void 0 : injectedExtension.enable(dappName);
        if (!rawExtension) {
          throw new NotInstalledError(`${this.title} is installed but is not returned by the 'Wallet.enable(dappname)' function`, this);
        }
        const extension = Object.assign(Object.assign({}, rawExtension), { name: this.extensionName, version: injectedExtension.version });
        this._extension = extension;
        this._signer = extension === null || extension === void 0 ? void 0 : extension.signer;
      } catch (err) {
        throw this.transformError(err);
      }
    });
    this.getAccounts = (anyType) => __awaiter(this, void 0, void 0, function* () {
      if (!this._extension) {
        throw new NotInstalledError(`The 'Wallet.enable(dappname)' function should be called first.`, this);
      }
      const accounts = yield this._extension.accounts.get(anyType);
      const accountsWithWallet = accounts.map((account) => {
        var _a, _b;
        return Object.assign(Object.assign({}, account), { source: (_a = this._extension) === null || _a === void 0 ? void 0 : _a.name, wallet: this, signer: (_b = this._extension) === null || _b === void 0 ? void 0 : _b.signer });
      });
      return accountsWithWallet;
    });
    this.subscribeAccounts = (callback) => __awaiter(this, void 0, void 0, function* () {
      if (!this._extension) {
        throw new NotInstalledError(`The 'Wallet.enable(dappname)' function should be called first.`, this);
      }
      const unsubscribe = this._extension.accounts.subscribe((accounts) => {
        const accountsWithWallet = accounts.map((account) => {
          var _a, _b;
          return Object.assign(Object.assign({}, account), { source: (_a = this._extension) === null || _a === void 0 ? void 0 : _a.name, wallet: this, signer: (_b = this._extension) === null || _b === void 0 ? void 0 : _b.signer });
        });
        callback(accountsWithWallet);
      });
      return unsubscribe;
    });
  }
  get extension() {
    return this._extension;
  }
  get signer() {
    return this._signer;
  }
  get installed() {
    var _a;
    const injectedWindow = window;
    const injectedExtension = (_a = injectedWindow === null || injectedWindow === void 0 ? void 0 : injectedWindow.injectedWeb3) === null || _a === void 0 ? void 0 : _a[this.extensionName];
    return !!injectedExtension;
  }
  get rawExtension() {
    var _a;
    const injectedWindow = window;
    const injectedExtension = (_a = injectedWindow === null || injectedWindow === void 0 ? void 0 : injectedWindow.injectedWeb3) === null || _a === void 0 ? void 0 : _a[this.extensionName];
    return injectedExtension;
  }
};
var logo$3 = "data:image/svg+xml,%3Csvg%20width%3D%2224%22%20height%3D%2224%22%20viewBox%3D%220%200%2024%2024%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%3Crect%20width%3D%2224%22%20height%3D%2224%22%20rx%3D%224%22%20fill%3D%22%23D5FF5C%22%2F%3E%3Cpath%20fill-rule%3D%22evenodd%22%20clip-rule%3D%22evenodd%22%20d%3D%22M16.5349%2012.9159C16.6871%2013.2472%2017.135%2013.3643%2017.3929%2013.1065L17.8651%2012.6345C18.3535%2012.1464%2019.1453%2012.1464%2019.6337%2012.6345C20.1221%2013.1227%2020.1221%2013.9141%2019.6337%2014.4023L15.8097%2018.2246C14.8921%2019.3104%2013.5198%2020%2011.9865%2020C10.3851%2020%208.95942%2019.2478%208.04385%2018.0777L4.36629%2014.4018C3.87791%2013.9136%203.87791%2013.1222%204.36629%2012.634C4.85467%2012.1459%205.64649%2012.1459%206.13487%2012.634L6.60044%2013.0994C6.85253%2013.3514%207.29002%2013.238%207.43894%2012.9141V12.9141C7.46838%2012.8501%207.48439%2012.7814%207.48439%2012.711L7.48438%207.00059C7.48438%206.30991%208.04428%205.75001%208.73496%205.75001C9.42563%205.75001%209.98553%206.30991%209.98553%207.00058L9.98553%209.88892C9.98553%2010.1376%2010.2403%2010.3065%2010.4774%2010.2315V10.2315C10.6276%2010.1841%2010.736%2010.0474%2010.736%209.89001L10.736%205.25041C10.736%204.55974%2011.2959%203.99984%2011.9866%203.99984C12.6773%203.99984%2013.2372%204.55974%2013.2372%205.25041L13.2372%209.89018C13.2372%2010.0476%2013.3456%2010.1842%2013.4957%2010.2316V10.2316C13.7327%2010.3065%2013.9874%2010.1377%2013.9874%209.88909L13.9874%207.00059C13.9874%206.30991%2014.5473%205.75001%2015.2379%205.75001C15.9286%205.75001%2016.4885%206.30991%2016.4885%207.00058L16.4885%2012.7086C16.4885%2012.7805%2016.5049%2012.8506%2016.5349%2012.9159V12.9159Z%22%20fill%3D%22%23FD4848%22%2F%3E%3Cpath%20d%3D%22M15.9885%2015.5C15.9885%2015.5%2014.1969%2018%2011.9867%2018C9.77655%2018%207.98486%2015.5%207.98486%2015.5C7.98486%2015.5%209.77655%2013%2011.9867%2013C14.1969%2013%2015.9885%2015.5%2015.9885%2015.5Z%22%20fill%3D%22%23D5FF5C%22%2F%3E%3Cpath%20d%3D%22M13.8543%2015.5C13.8543%2016.5311%2013.018%2017.3671%2011.9863%2017.3671C10.9545%2017.3671%2010.1183%2016.5311%2010.1183%2015.5C10.1183%2014.4689%2010.9545%2013.6329%2011.9863%2013.6329C13.018%2013.6329%2013.8543%2014.4689%2013.8543%2015.5Z%22%20stroke%3D%22%23FD4848%22%20stroke-width%3D%220.265831%22%2F%3E%3Cpath%20d%3D%22M13.1041%2015.5C13.1041%2016.1169%2012.6037%2016.6171%2011.9864%2016.6171C11.3691%2016.6171%2010.8688%2016.1169%2010.8688%2015.5C10.8688%2014.8831%2011.3691%2014.3829%2011.9864%2014.3829C12.6037%2014.3829%2013.1041%2014.8831%2013.1041%2015.5Z%22%20stroke%3D%22%23FD4848%22%20stroke-width%3D%220.265831%22%2F%3E%3Cpath%20d%3D%22M14.605%2015.5C14.605%2016.9453%2013.4327%2018.1171%2011.9866%2018.1171C10.5405%2018.1171%209.36827%2016.9453%209.36827%2015.5C9.36827%2014.0547%2010.5405%2012.8829%2011.9866%2012.8829C13.4327%2012.8829%2014.605%2014.0547%2014.605%2015.5Z%22%20stroke%3D%22%23FD4848%22%20stroke-width%3D%220.265831%22%2F%3E%3Cpath%20d%3D%22M15.3552%2015.5C15.3552%2017.3595%2013.847%2018.8671%2011.9865%2018.8671C10.1259%2018.8671%208.61778%2017.3595%208.61778%2015.5C8.61778%2013.6405%2010.1259%2012.1329%2011.9865%2012.1329C13.847%2012.1329%2015.3552%2013.6405%2015.3552%2015.5Z%22%20stroke%3D%22%23FD4848%22%20stroke-width%3D%220.265831%22%2F%3E%3Cpath%20d%3D%22M12.3534%2015.5C12.3534%2015.7027%2012.1891%2015.8671%2011.9863%2015.8671C11.7836%2015.8671%2011.6192%2015.7027%2011.6192%2015.5C11.6192%2015.2973%2011.7836%2015.1329%2011.9863%2015.1329C12.1891%2015.1329%2012.3534%2015.2973%2012.3534%2015.5Z%22%20fill%3D%22%23162BEB%22%20stroke%3D%22%23FD4848%22%20stroke-width%3D%220.265831%22%2F%3E%3Cellipse%20cx%3D%2211.9863%22%20cy%3D%2215.5%22%20rx%3D%220.5%22%20ry%3D%220.5%22%20fill%3D%22%23FD4848%22%2F%3E%3Cmask%20id%3D%22path-10-inside-1_4684_17034%22%20fill%3D%22white%22%3E%3Cpath%20d%3D%22M15.9885%2015.5C15.9885%2015.5%2014.1969%2018%2011.9867%2018C9.77655%2018%207.98486%2015.5%207.98486%2015.5C7.98486%2015.5%209.77655%2013%2011.9867%2013C14.1969%2013%2015.9885%2015.5%2015.9885%2015.5Z%22%2F%3E%3C%2Fmask%3E%3Cpath%20d%3D%22M15.9885%2015.5C15.9885%2015.5%2014.1969%2018%2011.9867%2018C9.77655%2018%207.98486%2015.5%207.98486%2015.5C7.98486%2015.5%209.77655%2013%2011.9867%2013C14.1969%2013%2015.9885%2015.5%2015.9885%2015.5Z%22%20stroke%3D%22%23D5FF5C%22%20stroke-width%3D%220.531663%22%20mask%3D%22url%28%23path-10-inside-1_4684_17034%29%22%2F%3E%3C%2Fsvg%3E";
var TalismanWallet = class extends BaseDotsamaWallet {
  constructor() {
    super(...arguments);
    this.extensionName = "talisman";
    this.title = "Talisman";
    this.installUrl = "https://talisman.xyz/download";
    this.noExtensionMessage = "You can use any Polkadot compatible wallet but we recommend using Talisman";
    this.logo = {
      src: logo$3,
      alt: "Talisman Logo"
    };
  }
};
var logo$2 = "data:image/svg+xml,%3C%3Fxml%20version%3D%221.0%22%20encoding%3D%22utf-8%22%20standalone%3D%22yes%22%3F%3E%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20xmlns%3Axlink%3D%22http%3A%2F%2Fwww.w3.org%2F1999%2Fxlink%22%20version%3D%221.1%22%20id%3D%22Layer_1%22%20x%3D%220px%22%20y%3D%220px%22%20viewBox%3D%2215%2015%20140%20140%22%20style%3D%22enable-background%3Anew%200%200%20170%20170%3Bzoom%3A%201%3B%22%20xml%3Aspace%3D%22preserve%22%3E%3Cstyle%20type%3D%22text%2Fcss%22%3E.bg0%7Bfill%3A%23FF8C00%7D%20.st0%7Bfill%3A%23FFFFFF%7D%3C%2Fstyle%3E%3Cg%3E%3Ccircle%20class%3D%22bg0%22%20cx%3D%2285%22%20cy%3D%2285%22%20r%3D%2270%22%3E%3C%2Fcircle%3E%3Cg%3E%3Cpath%20class%3D%22st0%22%20d%3D%22M85%2C34.7c-20.8%2C0-37.8%2C16.9-37.8%2C37.8c0%2C4.2%2C0.7%2C8.3%2C2%2C12.3c0.9%2C2.7%2C3.9%2C4.2%2C6.7%2C3.3c2.7-0.9%2C4.2-3.9%2C3.3-6.7%20c-1.1-3.1-1.6-6.4-1.5-9.7C58.1%2C57.6%2C69.5%2C46%2C83.6%2C45.3c15.7-0.8%2C28.7%2C11.7%2C28.7%2C27.2c0%2C14.5-11.4%2C26.4-25.7%2C27.2%20c0%2C0-5.3%2C0.3-7.9%2C0.7c-1.3%2C0.2-2.3%2C0.4-3%2C0.5c-0.3%2C0.1-0.6-0.2-0.5-0.5l0.9-4.4L81%2C73.4c0.6-2.8-1.2-5.6-4-6.2%20c-2.8-0.6-5.6%2C1.2-6.2%2C4c0%2C0-11.8%2C55-11.9%2C55.6c-0.6%2C2.8%2C1.2%2C5.6%2C4%2C6.2c2.8%2C0.6%2C5.6-1.2%2C6.2-4c0.1-0.6%2C1.7-7.9%2C1.7-7.9%20c1.2-5.6%2C5.8-9.7%2C11.2-10.4c1.2-0.2%2C5.9-0.5%2C5.9-0.5c19.5-1.5%2C34.9-17.8%2C34.9-37.7C122.8%2C51.6%2C105.8%2C34.7%2C85%2C34.7z%20M87.7%2C121.7%20c-3.4-0.7-6.8%2C1.4-7.5%2C4.9c-0.7%2C3.4%2C1.4%2C6.8%2C4.9%2C7.5c3.4%2C0.7%2C6.8-1.4%2C7.5-4.9C93.3%2C125.7%2C91.2%2C122.4%2C87.7%2C121.7z%22%3E%3C%2Fpath%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E";
var PolkadotjsWallet = class extends BaseDotsamaWallet {
  constructor() {
    super(...arguments);
    this.extensionName = "polkadot-js";
    this.title = "Polkadot.js";
    this.noExtensionMessage = "You can use any Polkadot compatible wallet but we recommend using Talisman";
    this.installUrl = "https://polkadot.js.org/extension/";
    this.logo = {
      src: logo$2,
      alt: "Polkadotjs Logo"
    };
  }
};
var logo$1 = "data:image/svg+xml,%3Csvg%20width%3D%22134%22%20height%3D%22134%22%20viewBox%3D%220%200%20134%20134%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%3Cmask%20id%3D%22mask0_699_5101%22%20style%3D%22mask-type%3Aalpha%22%20maskUnits%3D%22userSpaceOnUse%22%20x%3D%220%22%20y%3D%220%22%20width%3D%22134%22%20height%3D%22134%22%3E%3Crect%20width%3D%22134%22%20height%3D%22134%22%20fill%3D%22%23C4C4C4%22%2F%3E%3C%2Fmask%3E%3Cg%20mask%3D%22url%28%23mask0_699_5101%29%22%3E%3Cpath%20d%3D%22M87.9615%2064.3201L87.9456%2047.7455L27.1191%2016.2236V64.3041L66.0589%2085.106L80.2884%2078.8367L37.4403%2056.1046L37.4722%2037.887L87.9615%2064.3201Z%22%20fill%3D%22url%28%23paint0_linear_699_5101%29%22%2F%3E%3Cpath%20d%3D%22M50.7607%2044.8421V50.5052L37.3926%2056.2321L37.4883%2037.6636L50.7607%2044.8421Z%22%20fill%3D%22url%28%23paint1_linear_699_5101%29%22%2F%3E%3Cpath%20d%3D%22M50.8095%2091.822L80.2895%2078.8368L37.4414%2056.2163L50.6819%2050.5054L105.765%2079.2835L50.9212%20103.212L50.8095%2091.822Z%22%20fill%3D%22url%28%23paint2_linear_699_5101%29%22%2F%3E%3Cpath%20d%3D%22M37.4886%2087.9773L50.6493%2082.2982L50.9365%20103.196L105.765%2079.2832V97.118L37.377%20127.077L37.4886%2087.9773Z%22%20fill%3D%22url%28%23paint3_linear_699_5101%29%22%2F%3E%3Cpath%20d%3D%22M27.1191%2082.5857L37.4403%2087.9776L37.3765%20127.013L27.1191%20121.86V82.5857Z%22%20fill%3D%22url%28%23paint4_linear_699_5101%29%22%2F%3E%3Cpath%20d%3D%22M40.1522%2076.7791L50.6489%2082.2986L37.4403%2087.9776L27.1191%2082.5857L40.1522%2076.7791Z%22%20fill%3D%22url%28%23paint5_linear_699_5101%29%22%2F%3E%3Cpath%20d%3D%22M105.765%2056.5993L105.702%2039.9131L87.9785%2047.7457V64.3362L105.765%2056.5993Z%22%20fill%3D%22url%28%23paint6_linear_699_5101%29%22%2F%3E%3Cpath%20d%3D%22M27.1191%2016.2237L45.0337%207.97632L105.732%2039.8811L87.9775%2047.7456L27.1191%2016.2237Z%22%20fill%3D%22url%28%23paint7_linear_699_5101%29%22%2F%3E%3C%2Fg%3E%3Cdefs%3E%3ClinearGradient%20id%3D%22paint0_linear_699_5101%22%20x1%3D%2211.9006%22%20y1%3D%2250.6648%22%20x2%3D%22119.372%22%20y2%3D%2250.6648%22%20gradientUnits%3D%22userSpaceOnUse%22%3E%3Cstop%20stop-color%3D%22%23FFD4B2%22%2F%3E%3Cstop%20offset%3D%220.36%22%20stop-color%3D%22%239ACEB7%22%2F%3E%3Cstop%20offset%3D%220.67%22%20stop-color%3D%22%2347C8BB%22%2F%3E%3Cstop%20offset%3D%220.89%22%20stop-color%3D%22%2314C5BE%22%2F%3E%3Cstop%20offset%3D%221%22%20stop-color%3D%22%2300C4BF%22%2F%3E%3C%2FlinearGradient%3E%3ClinearGradient%20id%3D%22paint1_linear_699_5101%22%20x1%3D%2244.0766%22%20y1%3D%2262.8524%22%20x2%3D%2244.0766%22%20y2%3D%2221.2167%22%20gradientUnits%3D%22userSpaceOnUse%22%3E%3Cstop%20stop-color%3D%22%2300FECF%22%2F%3E%3Cstop%20offset%3D%220.08%22%20stop-color%3D%22%2300E5D0%22%2F%3E%3Cstop%20offset%3D%220.24%22%20stop-color%3D%22%2300A5D1%22%2F%3E%3Cstop%20offset%3D%220.48%22%20stop-color%3D%22%230040D4%22%2F%3E%3Cstop%20offset%3D%220.54%22%20stop-color%3D%22%230025D5%22%2F%3E%3Cstop%20offset%3D%221%22%2F%3E%3C%2FlinearGradient%3E%3ClinearGradient%20id%3D%22paint2_linear_699_5101%22%20x1%3D%2237.4414%22%20y1%3D%2276.8587%22%20x2%3D%22146.891%22%20y2%3D%2276.8587%22%20gradientUnits%3D%22userSpaceOnUse%22%3E%3Cstop%20stop-color%3D%22%23FDEC9F%22%2F%3E%3Cstop%20offset%3D%220.08%22%20stop-color%3D%22%23E4D8A4%22%2F%3E%3Cstop%20offset%3D%220.24%22%20stop-color%3D%22%23A4A6B2%22%2F%3E%3Cstop%20offset%3D%220.47%22%20stop-color%3D%22%233F57C8%22%2F%3E%3Cstop%20offset%3D%220.61%22%20stop-color%3D%22%230025D5%22%2F%3E%3Cstop%20offset%3D%221%22%2F%3E%3C%2FlinearGradient%3E%3ClinearGradient%20id%3D%22paint3_linear_699_5101%22%20x1%3D%2215.0596%22%20y1%3D%22103.18%22%20x2%3D%22155.01%22%20y2%3D%22103.18%22%20gradientUnits%3D%22userSpaceOnUse%22%3E%3Cstop%20offset%3D%220.05%22%20stop-color%3D%22%2362A5FF%22%2F%3E%3Cstop%20offset%3D%220.45%22%20stop-color%3D%22%231032D1%22%2F%3E%3Cstop%20offset%3D%221%22%2F%3E%3C%2FlinearGradient%3E%3ClinearGradient%20id%3D%22paint4_linear_699_5101%22%20x1%3D%22628.741%22%20y1%3D%223244.93%22%20x2%3D%22797.782%22%20y2%3D%223247.12%22%20gradientUnits%3D%22userSpaceOnUse%22%3E%3Cstop%20stop-color%3D%22%23FFD4B2%22%2F%3E%3Cstop%20offset%3D%220.36%22%20stop-color%3D%22%239ACEB7%22%2F%3E%3Cstop%20offset%3D%220.67%22%20stop-color%3D%22%2347C8BB%22%2F%3E%3Cstop%20offset%3D%220.89%22%20stop-color%3D%22%2314C5BE%22%2F%3E%3Cstop%20offset%3D%221%22%20stop-color%3D%22%2300C4BF%22%2F%3E%3C%2FlinearGradient%3E%3ClinearGradient%20id%3D%22paint5_linear_699_5101%22%20x1%3D%2224.5987%22%20y1%3D%2282.3783%22%20x2%3D%2272.5834%22%20y2%3D%2282.3783%22%20gradientUnits%3D%22userSpaceOnUse%22%3E%3Cstop%20stop-color%3D%22%2300FECF%22%2F%3E%3Cstop%20offset%3D%220.08%22%20stop-color%3D%22%2300E5D0%22%2F%3E%3Cstop%20offset%3D%220.25%22%20stop-color%3D%22%2300A5D1%22%2F%3E%3Cstop%20offset%3D%220.49%22%20stop-color%3D%22%230040D4%22%2F%3E%3Cstop%20offset%3D%220.56%22%20stop-color%3D%22%230025D5%22%2F%3E%3C%2FlinearGradient%3E%3ClinearGradient%20id%3D%22paint6_linear_699_5101%22%20x1%3D%2270.9573%22%20y1%3D%2252.5952%22%20x2%3D%22189.069%22%20y2%3D%2250.4576%22%20gradientUnits%3D%22userSpaceOnUse%22%3E%3Cstop%20stop-color%3D%22%2300FECF%22%2F%3E%3Cstop%20offset%3D%220.05%22%20stop-color%3D%22%2300E5D0%22%2F%3E%3Cstop%20offset%3D%220.15%22%20stop-color%3D%22%2300A5D1%22%2F%3E%3Cstop%20offset%3D%220.29%22%20stop-color%3D%22%230040D4%22%2F%3E%3Cstop%20offset%3D%220.33%22%20stop-color%3D%22%230025D5%22%2F%3E%3C%2FlinearGradient%3E%3ClinearGradient%20id%3D%22paint7_linear_699_5101%22%20x1%3D%2227.1191%22%20y1%3D%2227.8689%22%20x2%3D%22173.642%22%20y2%3D%2227.8689%22%20gradientUnits%3D%22userSpaceOnUse%22%3E%3Cstop%20stop-color%3D%22%23FFD4AF%22%2F%3E%3Cstop%20offset%3D%220.1%22%20stop-color%3D%22%23E6D5BA%22%2F%3E%3Cstop%20offset%3D%220.31%22%20stop-color%3D%22%23A7D6D5%22%2F%3E%3Cstop%20offset%3D%220.61%22%20stop-color%3D%22%2343D9FF%22%2F%3E%3Cstop%20offset%3D%220.63%22%20stop-color%3D%22%2337B1D0%22%2F%3E%3Cstop%20offset%3D%220.65%22%20stop-color%3D%22%232B8CA5%22%2F%3E%3Cstop%20offset%3D%220.67%22%20stop-color%3D%22%23216B7D%22%2F%3E%3Cstop%20offset%3D%220.7%22%20stop-color%3D%22%23184E5B%22%2F%3E%3Cstop%20offset%3D%220.72%22%20stop-color%3D%22%2310353F%22%2F%3E%3Cstop%20offset%3D%220.75%22%20stop-color%3D%22%230A2228%22%2F%3E%3Cstop%20offset%3D%220.78%22%20stop-color%3D%22%23061316%22%2F%3E%3Cstop%20offset%3D%220.82%22%20stop-color%3D%22%23020809%22%2F%3E%3Cstop%20offset%3D%220.88%22%20stop-color%3D%22%23010202%22%2F%3E%3Cstop%20offset%3D%221%22%2F%3E%3C%2FlinearGradient%3E%3C%2Fdefs%3E%3C%2Fsvg%3E";
var SubWallet = class extends BaseDotsamaWallet {
  constructor() {
    super(...arguments);
    this.extensionName = "subwallet-js";
    this.title = "SubWallet";
    this.installUrl = "https://chrome.google.com/webstore/detail/subwallet/onhogfjeacnfoofkfgppdlbmlmnplgbn?hl=en&authuser=0";
    this.noExtensionMessage = "You can use any Polkadot compatible wallet but we recommend using Talisman";
    this.logo = {
      src: logo$1,
      alt: "Subwallet Logo"
    };
  }
};
var logo = "data:image/svg+xml,%3Csvg%20id%3D%22Layer_1%22%20data-name%3D%22Layer%201%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20xmlns%3Axlink%3D%22http%3A%2F%2Fwww.w3.org%2F1999%2Fxlink%22%20viewBox%3D%220%200%20120%20120%22%3E%3Cdefs%3E%3Cstyle%3E.cls-1%7Bfill-rule%3Aevenodd%3Bfill%3Aurl%28%23radial-gradient%29%3B%7D%3C%2Fstyle%3E%3CradialGradient%20id%3D%22radial-gradient%22%20cx%3D%22-164.98%22%20cy%3D%22566.58%22%20r%3D%221%22%20gradientTransform%3D%22matrix%28-15%2C%20136.5%2C%20135.36%2C%2014.88%2C%20-79032.62%2C%2014076.16%29%22%20gradientUnits%3D%22userSpaceOnUse%22%3E%3Cstop%20offset%3D%220%22%20stop-color%3D%22%23704bff%22%2F%3E%3Cstop%20offset%3D%220.21%22%20stop-color%3D%22%23c549ff%22%2F%3E%3C%2FradialGradient%3E%3C%2Fdefs%3E%3Ctitle%3Eenkrypt-logo-color%3C%2Ftitle%3E%3Cpath%20class%3D%22cls-1%22%20d%3D%22M0%2C23.09A23.09%2C23.09%2C0%2C0%2C1%2C23.09%2C0H120V17.72a16.6%2C16.6%2C0%2C0%2C1-16.6%2C16.6H57.84A23.09%2C23.09%2C0%2C0%2C0%2C34.75%2C57.41v5.87a23.1%2C23.1%2C0%2C0%2C0%2C23.09%2C23.1H103.4A16.6%2C16.6%2C0%2C0%2C1%2C120%2C103v17H23.09A23.09%2C23.09%2C0%2C0%2C1%2C0%2C96.91ZM58.84%2C44h46.73A14.43%2C14.43%2C0%2C0%2C1%2C120%2C58.4v3.89a14.43%2C14.43%2C0%2C0%2C1-14.43%2C14.43H58.84A14.44%2C14.44%2C0%2C0%2C1%2C44.4%2C62.29V58.4A14.44%2C14.44%2C0%2C0%2C1%2C58.84%2C44Z%22%2F%3E%3C%2Fsvg%3E";
var EnkryptWallet = class extends BaseDotsamaWallet {
  constructor() {
    super(...arguments);
    this.extensionName = "enkrypt";
    this.title = "Enkrypt";
    this.installUrl = "https://www.enkrypt.com/#overview";
    this.noExtensionMessage = "You can use any Polkadot compatible wallet but we recommend using Talisman";
    this.logo = {
      src: logo,
      alt: "Enkrypt Logo"
    };
  }
};
var supportedWallets = [
  new TalismanWallet(),
  new SubWallet(),
  new PolkadotjsWallet(),
  new EnkryptWallet()
];

// src/core/substrate/registry/tokens.ts
var chainTokenSymbol = (api) => api.registry.chainTokens[0];

// src/utils/events/system/ExtrinsicFailed.ts
var isExtrinsicFailedEvent = ({ event }) => event.section === "system" && event.method === "ExtrinsicFailed";
var asExtrinsicFailedEvent = (event) => {
  if (!isExtrinsicFailedEvent(event))
    return;
  return { ...event, name: formatEventName(event) };
};
var formatExtrinsicFailed = (event, api) => {
  if (!api)
    return;
  const ev = asExtrinsicFailedEvent(event);
  if (!ev)
    return;
  return decodeError(ev.event?.data?.dispatchError, { contract: { api } });
};

// src/utils/helpers/NOOP.ts
var NOOP = () => void 0;

// src/utils/helpers/pseudoRandomU8a.ts
var pseudoRandomU8a = (t = 21) => crypto.getRandomValues(new Uint8Array(t));

// src/utils/helpers/encodeSalt.ts
var encodeSalt = (salt = pseudoRandomU8a(32)) => {
  const encoder2 = new TextEncoder();
  return typeof salt === "string" ? encoder2.encode(salt) : salt;
};

// src/utils/helpers/formatFileName.ts
var formatFileName = ({ name, size }) => size !== void 0 ? `${name} (${(size / 1e3).toFixed(2)}kb)` : name;

// src/utils/helpers/getExpiredItem.ts
var FIVE_SECONDS = 5e3;
function getExpiredItem(items, expirationPeriod) {
  if (expirationPeriod === 0)
    return [];
  const timeFromCreation = (creationTime) => Date.now() - creationTime;
  return items.filter(
    (item) => timeFromCreation(item.createdAt) >= (expirationPeriod || FIVE_SECONDS)
  );
}

// src/utils/helpers/isTxCancelledError.ts
var isTxCancelledError = (e) => e?.toString() === "Error: Cancelled" || e?.toString() === "Error: Canceled";

// src/utils/helpers/isValidHash.ts
var isValidHash = (value, length = 64) => {
  if (!value)
    return false;
  return new RegExp(`^0x[0-9a-f]{${length}}$`).test(value);
};

// src/utils/helpers/parseUnits/parseUnits.ts
var stringNumberToBN = (valWithCommas) => {
  const v = valWithCommas.split(",").join("");
  return new import_bn.default(v);
};
var planckToDecimal = (amount, api) => {
  const decimals = api?.registry.chainDecimals[0];
  if (!decimals || !amount)
    return;
  if (decimals === void 0 || amount === void 0)
    return;
  const base = new import_bn.default(10).pow(new import_bn.default(decimals));
  const { div, mod } = new import_bn.default(amount).divmod(base);
  return parseFloat(
    `${div.toString()}.${mod.toString().padStart(decimals, "0")}`
  );
};
var planckToDecimalFormatted = (amount, api, options) => {
  const decimalAmount = planckToDecimal(amount, api);
  if (decimalAmount === void 0 || !api)
    return;
  const formattedVal = options?.decimals === void 0 ? decimalAmount.toString() : decimalAmount.toFixed(options?.decimals).toString();
  return `${formattedVal} ${chainTokenSymbol(api)}`;
};
var decimalToPlanck = (amount, api) => {
  const decimals = api?.registry.chainDecimals[0];
  if (!decimals)
    return;
  const convertedValue = BigInt(amount * 10 ** decimals);
  return convertedValue;
};

// src/utils/helpers/pseudoRandomHex.ts
var pseudoRandomHex = (size = 64) => `0x${[...Array(size)].map(() => Math.floor(Math.random() * 16).toString(16)).join("")}`;

// src/utils/helpers/pseudoRandomId.ts
var import_nanoid = require("nanoid");
var pseudoRandomId = (t = 21) => (0, import_nanoid.nanoid)(t);

// src/utils/helpers/unixMilliToDate.ts
var unixMilliToDate = (unixInMilliSeconds) => unixInMilliSeconds ? new Date(unixInMilliSeconds) : void 0;

// src/utils/pick/pickCallInfo.ts
function pickCallInfo(decoded) {
  if (!decoded?.ok)
    return;
  const { gasRequired, gasConsumed, storageDeposit } = decoded.value;
  return { gasRequired, gasConsumed, storageDeposit };
}

// src/utils/pick/pickDecoded.ts
function pickDecoded(decoded) {
  return decoded?.ok ? decoded.value.decoded : void 0;
}

// src/utils/pick/pickDecodedError.ts
function pickDecodedError(call, contract, moduleMessages, defaultMessage) {
  const { result } = call || {};
  if (!result || result?.ok)
    return;
  return decodeError(result.error, contract, moduleMessages, defaultMessage);
}

// src/utils/pick/pickError.ts
function pickError(decoded) {
  return decoded?.ok ? void 0 : decoded?.error;
}

// src/utils/pick/pickResultErr.ts
function pickResultErr(decoded) {
  return decoded?.ok ? decoded.value.decoded?.Err : void 0;
}

// src/utils/pick/pickResultOk.ts
function pickResultOk(decoded) {
  return decoded?.ok ? decoded.value.decoded?.Ok : void 0;
}

// src/utils/pick/pickTxInfo.ts
function pickTxInfo(result) {
  if (!result?.ok)
    return;
  const { gasRequired, gasConsumed, storageDeposit, partialFee } = result.value;
  return { gasRequired, gasConsumed, storageDeposit, partialFee };
}

// src/utils/substrate/bnToBalance.ts
var bnToBalance = (api, bn) => api?.createType("Balance", bn);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  BN,
  BN_BILLION,
  BN_EIGHT,
  BN_FIVE,
  BN_FOUR,
  BN_HUNDRED,
  BN_MAX_INTEGER,
  BN_MILLION,
  BN_NINE,
  BN_ONE,
  BN_QUINTILL,
  BN_SEVEN,
  BN_SIX,
  BN_SQRT_MAX_INTEGER,
  BN_TEN,
  BN_THOUSAND,
  BN_THREE,
  BN_TWO,
  BN_ZERO,
  MetadataError,
  NOOP,
  U8A_WRAP_ETHEREUM,
  U8A_WRAP_POSTFIX,
  U8A_WRAP_PREFIX,
  _0n,
  _1000n,
  _100n,
  _10n,
  _1Bn,
  _1Mn,
  _1Qn,
  _1n,
  _2n,
  _2pow53n,
  _3n,
  _4n,
  _5n,
  _6n,
  _7n,
  _8n,
  _9n,
  _sqrt2pow53n,
  arrayChunk,
  arrayFilter,
  arrayFlatten,
  arrayRange,
  arrayShuffle,
  arrayUnzip,
  arrayZip,
  asContractInstantiatedEvent,
  asExtrinsicFailedEvent,
  assert,
  assertReturn,
  assertUnreachable,
  bnFromHex,
  bnMax,
  bnMin,
  bnSqrt,
  bnToBalance,
  bnToBn,
  bnToHex,
  bnToU8a,
  bufferToU8a,
  calcSi,
  compactAddLength,
  compactFromU8a,
  compactFromU8aLim,
  compactStripLength,
  compactToU8a,
  decimalToPlanck,
  detectPackage,
  encodeSalt,
  extractTime,
  findSi,
  floatToU8a,
  formatBalance,
  formatDate,
  formatDecimal,
  formatElapsed,
  formatEventName,
  formatExtrinsicFailed,
  formatFileName,
  formatNumber,
  getExpiredItem,
  hasAny,
  hasBigInt,
  hasBuffer,
  hasCjs,
  hasDirname,
  hasEsm,
  hasProcess,
  hasWasm,
  hexAddPrefix,
  hexFixLength,
  hexHasPrefix,
  hexStripPrefix,
  hexToBigInt,
  hexToBn,
  hexToNumber,
  hexToString,
  hexToU8a,
  identity,
  isArray,
  isAscii,
  isBigInt,
  isBn,
  isBoolean,
  isBroadcast,
  isBuffer,
  isChildClass,
  isClass,
  isCodec,
  isCompact,
  isContractInstantiatedEvent,
  isDropped,
  isDryRun,
  isError,
  isErrored,
  isExtrinsicFailedEvent,
  isFinalityTimeout,
  isFinalized,
  isFunction,
  isFuture,
  isHex,
  isInBlock,
  isInstanceOf,
  isInvalid,
  isIp,
  isJsonObject,
  isNone,
  isNull,
  isNumber,
  isObject,
  isObservable,
  isPendingSignature,
  isPromise,
  isReady,
  isRetracted,
  isString,
  isTestChain,
  isToBigInt,
  isToBn,
  isTxCancelledError,
  isU8a,
  isUndefined,
  isUsurped,
  isUtf8,
  isValidHash,
  isWasm,
  lazyMethod,
  lazyMethods,
  logger,
  loggerFormat,
  memoize,
  nMax,
  nMin,
  nSqrt,
  nToBigInt,
  nToHex,
  nToU8a,
  nextTick,
  noop,
  numberToHex,
  numberToU8a,
  objectClear,
  objectCopy,
  objectEntries,
  objectKeys,
  objectProperties,
  objectProperty,
  objectSpread,
  objectValues,
  packageInfo,
  pickCallInfo,
  pickDecoded,
  pickDecodedError,
  pickError,
  pickResultErr,
  pickResultOk,
  pickTxInfo,
  planckToDecimal,
  planckToDecimalFormatted,
  promisify,
  pseudoRandomHex,
  pseudoRandomId,
  pseudoRandomU8a,
  shouldDisable,
  shouldDisableStrict,
  stringCamelCase,
  stringLowerFirst,
  stringNumberToBN,
  stringPascalCase,
  stringShorten,
  stringToHex,
  stringToU8a,
  stringUpperFirst,
  stringify,
  toBasicMetadata,
  toJSType,
  toMessageParams,
  u8aCmp,
  u8aConcat,
  u8aConcatStrict,
  u8aEmpty,
  u8aEq,
  u8aFixLength,
  u8aIsWrapped,
  u8aSorted,
  u8aToBigInt,
  u8aToBn,
  u8aToBuffer,
  u8aToFloat,
  u8aToHex,
  u8aToNumber,
  u8aToString,
  u8aToU8a,
  u8aUnwrapBytes,
  u8aWrapBytes,
  unixMilliToDate,
  validateMetadata
});
