// useink/dist/utils.mjs
import {
  BN_BILLION,
  BN_EIGHT,
  BN_FIVE,
  BN_FOUR,
  BN_HUNDRED,
  BN_MAX_INTEGER,
  BN_MILLION,
  BN_NINE,
  BN_ONE,
  BN_QUINTILL,
  BN_SEVEN,
  BN_SIX,
  BN_SQRT_MAX_INTEGER,
  BN_TEN,
  BN_THOUSAND,
  BN_THREE,
  BN_TWO,
  BN_ZERO,
  MetadataError,
  NOOP,
  U8A_WRAP_ETHEREUM,
  U8A_WRAP_POSTFIX,
  U8A_WRAP_PREFIX,
  _0n,
  _1000n,
  _100n,
  _10n,
  _1Bn,
  _1Mn,
  _1Qn,
  _1n,
  _2n,
  _2pow53n,
  _3n,
  _4n,
  _5n,
  _6n,
  _7n,
  _8n,
  _9n,
  _sqrt2pow53n,
  arrayChunk,
  arrayFilter,
  arrayFlatten,
  arrayRange,
  arrayShuffle,
  arrayUnzip,
  arrayZip,
  asContractInstantiatedEvent,
  asExtrinsicFailedEvent,
  assert,
  assertReturn,
  assertUnreachable,
  bnMax,
  bnMin,
  bnSqrt,
  bnToBalance,
  bnToBn,
  bnToHex,
  bnToU8a,
  bufferToU8a,
  calcSi,
  compactAddLength,
  compactFromU8a,
  compactFromU8aLim,
  compactStripLength,
  compactToU8a,
  decimalToPlanck,
  detectPackage,
  encodeSalt,
  extractTime,
  findSi,
  floatToU8a,
  formatBalance,
  formatDate,
  formatDecimal,
  formatElapsed,
  formatEventName,
  formatExtrinsicFailed,
  formatFileName,
  formatNumber,
  getExpiredItem,
  hasAny,
  hasBigInt,
  hasBuffer,
  hasCjs,
  hasDirname,
  hasEsm,
  hasProcess,
  hasWasm,
  hexAddPrefix,
  hexFixLength,
  hexHasPrefix,
  hexStripPrefix,
  hexToBigInt,
  hexToBn,
  hexToNumber,
  hexToString,
  hexToU8a,
  identity,
  import_bn,
  isArray,
  isAscii,
  isBigInt,
  isBn,
  isBoolean,
  isBroadcast,
  isBuffer,
  isChildClass,
  isClass,
  isCodec,
  isCompact,
  isContractInstantiatedEvent,
  isDropped,
  isDryRun,
  isError,
  isErrored,
  isExtrinsicFailedEvent,
  isFinalityTimeout,
  isFinalized,
  isFunction,
  isFuture,
  isHex,
  isInBlock,
  isInstanceOf,
  isInvalid,
  isIp,
  isJsonObject,
  isNone,
  isNull,
  isNumber,
  isObject,
  isObservable,
  isPendingSignature,
  isPromise,
  isReady,
  isRetracted,
  isString,
  isTestChain,
  isToBigInt,
  isToBn,
  isTxCancelledError,
  isU8a,
  isUndefined,
  isUsurped,
  isUtf8,
  isValidHash,
  isWasm,
  lazyMethod,
  lazyMethods,
  logger,
  loggerFormat,
  memoize,
  nMax,
  nMin,
  nSqrt,
  nToBigInt,
  nToHex,
  nToU8a,
  nextTick,
  noop,
  numberToHex,
  numberToU8a,
  objectClear,
  objectCopy,
  objectEntries,
  objectKeys,
  objectProperties,
  objectProperty,
  objectSpread,
  objectValues,
  packageInfo,
  pickCallInfo,
  pickDecoded,
  pickDecodedError,
  pickError,
  pickResultErr,
  pickResultOk,
  pickTxInfo,
  planckToDecimal,
  planckToDecimalFormatted,
  promisify,
  pseudoRandomHex,
  pseudoRandomId,
  pseudoRandomU8a,
  shouldDisable,
  shouldDisableStrict,
  stringCamelCase,
  stringLowerFirst,
  stringNumberToBN,
  stringPascalCase,
  stringShorten,
  stringToHex,
  stringToU8a,
  stringUpperFirst,
  stringify,
  toBasicMetadata,
  toJSType,
  toMessageParams,
  u8aCmp,
  u8aConcat,
  u8aConcatStrict,
  u8aEmpty,
  u8aEq,
  u8aFixLength,
  u8aIsWrapped,
  u8aSorted,
  u8aToBigInt,
  u8aToBn,
  u8aToBuffer,
  u8aToFloat,
  u8aToHex,
  u8aToNumber,
  u8aToString,
  u8aToU8a,
  u8aUnwrapBytes,
  u8aWrapBytes,
  unixMilliToDate,
  validateMetadata
} from "./chunk-K4CFJTMJ.mjs";
import "./chunk-UT43TYHK.mjs";
var export_BN = import_bn.default;
export {
  export_BN as BN,
  BN_BILLION,
  BN_EIGHT,
  BN_FIVE,
  BN_FOUR,
  BN_HUNDRED,
  BN_MAX_INTEGER,
  BN_MILLION,
  BN_NINE,
  BN_ONE,
  BN_QUINTILL,
  BN_SEVEN,
  BN_SIX,
  BN_SQRT_MAX_INTEGER,
  BN_TEN,
  BN_THOUSAND,
  BN_THREE,
  BN_TWO,
  BN_ZERO,
  MetadataError,
  NOOP,
  U8A_WRAP_ETHEREUM,
  U8A_WRAP_POSTFIX,
  U8A_WRAP_PREFIX,
  _0n,
  _1000n,
  _100n,
  _10n,
  _1Bn,
  _1Mn,
  _1Qn,
  _1n,
  _2n,
  _2pow53n,
  _3n,
  _4n,
  _5n,
  _6n,
  _7n,
  _8n,
  _9n,
  _sqrt2pow53n,
  arrayChunk,
  arrayFilter,
  arrayFlatten,
  arrayRange,
  arrayShuffle,
  arrayUnzip,
  arrayZip,
  asContractInstantiatedEvent,
  asExtrinsicFailedEvent,
  assert,
  assertReturn,
  assertUnreachable,
  hexToBn as bnFromHex,
  bnMax,
  bnMin,
  bnSqrt,
  bnToBalance,
  bnToBn,
  bnToHex,
  bnToU8a,
  bufferToU8a,
  calcSi,
  compactAddLength,
  compactFromU8a,
  compactFromU8aLim,
  compactStripLength,
  compactToU8a,
  decimalToPlanck,
  detectPackage,
  encodeSalt,
  extractTime,
  findSi,
  floatToU8a,
  formatBalance,
  formatDate,
  formatDecimal,
  formatElapsed,
  formatEventName,
  formatExtrinsicFailed,
  formatFileName,
  formatNumber,
  getExpiredItem,
  hasAny,
  hasBigInt,
  hasBuffer,
  hasCjs,
  hasDirname,
  hasEsm,
  hasProcess,
  hasWasm,
  hexAddPrefix,
  hexFixLength,
  hexHasPrefix,
  hexStripPrefix,
  hexToBigInt,
  hexToBn,
  hexToNumber,
  hexToString,
  hexToU8a,
  identity,
  isArray,
  isAscii,
  isBigInt,
  isBn,
  isBoolean,
  isBroadcast,
  isBuffer,
  isChildClass,
  isClass,
  isCodec,
  isCompact,
  isContractInstantiatedEvent,
  isDropped,
  isDryRun,
  isError,
  isErrored,
  isExtrinsicFailedEvent,
  isFinalityTimeout,
  isFinalized,
  isFunction,
  isFuture,
  isHex,
  isInBlock,
  isInstanceOf,
  isInvalid,
  isIp,
  isJsonObject,
  isNone,
  isNull,
  isNumber,
  isObject,
  isObservable,
  isPendingSignature,
  isPromise,
  isReady,
  isRetracted,
  isString,
  isTestChain,
  isToBigInt,
  isToBn,
  isTxCancelledError,
  isU8a,
  isUndefined,
  isUsurped,
  isUtf8,
  isValidHash,
  isWasm,
  lazyMethod,
  lazyMethods,
  logger,
  loggerFormat,
  memoize,
  nMax,
  nMin,
  nSqrt,
  nToBigInt,
  nToHex,
  nToU8a,
  nextTick,
  noop,
  numberToHex,
  numberToU8a,
  objectClear,
  objectCopy,
  objectEntries,
  objectKeys,
  objectProperties,
  objectProperty,
  objectSpread,
  objectValues,
  packageInfo,
  pickCallInfo,
  pickDecoded,
  pickDecodedError,
  pickError,
  pickResultErr,
  pickResultOk,
  pickTxInfo,
  planckToDecimal,
  planckToDecimalFormatted,
  promisify,
  pseudoRandomHex,
  pseudoRandomId,
  pseudoRandomU8a,
  shouldDisable,
  shouldDisableStrict,
  stringCamelCase,
  stringLowerFirst,
  stringNumberToBN,
  stringPascalCase,
  stringShorten,
  stringToHex,
  stringToU8a,
  stringUpperFirst,
  stringify,
  toBasicMetadata,
  toJSType,
  toMessageParams,
  u8aCmp,
  u8aConcat,
  u8aConcatStrict,
  u8aEmpty,
  u8aEq,
  u8aFixLength,
  u8aIsWrapped,
  u8aSorted,
  u8aToBigInt,
  u8aToBn,
  u8aToBuffer,
  u8aToFloat,
  u8aToHex,
  u8aToNumber,
  u8aToString,
  u8aToU8a,
  u8aUnwrapBytes,
  u8aWrapBytes,
  unixMilliToDate,
  validateMetadata
};
