// @polkadot/api-derive/cjs/society/members.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.members = exports._members = void 0;
const rxjs_1 = require("rxjs");
const index_js_1 = require("../util/index.js");
function _members(instanceId, api) {
    return (0, index_js_1.memo)(instanceId, (accountIds) => (0, rxjs_1.combineLatest)([
        (0, rxjs_1.of)(accountIds),
        api.query.society.payouts.multi(accountIds),
        api.query.society.strikes.multi(accountIds),
        api.query.society.defenderVotes.multi(accountIds),
        api.query.society.suspendedMembers.multi(accountIds),
        api.query.society.vouching.multi(accountIds)
    ]).pipe((0, rxjs_1.map)(([accountIds, payouts, strikes, defenderVotes, suspended, vouching]) => accountIds.map((accountId, index) => ({
        accountId,
        isDefenderVoter: defenderVotes[index].isSome,
        isSuspended: suspended[index].isTrue,
        payouts: payouts[index],
        strikes: strikes[index],
        vote: defenderVotes[index].unwrapOr(undefined),
        vouching: vouching[index].unwrapOr(undefined)
    })))));
}
exports._members = _members;
/**
 * @description Get the member info for a society
 */
function members(instanceId, api) {
    return (0, index_js_1.memo)(instanceId, () => api.query.society.members().pipe((0, rxjs_1.switchMap)((members) => api.derive.society._members(members))));
}
exports.members = members;
